---
title: "2024_NEW_Oculina_scRNAseq"
author: "Maria Valadez Ingersoll"
date: "2024-09-10"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Set working directory, load packages
```{r}
setwd("/projectnb/coral/MVI_Oculina/Oculina_sc_manuscript/SingleCell")

library(stringr)
library(deldir)
library(Seurat)
library(tidyverse)
library(openxlsx)
library(MAST)
library(dplyr)
library(microseq)
library(sctransform)
library(RColorBrewer)
library(ggplot2)
library(glmGamPoi)
library(gplots)
library(ggplot2)
library(pheatmap)
library(superheat)
library(purrr)
library(tibble)
library(DESeq2)
library(ggrepel)
library(cowplot)
library(devtools)
```

The proper reference files for astrangia are in the directory /projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/new_Ap_ref
The gtf is called apoculata_collapsed.gtf and the genome is called apoculata.genome.fasta

Concat apoc genome and apoc gtf to the algae genome and gtf
```{bash}
cat ./new_Ap_ref/apoculata.genome.fasta algae_reftranscriptome.fasta > new_holobiont_genome.fasta

cellranger mkgtf ./new_Ap_ref/apoculata_collapsed.gtf ./new_Ap_ref/apoculata_collapsed_filt.gtf --attribute=gene_biotype:protein_coding

cat ./new_Ap_ref/apoculata_collapsed_filt.gtf algae_ref5_filt.gtf > new_holobiont_gtf.gtf
```

Mkref for reference of concatenated host and sym files


Skipping to line 200 R analysis for now

Make reference files; might have to think about sym reads in here, but can come back

```{r}
# load gene annotations
annotations <- read.delim("./files/apoculata_GeneAnnotation_combined.txt", sep="\t", header=TRUE)
head(annotations)

#seq2iso <- annotations %>%
 # mutate(gene = gsub("evm.model.", "EVM%20prediction%20", Protein_ID)) %>%
 # select(Protein_ID, gene)
seq2iso <- annotations %>%
  mutate(gene = gsub("model", "TU", Protein_ID)) %>%
  mutate(gene=gsub("_", ".", gene)) %>%
  select(Protein_ID, gene)
head(seq2iso)
write.table(seq2iso, file="./files/astrangia_seq2iso_new.txt", quote=F, sep="\t", row.names=FALSE)

gene = annotations %>%
  mutate(gene_symbol = gsub("_.*", "", Swiss.Prot_Hit)) %>%
  mutate(gene_symbol = gsub(".*[|]", "", gene_symbol))%>%
       mutate(swissprot_symbol = gsub(" OS=.*", "", Swiss.Prot_Description)) %>%
  mutate(trembl_symbol = gsub(" OS=.*", "", Trembl_Description)) %>%
  mutate(ncbi_symbol = gsub(" OS=.*", "", NCBI_Description)) %>%
  select(Protein_ID, gene_symbol, swissprot_symbol, trembl_symbol, ncbi_symbol)
head(gene)

seq2iso2gene <- seq2iso %>%
  left_join(gene)
head(seq2iso2gene)
seq2iso2gene <- 
    transform(
        seq2iso2gene ,
        gene_symbol =
            ifelse( is.na(gene_symbol), gene , gene_symbol)) %>%
  mutate(gene_symbol = make.names(gene_symbol, unique = TRUE))
head(seq2iso2gene)
write.table(seq2iso2gene, file="./files/astrangia_seq2iso2gene_new.txt", quote=F, sep="\t", row.names=FALSE)

gene_symbol_only <- seq2iso2gene[,c(2,3)] %>%
  column_to_rownames(var="gene")
gene2gene_symbol <- seq2iso2gene[,c(2,3)]
```

Prepare data from cellranger
```{r}
#Set up your experiment
results_dir <- "catcat_analysis"
proj_name_lead <- "Oculina_catcat"

dir.create(results_dir, showWarnings = F)
proj_name <- paste0(results_dir, "/", proj_name_lead)

#these are not on the github or in the Rproject
F8.host.data <- Read10X(data.dir <- "/projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/A3_cat_cat_new/outs/filtered_feature_bc_matrix/")
F1.host.data <- Read10X(data.dir <- "/projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/S3_cat_cat_new/outs/filtered_feature_bc_matrix/")

# check for duplicated barcodes(column names) within a dataset
any(duplicated(x = colnames(x = F8.host.data)))
#FALSE
any(duplicated(x = colnames(x = F1.host.data)))
#FALSE

# Rename columns to avoid trouble when merging data sets
colnames(x = F8.host.data) <- paste('F8', colnames(x = F8.host.data), sep = '_')
colnames(x = F1.host.data) <- paste('F1', colnames(x = F1.host.data), sep = '_')
rownames(x = F8.host.data) <- gsub("_", "-", rownames(F8.host.data))
head(rownames(F8.host.data))
rownames(x = F1.host.data) <- gsub("_", "-", rownames(F1.host.data))
head(rownames(F1.host.data))

#rename rownames
length(gene2gene_symbol$gene_symbol)
#45867
length(rownames(F8.host.data))
#75579 bc of symbiont gtf input
length(rownames(F1.host.data))
#75579 bc of symbiont gtf input

#replace the "merged" from F8_rownames with "" 
#------->>>>>>> THIS MIGHT CHANGE
####---->>>> no longer doing this, now instead of the "merged" the isoforms are in the format: "evm.TU.Ap11.1003-evm.TU.Ap11.1005"
F8.host.data_names <- F8.host.data
F8_host_rownames <- as.data.frame(rownames(F8.host.data_names))
colnames(F8_host_rownames) <-paste("X")
#F8_host_rownames$X <- gsub("MERGED%3A%20", "", F8_host_rownames$X)
F8_host_gene2gene_symbol <- F8_host_rownames %>%
  left_join(gene2gene_symbol, by=c("X"="gene"))
F8_host_gene2gene_symbol <- 
    transform(
        F8_host_gene2gene_symbol ,
        gene_symbol =
            ifelse( is.na(gene_symbol), X, gene_symbol)) %>%
  mutate(gene_symbol = make.names(gene_symbol, unique = TRUE))
rownames(x=F8.host.data_names) <- F8_host_gene2gene_symbol$gene_symbol
head(rownames(F8.host.data_names))
head(rownames(F8.host.data))

#do the same for F1
F1.host.data_names <- F1.host.data
F1_host_rownames <- as.data.frame(rownames(F1.host.data_names))
colnames(F1_host_rownames) <-paste("X")
#F1_host_rownames$X <- gsub("MERGED%3A%20", "", F1_host_rownames$X)
F1_host_gene2gene_symbol <- F1_host_rownames %>%
  left_join(gene2gene_symbol, by=c("X"="gene"))
F1_host_gene2gene_symbol <- 
    transform(
        F1_host_gene2gene_symbol ,
        gene_symbol =
            ifelse( is.na(gene_symbol), X, gene_symbol)) %>%
  mutate(gene_symbol = make.names(gene_symbol, unique = TRUE))
rownames(x=F1.host.data_names) <- F1_host_gene2gene_symbol$gene_symbol
head(rownames(F1.host.data_names))
head(rownames(F1.host.data))

### Initialize the Seurat object with the raw, non-normalized data

F8 <- CreateSeuratObject(counts = F8.host.data_names, 
                             min.cells = 3, 
                             min.features = 200, 
                             project = "F8_host")
F1 <- CreateSeuratObject(counts = F1.host.data_names, 
                             min.cells = 3, 
                             min.features = 200, 
                             project = "F1_host")

### add metadata to objects

# add individual
F8[["individual"]] <- "A_2F8"
F8@meta.data$individual <- as.factor(F8@meta.data$individual)
F1[["individual"]] <- "S_2F1"
F1@meta.data$individual <- as.factor(F1@meta.data$individual)

# add symstate
F8[["symstate"]] <- "apo"
F8@meta.data$symstate <- as.factor(F8@meta.data$symstate)
F1[["symstate"]] <- "sym"
F1@meta.data$symstate <- as.factor(F1@meta.data$symstate)

# add genet
F8[["genet"]] <- "F"
F8@meta.data$genet <- as.factor(F8@meta.data$genet)
F1[["genet"]] <- "F"
F1@meta.data$genet <- as.factor(F1@meta.data$genet)

str(F8@meta.data)
str(F1@meta.data)

F8
#24844 features across 3828 samples within 1 assay 
#Active assay: RNA (24844 features, 0 variable features)

F1
#29251 features across 3151 samples within 1 assay 
#Active assay: RNA (29251 features, 0 variable features)
```

Quality control and filtering
```{r}
obj.list <- list(F8, F1)

rownames(obj.list[[1]]) %>% as.data.frame() %>% View()

for (i in 1:length(obj.list)) {
  obj.list[[i]] <- PercentageFeatureSet(obj.list[[i]], pattern = "^MT-", col.name = 'percent.mt')
  obj.list[[i]][["per.mito.bin"]] <- ifelse(obj.list[[i]][["percent.mt"]] > 10, "> 10%", ifelse(obj.list[[i]][["percent.mt"]] > 5, "5%-10%", ifelse(obj.list[[i]][["percent.mt"]] > 2.5, "2.5%-5%", "< 2.5%")))
}

plts_qc <- lapply(obj.list, function(x) VlnPlot(x, 
                                             features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), 
                                             group.by = "individual",
                                             pt.size = .25,
                                             ncol = 3))

patchwork::wrap_plots(plts_qc, ncol = 1)
```

Perform integration of the two seurat objects, according to seurat v5
```{r}
F8_new <- obj.list[[1]]
F1_new <- obj.list[[2]]
obj.list_merge <- merge(F8_new, F1_new)
obj.list_merge
#31026 features across 6979 samples within 1 assay 
#Active assay: RNA (31026 features, 0 variable features)
# 2 layers present: counts.F8_host, counts.F1_host


#following the "Perform analysis without integration" first:
# run standard anlaysis workflow
merge <- NormalizeData(obj.list_merge)
merge <- FindVariableFeatures(merge)
#merge <- ScaleData(merge, features = rownames(obj.list_merge@assays$RNA@features))
merge<- ScaleData(merge, features = features_to_scale)
merge <- RunPCA(merge)
merge <- FindNeighbors(merge, dims = 1:30, reduction = "pca")
merge <- FindClusters(merge, resolution = 2, cluster.name = "unintegrated_clusters")
#41 clusters; unintegrated and with a high res
#now 40 clusters

merge <- RunUMAP(merge, dims = 1:30, reduction = "pca", reduction.name = "umap.unintegrated")
DimPlot(merge, reduction = "umap.unintegrated", group.by = c("individual", "seurat_clusters"))
#the individuals overlap really well

#"Perform integration"
merge <- IntegrateLayers(object = merge, method = CCAIntegration, orig.reduction = "pca", new.reduction = "integrated.cca",
    verbose = TRUE)
# re-join layers after integration
merge[["RNA"]] <- JoinLayers(merge[["RNA"]])

merge <- FindNeighbors(merge, reduction = "integrated.cca", dims = 1:30)
merge <- FindClusters(merge, resolution = 0.5)
#28 clusters at res of 0.5
#22 clusters at res of 0.25

#now 27

merge <- RunUMAP(merge, dims = 1:30, reduction = "integrated.cca")

# Visualization
plt1merge <- DimPlot(merge, reduction = "umap", label = T, shuffle=TRUE)
plt2merge <- DimPlot(merge, reduction = "umap", group.by = "individual", shuffle=TRUE)
plt3merge <- DimPlot(merge, reduction = "umap", group.by = "symstate", shuffle=TRUE)
#saved plt1_plt3_merge_091124 as landscape 12x6
#new saved as plt1_plt3_merge_091624
#keeping with res of 0.5 for now, can merge clusters later

sym_cell_feat_plot_new <- DimPlot(merge, reduction = "umap", cells.highlight = alga_cells, cols.highlight=c("black"))
#saved as sym_cells_umap_091624 as 6x6

sym_feat_plot1 <- FeaturePlot(merge, features=c("Sym.152125.c5.g2.i4"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98))
sym_feat_plot2 <- FeaturePlot(merge, features=c("Sym.146798.c0.g2.i7"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98))
sym_feat_plot3 <- FeaturePlot(merge, features=c("Sym.111787.c1.g1.i6"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98))

#apo cells assigned to "algal-hosting" don't even express symbiont genes
#sym cells assigned to "algal-hosting" but outside of gderm cluster either don't express symbiont reads or express them at very low conc
#so those 45 sym cells identified in algal-hosting cell from before and now in gderm are the actual algal-hosting cells


################ trying to now remove the sym genes
counts <- GetAssayData(merge, assay = "RNA")
length(rownames(counts))
#31026
tail(counts)
counts_lim <- counts[1:24716,]
#counts_names <- rownames(counts)
#nrow(gene)
#45867
#counts_names <- counts_names[1:24716]
#counts_names_df <- as.data.frame(counts_names)

#get list of sym genes just to have
counts_sym <-rownames(counts[24717:31026,])

#remove cells with high percentage of sym genes that are messing shit up
merge_sym_percent <- PercentageFeatureSet(merge, features = counts_sym, col.name = "sym_percent")
#filter out cells that have over 50% symbiont reads
#merge_metadata <- merge_sym_percent@meta.data
#merge_metadata_highsym <- filter(merge_metadata, sym_percent > 50)
merge_nosym <- subset(merge_sym_percent, subset = sym_percent < 50)

sym_cell_feat_plot_nosym <- DimPlot(merge_nosym, reduction = "umap", cells.highlight = alga_cells, cols.highlight=c("black"))

# now remove sym genes all together
ultimate_merge <- subset(merge_nosym, features = rownames(counts_lim))
tail(rownames(ultimate_merge@assays$RNA))

```

```{r}
DefaultAssay(ultimate_merge) <- "RNA"
ultimate_merge@meta.data$seurat_clusters <- as.numeric(as.character(ultimate_merge@meta.data$seurat_clusters))+1
Idents(ultimate_merge)<- "seurat_clusters"
ultimate_merge$seurat_clusters
ultimate_merge@active.ident <- factor(x = ultimate_merge@active.ident, levels = 1:27)
ultimate_merge@active.ident

saveRDS(ultimate_merge, paste0(proj_name, '_ultimate_091724.rds'))
ultimate_merge <- readRDS(file.choose())
```

One method for finding marker genes, there are multiple ways of doing this
```{r}
###
markers <- FindAllMarkers(ultimate_merge,
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          return.thresh = 0.01,
                          latent.vars = c('percent.mt', 'nFeature_RNA'))



markers <- markers %>%
  filter(avg_log2FC > 0)
nrow(markers)
#new 24065

markers_fullnames <- markers
rownames(markers_fullnames) =NULL
head(markers_fullnames)
head(seq2iso2gene)
col_names <- c("p_val", "avg_log2FC", "pct.1", "pct.2", "p_val_adj", "cluster", "gene_symbol")
colnames(markers_fullnames) <- col_names
markers_fullnames <- markers_fullnames %>%
  left_join(seq2iso2gene) %>%
  write_csv('./files/Oculina_ultimate_markers_091824_logFC.5.csv')

markers_fullnames <- read.csv(file.choose())
head(markers_fullnames)

markers_strong <- markers_fullnames %>%
  filter(p_val_adj<0.05) %>%
  filter(pct.1 > 0.1)

nrow(markers_strong)
#new 7132
write_csv(markers_strong, "./files/SuppDataset_oculina_markers_strong_091824.csv")

#from the inclusion of the symbiont genome/gtf, strong symbiont genes really only in cluster 21, a few others scattered
```

###Now work on identifying your cell types

make "oculina" to manipulate around with and keep "merge" unmanipulated
```{r}
#oculina <- merge 
# oculina is the one that has the cluster 21 with the algal-hosting cells ONLY
#merge@meta.data
#merge@active.ident
#head(merge)

#oculina_new is the one that was PCA-ed and UMAP-ed without the symbiont reads then symbiont reads were removed
oculina_new <- ultimate_merge
```

First, look at symbiont genes
```{r}
#Sym.1 <- c("Sym.205261.c0.g1.i1") #marker for 21, sym only
#Sym.2 <- c("Sym.84646.c0.g2.i1") #marker for 1 but in a lot of cells and apo and sym; almost certainly not actually a sym read but a mismatch
#Sym.3 <- c("Sym.84646.c0.g2.i1") #same as sym.2
#VlnPlot(object =oculina, features = Sym.3, pt.size= 0,combine = F,assay = "RNA", split.by = "symstate") 
```

Use genes of interest identified in previous cnidarian datasets
```{r}
#TF highly expressed in spist cnidocytes
pax6 <- c("PAX6", "PAX6.1", "PAX6.2") #PAX6.2 marker for 3, 10, 14
six1 <- c("SIX1B", "SIX1B.1") #SIX1B marker for 13
six2 <- c("SO") #ns

#TF highly expressed in spist gastrodermis etc
foxc2 <- c("FXC2B") #spist gastro muscle; ns
deaf1 <- c("DEAF1") #spist gastro muscle; --> marker for 2, 4
erg <- c("ERG") #all spist gastros, --> marker for 1 and 4 but high in lots of cells
fli1 <- c("FLII") #same spist as erg; ns
etv6 <- c("ETV6","ETV6.1") #gastro algal hosting in spist, --> 6.1 is marker for 4
elf2 <- c("ELF2") #gastro algal hosting in spist, ns
elf4 <- c("ELF4", "ELF4.1", "ETS1A.3") #gastro algal hosting in spist --> marker for 4; 2, 4, 13, 17; 13, 17; ELF4.1 also in some others (3, 7, 14, 19)
spdef <- c("SPDEF") #gastro algal hosting in spist, marker for 4, 10, 13 (also in 12, 20)
usf2 <- c("USF2", "USF2.1") #in gastrodermis in spist, ns
otx1 <-c("OTX1", "OTX1B", "OTX1B.1") #gastro in spist, markers for 1 and 4
six4_6 <- c("SIX4", "SIX6") #ns

#TFs for neuron
gata2 <- c("GATA2") #marker for 17 and 26 (especially 26)
pax7 <- c("PAX7", "PAX7.1", "PAX7.2") #markers for 3, 10, 14, highest in 14
asc <- c("ASCC1", "ASCC2", "ASCC3") #ns
pou4 <- c("PO4F3") #marker for 13, 19, 25, 27; highest in 25 and 27
islet <- c("ICA69") #ns
tbx20 <- c("TBX20.2") #marker for 13, 17, 20; highest in 17

#TFs for Immune
irf1 <- c("IRF1") #marker gene for 15
irf2 <- c("IRF2", "IRF2.1", "IRF2.2", "IRF2.3") #markers 15
nfat5 <- c("NFAT5.4") #marker gene for 2 but high in everything
atf3 <- c("ATF3") #marker for 15
atf5 <- c("ATF5") #marker for 11 and 20 but in all cells

#TFs for spist epidermis 
pax2 <- c("PAX2", "PAX2A", "PAX2A.1", "PAX2A.2") #markers for 25 (pax2) and 26 (pax2a) but very weak
pax8 <- c("PAX8", "PAX8.1", "PAX8.2", "PAX8.3", "PAX8.4", "PAX8.5", "PAX8.6", "PAX8.7") #ns
bach <- c("BACH", "BACH.1") #ns
nfe2 <- c("NF2L1") #marker for 4 but high in everything

#TFs for spist gland
rfx <- c("RFX6", "RFX4") #marker for 11, 18 (RFX6) and 16, 20 (RFX4)

#TFs for calicoblast
dnajc21 <- c("DJC21") #this is spist zf-met.like DNAJC21; highest in 23
prdm9 <- c("PRDM6.7") #marker for 14 and 19, PRDM9 in description, kinda weak
myb <- c("evm.TU.Ap3.186") #myb-like, marker for 1, 2, 23; mostly 23
ncor <- c("NCOR1", "NCOR1.1") #markers for 2 but also NCOR1 in 23
ssrp1 <- c("SSRP1") #marker for 7
dmrt <- c("DMRTD", "DMTA2.3") #2.3 is marker for 3 and 4 but all weak
csd <- c("CSD") #ns
atf <- c("ZHANG", "ZHANG.1", "CREM") #ZHANGs markers for 2 but weak, crem ns
dbp <- c("DBP", "DBP.2", "HLF.2", "HLF.3", "TEF") #dbps markers for 5; hlf markers for 5 and 1; tef ns
atf2 <- c("ATF2") #ATF2 marker for 25 but weak
# forkhead (fox) tf marker for many cells including 23, 25, 27

###random genes of interest from spist paper
#cnidocyte (minicollagen but we don't have that)
tal2 <- c("TAL2.4") #tachylectins or galactosamine binding lectins, marker for 8, 16; mostly 8, a tiny bit 21

#immune
alps <- c("ALPS", "ALPS.1") #Anti-lipopolysaccharide factor, marker for 15, immune cell linked in spistillata; also 4
ikka <- c("IKKA") #marker for 7, 11, 18
ikbp1 <- c("IKBP1") #marker for 23
ilf2 <- c("ILF2") #marker for 1 and 23
hsp70 <- c("HSP71.1", "HSP71", "HSP7C.1") #marker for 15 (first), and 4 (second and third, also in all)
hsp90 <- c("H90A1", "HSP97") #first is marker for 1 and 4 but in all cells; second is marker for 15 and 1 also in a lot of cells
alphaA <- c("L2EFL.1") #alpha-crystallin A, marker for 11 and 12
alphaB <- c("CRYAB") #alpha-crystallin B, marker for 15
cathepsin <- c("CATL.1", "CATL.3", "CATB.1") #lots of cathepsin marker genes in 1, 4, and 15
bcl2 <- c( "BCL2") #bcl2 is marker for 15 and others; in lots of cells
#other random immune genes
nfkb1 <- c("NFKB1", "NFKB1.1") #ns
my881 <- c("MY88A") #ns
bcl3 <- c("BCL3", "BCL3.1") #ns
lbp <- c("LBP", "LBP.1", "LBP.2", "LBP.3") # lbp.3 and lbp.1 markers for 15
prosaposin <- c("SAP") # marker for 1, 4
endop <- c("ENDD1", "ENDD1.1", "evm.TU.Ap10.801") #endonuclease domain containing protein; idk if TF; last is marker for 15
cd36 <- c("CD36", "CD36.1", "CD36.2") #ns
sting <- c("STING", "STING.1", "STING.2", "STING.3", "STING.4", "STING.5") #ns
ifih1 <- c("IFIH1", "IFIH1.1", "IFIH1.2") #ns interferon-induced helicase c domain-containing protein
dock <- c("DOCK1", "DOCK3", "DOCK7", "DOCK9") #dedicator of cytokinesis; markers for 2
gbp6 <- c("GBP6.6", "GBP6", "GBP6.2") #marker for 15, guanylate binding protein 6; NICE (also markers for 4 and 11)
rab43 <- c("RAB43", "RAB43.1") #ns
bag <- c("BAG1", "BAG4", "BAG.2", "BAG.1") #BAG family molecular chaperone regulator 1 and 4 (ns) and IgA FC receptor; bag.1 and bag.2 are markers for 6
apoptosis <- c("BFAR", "BAX", "SIVA", "AR1", "CFLAR") #apoptosis regulators; markers for 2/4/13/17 (AR1) - mostly 4
casp3 <- c("CASP3.2") #caspase3 marker for 15
mvp <- c("MVP", "MVP.1", "MVP.2") #major vault protein; MVP is in a lot of things; MVP marker for 15 and high in others
ttc28 <- c("TTC28.15", "TTC28.32", "TTC28.26", "TTC28.240", "TTC28.118", "TTC28.152") #markers for many including 15; Tetratricopeptide repeat protein 28 

#gastro
tropomyosin <- c("TPM2", "TPM2.1", "TPM.1", "TPM.2", "TPM.6") #markers for many (1,3,4,5,6,9,10,11,12,14,15)
apoB <- c("APOB","APOB.1") #apolipoprotein markers for 1, 2, 4; also in 7
aplp <- c("APLP") #apolipophorin marker for 1, 2, 4; also in 7, 13, 17, 25 a bit
galk <- c("GALK1", "GALK2") #galactokinase; marker for 1
carb <- c("CAH2", "CAH2.2", "CAH2.5", "CAH2.6") #carbonic anhydrase; high exp in algal-hosting cells in spist; markers for 1, 3, 6 (many), 10, 14, 16, 24
npc2 <- c("NPC2", "NPC2.5", "NPC2.6", "NPC2.7") #markers for 1 and 15

#others to note some of which are markers: granulins (1, 2, 4), S2611 (transporter, 1, 2), lysosome associated things (LAMP1 in 1, 4, 15), GRP78 (glucose-reg, 1), PLCL.5 and .1 (1, 2, 4, 18), glutamic acid-rich proteins in 1, 3, 5, 7, 10, 14, 15; CMLO2.3 and NAT8L (acetyltransf., 1, 2, 4; 1, 4); acid phosphatase (1, 2, 4, 21)

#mitotic/germline
#look for cyclins and tubulins
sy65 <- c("SYT1", "SY65", "CO6A5.7", "ESYT3") #synaptotagmin, mitotic marker in spist; first is in 14; second in 11; third is in 14; fourth is in 3, 5, 10, 14, 19

#epidermis
pxdn <- c("PXDN.2", "PXDN.10", "PXDN.18", "PXDN.16") #first high in 3 and 10; second high in 10; third in 25; fourth in 3/10
moxd1 <- c("MOXD1.11", "MOXD1.1", "MOXD1") #DBH-like monooxygenase protein, markers for many; these high in 11; others in 1/2/3/5/6/9/10/13/14/19
gst1 <- c("GST1.2","MGST1") #GST1.2 in 1, 6, 11, 15; 1, 11
cadh <- c("FAT4.10", "FAT2.3", "PCDL") #FAT4s in 1, 2, 4, 24; fat2 in 13; pcdls in 5, 19, 27

#neuron
pkd1 <- c("PKD1.1", "PKD1.2", "PKD1.4") #markers for 13, 16, 19
dclk2 <- c("DCLK2.1", "DCLK2.2") #markers for 5, 11, 17, 19, 25 ; in many cells
cel3b <- c("CEL3B") #chymotrypsin-like elastase family member 3B, marker for 5 and 19
cabp5 <- c("CABP5") #calcium binding protein 5, calcium gated channel (neuron-y), marker for 17
trpa <- c("TRPA1.17") #marker for many, this one highest in 8, 16, 21
atoh8 <- c("ATOH8") #neuron marker in spist, ns
kcn <- c("KCNF1", "KCNAE") #potassium voltage gated channel thing markers for 5, 9, 13, 19; 12, 13, 17, 19 --> mostly 13
cbpd <- c("CBPA2.2", "CBPD.3", "CBPB1", "CBPA4") #carboxypeptidase, markers for 5; 18; ns; 12
lwa <- c("LWA", "LWA.1") #ns
chymo <- c("CEL3B", "CPP1.13", "CTRL.2", "CEL2A.4", "PLMN.2", "TMPS3", "CTRA.1") #all could be chymptrypsin homologs; in 5/19 (CEL3B); 20/21 (TMPS3)
cabo <- c("CABO.1") #squidulin (calmodulin homolog) marker for many including 5, 17, 19
calm <- c("CALM.13") #calmodulin, many calmodulin markers including for 5, 11, 19
cac1 <- c("CAC1A") #marker for many; voltage-dep calcium channel; 5, 12, 13, 17, 19, 20, 27
scn <- c("SCN2A") #marker for 5, 9, 13, 19, 27; sodium channel

#gland
olm2 <- c("OLM2A.6", "OLM2B") #olfactomedin, markers 18
muc5a <- c("MUC5A") #mucin 5AC marker for 8, 16, 21, 24
mlp <- c("MLP.12", "MLP") #mucin-like proteins, mlps in many, these mostly in 8/10/14/16/24
#some mlrp integumentary mucins in 8, 21, 22 and others
#lots of fbc fibrinogen in 18 and some in 22

#calicoblast
hemopexin <- c("EPRA1.2") #marker for 21 but still not great
myof <- c("MYOF","MYOF.5") #myoferlin/dysferlin markers many cells including 16, 21, 22
adam <- c("ADAM8.2") #metallopeptidase marker for 8 and 26, mostly 26
# a lot of Uncharacterized skeletal organic matrix protein in 18, some in 21
# hemicentin is in many cells and there are many orthologs
#sodium bicarbonate transporters in 6

VlnPlot(object =oculina_new, features = adam, pt.size= 0,combine = F,assay = "RNA") 
```

Predictions:
Gastro:
- 1 (erg, otx1, tropomyosin, apoB, aplp, galk, carb, npc2, granulins, S2611, LAMP1, GRP78, PLCL, glutamic acid-rich protein, CMLO2.3, NAT8L, acid phosphatase)
- 2 (deaf1, elf4, apoB, aplp, granulins, S2611, PLCL, CMLO2.3, acid phosphatase)
    - 3 (tropomyosin, carb, glutamic acid-rich protein)
- 4 (deaf1, erg, etv6, elf4, spdef, otx1, tropomyosin, apoB, aplp, granulins, LAMP1, PLCL, CMLO2.3, NAT8L, acid phosphatase)
    - 5 (tropomyosin, glutamic acid-rich protein)
    - 6 (tropomyosin, carb)
- 7 (apoB, aplp, glutamic acid-rich protein)
    - 9 (tropomyosin)
    - 10 (spdef, tropomyosin, carb, glutamic acid-rich protein)
    - 11 (tropomyosin)
    - 12 (tropomyosin)
    - 13 (elf4, spdef)
    - 14 (tropomyosin, carb, glutamic acid-rich protein)
    - 15 (tropomyosin, npc2, LAMP1, glutamic acid-rich protein)
    - 16 (carb)
    - 17 (elf4)
    - 18 (PLCL)
    - 21 (acid phosphatase)
    - 24 (carb)
    
Epidermis:
    - 1 (moxd1, gst1, FAT4)
    - 2 (moxd1, FAT4)
- 3 (pxdn, moxd1)
    - 4 (nfe2, FAT4)
    - 5 (moxd1, PCDL)
    - 6 (moxd1, gst1)
    - 9 (moxd1)
- 10 (pxdn, moxd1)
    - 11 (moxd1, gst1)
    - 13 (moxd1, FAT2)
- 14 (moxd1)
    - 15 (gst1)
    - 19 (moxd1, PCDL)
    - 24 (FAT4)
    - 25 (pax2, pxdn)
    - 26 (pax2)
    - 27 (PCDL)
    
Neuron:
    - 3 (pax7)
- 5 (dclk2, cel3b, kcn, cbpd, chymo, cabo, calm, cac1, scn)
    - 8 (trpa)
    - 9 (kcn, scn)
    - 10 (pax7)
    - 11 (dclk2, calm)
    - 12 (kcn, cbpd, cac1)
    - 13 (pou4, tbx20, pkd1, kcn, cac1, scn)
    - 14 (pax7)
    - 16 (pkd1, trpa)
- 17 (gata2, tbx20, dclk2, cabp5, kcn, cabo, cac1)
    - 18 (cbpd)
- 19 (pou4, pkd1, dclk2, cel3b, kcn, chymo, cabo, calm, cac1, scn)
    - 20 (tbx20, chymo, cac1)
    - 21 (trpa, chymo)
    - 25 (pou4, dclk2)
    - 26 (gata2)
    - 27 (pou4, cac1, scn)
    
Immune:
- 15 (irf1, irf2, atf3, alps, hsp70, hsp90, alphaB, cathepsin, bcl2, lbp, endop, gbp6, casp3, ttc28)

Gland:
    - 8 (muc5a, mlp, mlrp)
    - 10 (mlp)
    - 11 (rfx)
    - 14 (mlp)
    - 16 (rfx, muc5a, mlp)
    - 18 (rfx, olm2, fbc)
    - 20 (rfx)
    - 21 (muc5a, mlrp)
    - 22 (mlrp, fbc)
    - 24 (muc5a, mlp)
    
Calicoblast:
    - 1 (myb, dbp)
    - 2 (myb, ncor, atf)
    - 3 (dmrt)
    - 4 (dmrt)
    - 5 (dbp)
    - 6 (sodium bicarbonate transporter)
    - 7 (ssrp1)
    - 8 (adam)
    - 14 (prdm9)
    - 16 (myof)
    - 18 (usom)
    - 19 (prdm9)
    - 21 (hemopexin, myof, adam)
    - 22 (myof)
- 23 (DJC21, myb, ncor, forkhead)
    - 25 (atf2, forkhead)
    - 26 (adam)
    - 27 (forkhead)
    
Cnidocyte:
    - 8 (tal2)
    - 16 (tal2)
    
1 [Gastrodermis]
2 [Gastrodermis]
3 [Epidermis]
4 [Gastrodermis]
5 [Neuron]
6 [Epidermis] ??
7 [Gastrodermis]
8 [Cnidocyte]
9 [Epidermis]
10 [Epidermis]
11 [Gland] ?
12 [Gland] ?
13 [Neuron]
14 [Epidermis]
15 [Immune Cell]
16 [Cnidocyte]
17 [Neuron]
18 [Gland]
19 [Neuron]
20 [Gland]
21 [Gland]
22 [Neuron]
23 [Calicoblast]
24 [Gland]
25 [Neuron]
26 [Epidermis]
27 [Neuron]

```{r}
oc_markers_pre_new<- FindAllMarkers(oculina_new, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
nrow(oc_markers_pre_new)
#5766

oc_markers_pre_new %>% group_by(cluster) %>% top_n(20, avg_log2FC) -> top20_pre_new
top20_pre_new
nrow(top20_pre_new)
#540

color<-colorRampPalette(brewer.pal(n=9, name="Reds"),bias=0.5)(20)

DoHeatmap(oculina_new, features = top20_pre_new$gene, label=T) + theme(axis.text.y= element_text(size=0)) + scale_fill_gradientn(colors=color)
```

Rename cell clusters with new names
```{r}
oculina_named_new <- RenameIdents(oculina_new, "1" = "Gastrodermis 1")
oculina_named_new <- RenameIdents(oculina_named_new, "2" = "Gastrodermis 2")
oculina_named_new <- RenameIdents(oculina_named_new, "3" = "Epidermis 1")
oculina_named_new <- RenameIdents(oculina_named_new, "4" = "Gastrodermis 3")
oculina_named_new <- RenameIdents(oculina_named_new, "5" = "Neuron 1")
oculina_named_new <- RenameIdents(oculina_named_new, "6" = "Epidermis 2")
oculina_named_new <- RenameIdents(oculina_named_new, "7" = "Gastrodermis 4")
oculina_named_new <- RenameIdents(oculina_named_new, "8" = "Cnidocyte 1")
oculina_named_new <- RenameIdents(oculina_named_new, "9" = "Epidermis 3")
oculina_named_new <- RenameIdents(oculina_named_new, "10" = "Epidermis 4")
oculina_named_new <- RenameIdents(oculina_named_new, "11" = "Gland 1")
oculina_named_new <- RenameIdents(oculina_named_new, "12" = "Gland 2")
oculina_named_new <- RenameIdents(oculina_named_new, "13" = "Neuron 2")
oculina_named_new <- RenameIdents(oculina_named_new, "14" = "Epidermis 5")
oculina_named_new <- RenameIdents(oculina_named_new, "15" = "Immune Cell")
oculina_named_new <- RenameIdents(oculina_named_new, "16" = "Cnidocyte 2")
oculina_named_new <- RenameIdents(oculina_named_new, "17" = "Neuron 3")
oculina_named_new <- RenameIdents(oculina_named_new, "18" = "Gland 3")
oculina_named_new <- RenameIdents(oculina_named_new, "19" = "Neuron 4")
oculina_named_new <- RenameIdents(oculina_named_new, "20" = "Gland 4")
oculina_named_new <- RenameIdents(oculina_named_new, "21" = "Gland 5")
oculina_named_new <- RenameIdents(oculina_named_new, "22" = "Neuron 5")
oculina_named_new <- RenameIdents(oculina_named_new, "23" = "Calicoblast")
oculina_named_new <- RenameIdents(oculina_named_new, "24" = "Gland 6")
oculina_named_new <- RenameIdents(oculina_named_new, "25" = "Neuron 6")
oculina_named_new <- RenameIdents(oculina_named_new, "26" = "Epidermis 6")
oculina_named_new <- RenameIdents(oculina_named_new, "27" = "Neuron 7")

levels(oculina_named_new)
levels(oculina_named_new) <- c("Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gastrodermis 4", "Epidermis 1", "Epidermis 2", "Epidermis 3", "Epidermis 4", "Epidermis 5", "Epidermis 6", "Neuron 1", "Neuron 2", "Neuron 3", "Neuron 4", "Neuron 5", "Neuron 6", "Neuron 7", "Gland 1", "Gland 2", "Gland 3", "Gland 4", "Gland 5", "Gland 6", "Calicoblast", "Cnidocyte 1", "Cnidocyte 2", "Immune Cell")
levels(oculina_named_new)

DefaultAssay(oculina_named_new) <- "RNA"
saveRDS(oculina_named_new, paste0(proj_name, '_oculinanamed_NEW_091824.rds'))
oculina_named_new <- readRDS(file.choose())

install.packages("simplecolors")
library(simplecolors)

gastro_col <- colorRampPalette(c("#C7B8E6", "#330099"))    
gdc <- gastro_col(4) 
gdc4 <- c("#C7B8E6", "#957ACC", "#643DB2", "#330099")
epi_col <- colorRampPalette(c("#B6D8B6", "#006600"))
ec <- epi_col(6) 
ec6 <- c("#B6D8B6", "#91C191", "#6DAA6D", "#489348", "#247C24", "#006600")
neuro_col <- colorRampPalette(c("#B8DEE6", "#006699"))
nc <- neuro_col(7)
nc7 <- c("#B8DEE6", "#99CAD9", "#7AB6CC", "#5CA2BF", "#3D8EB2", "#1E7AA5", "#006699")
gland_col <- colorRampPalette(c("#E6B8BF","#9C1616"))
gc <- gland_col(6)
gc6 <- c("#E6B8BF", "#D7979D", "#C8777B", "#B95659", "#AA3637", "#9C1616")

cols <- c("#C7B8E6", "#957ACC", "#643DB2", "#330099", "#B6D8B6", "#91C191", "#6DAA6D", "#489348", "#247C24", "#006600", "#B8DEE6", "#99CAD9", "#7AB6CC", "#5CA2BF", "#3D8EB2", "#1E7AA5", "#006699", "#E6B8BF", "#D7979D", "#C8777B", "#B95659", "#AA3637", "#9C1616", "#996633", "#FF9900", "#FFCC66", "#333333")

plot1 <- DimPlot(oculina_named_new, reduction = "umap", label=F, label.size = 3, cols=cols, alpha=0.5) 
LabelClusters(plot1, id = "ident", size = 3, repel = T,  box.padding = 0.25, fontface="bold") + theme(legend.text=element_text(size=10))
#umap_oculina_091824 as landscape 8.5x6

plot2 <- DimPlot(oculina_named_new, reduction = "umap", label=F,  shuffle=TRUE, group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))
#umap_oculina_aposym_091824 as landscape 6.5x5.5


```

Reread about comparative analysis for calicoblast

Top 10 and top 20 genes per cluster
```{r}
#get the top10 from the re-leveled oculina_named seurat object
oc_markers_lvld_new<- FindAllMarkers(oculina_named_new, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
nrow(oc_markers_lvld_new)
#5766

oc_markers_lvld_new %>% group_by(cluster) %>% top_n(10, avg_log2FC) -> top10_lev_new
top10_lev_new
nrow(top10_lev_new)
#270
#gives the top 10 genes for each cluster; 27 clusters
saveRDS(top10_lev_new, paste0(proj_name, '_top10_releveled_named_091824.rds'))

#do top20
oc_markers_lvld_new %>% group_by(cluster) %>% top_n(20, avg_log2FC) -> top20_lev_new
top20_lev_new
nrow(top20_lev_new)
#540
saveRDS(top20_lev_new, paste0(proj_name, '_top20_releveled_named_091824.rds'))

DoHeatmap(oculina_named_new, features = top10_lev_new$gene,label=F, group.colors = cols) + theme(axis.text.y= element_text(size=0)) + scale_fill_gradientn(colors=color)
#saved as top10_releveled_named_091824 as landscape 9x6

DoHeatmap(oculina_named_new, features = top20_lev_new$gene,label=F, group.colors = cols) + theme(axis.text.y= element_text(size=0)) + scale_fill_gradientn(colors=color)
#saved as top20_releveled_named_091824 as landscape 9x6
```

###analyze specific cell clusters

Gastrodermis 1 #####COME BACK
```{r}
gderm1 <- subset(oculina_named_new, idents = "Gastrodermis 1") 

gderm1 <- FindVariableFeatures(gderm1)
gderm1 <- ScaleData(gderm1, features = features_to_scale)
gderm1 <- RunPCA(gderm1)
gderm1 <- FindNeighbors(gderm1, dims = 1:30, reduction = "pca")
gderm1 <- FindClusters(gderm1, resolution = 0.25)
#5 clusters
gderm1 <- RunUMAP(gderm1, reduction="pca", dims=1:30, return.model=TRUE)

#gderm1 <- RunUMAP(gderm1, dims = 1:30, reduction = "integrated.cca")

gderm1@meta.data$seurat_clusters <- as.numeric(as.character(gderm1@meta.data$seurat_clusters))+1
Idents(gderm1)<- "seurat_clusters"
gderm1$seurat_clusters
gderm1@active.ident <- factor(x = gderm1@active.ident, levels = 1:5)
gderm1@active.ident

levels(gderm1) <- c("1","2","3","4", "5")

plot5 <- DimPlot(gderm1, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99", "#3333FF", "#FF9900", "gray40"), alpha=0.5)
#5 clusters


plot6 <- DimPlot(gderm1, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot6b <- DimPlot(gderm1, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99", "#3333FF", "#FF9900", "gray50"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))
#NOW THINGS ARE IN THE RIGHT ORDER!!!!

#saved as umap_gderm1_plot5plot6_new 9x4
#umap_gderm1_plot6b_new 8x4 landscape
```

```{r}
gderm1_topfeat <- TopFeatures(object = gderm1[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gderm1[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gd1 <- FetchData(gderm1, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gd1 <- FetchData(gderm1, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gd1, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gd1, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as gderm1_pcplots_umap_091524

DimPlot(gderm1, reduction = "pca", label=F, order=c("sym", "apo"), group.by = "symstate", cols=alpha(c("gray40", "#CC6600"), 0.5))

stdev_gd1 <- Stdev(gderm1, reduction = "pca")

pca_gd1 <- pc_data_gd1
pca_gd1$sym_state <- gderm1@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gd1)) #<2e-16 *** 
summary(aov(PC_2 ~ sym_state, data = pca_gd1)) #0.00547 **
summary(aov(PC_3 ~ sym_state, data = pca_gd1)) #<2e-16 ***
summary(aov(PC_4 ~ sym_state, data = pca_gd1)) #2.09e-10 ***
summary(aov(PC_5 ~ sym_state, data = pca_gd1)) #<2e-16 ***                             
summary(aov(PC_6 ~ sym_state, data = pca_gd1)) #3.1e-06 ***
summary(aov(PC_7 ~ sym_state, data = pca_gd1)) #1.33e-12 ***
summary(aov(PC_8 ~ sym_state, data = pca_gd1)) #<2e-16 ***
summary(aov(PC_9 ~ sym_state, data = pca_gd1)) #0.53
summary(aov(PC_10 ~ sym_state, data = pca_gd1)) #0.0118 *

print(gderm1[["pca"]], dims = 1:5, nfeatures = 10)

DefaultAssay(gderm1) <- "RNA"
gderm1[["RNA"]]$counts<-as.matrix(gderm1[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gderm1_mark <- FindMarkers(gderm1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gderm1_mark)
#pretty sure in this case neg lfc means down in sym; pos lfc means up in sym
gderm1_mark <- rownames_to_column(gderm1_mark, var="gene_symbol")

#look at subcluster marker genes
gderm1_markers_subclust <- FindAllMarkers(gderm1, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gderm1_markers_subclust

#get apo vs sym differences for subcluster 1 of gderm1
gderm1_1 <- subset(gderm1, idents="1")
DefaultAssay(gderm1_1) <- "RNA"

gderm1_1_mark <- FindMarkers(gderm1_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm1_1_mark <- rownames_to_column(gderm1_1_mark, var="gene_symbol")
head(gderm1_1_mark)

#get apo vs sym differences for subcluster 2 of gderm1
gderm1_2 <- subset(gderm1, idents="2")
DefaultAssay(gderm1_2) <- "RNA"
gderm1_2_mark <- FindMarkers(gderm1_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm1_2_mark <- rownames_to_column(gderm1_2_mark, var="gene_symbol")
head(gderm1_2_mark)

#for 3
gderm1_3 <- subset(gderm1, idents="3")
DefaultAssay(gderm1_3) <- "RNA"
#can't do marker genes bc fewer than 3 cells in apo

# and for 4
gderm1_4 <- subset(gderm1, idents="4")
DefaultAssay(gderm1_4) <- "RNA"

gderm1_4_mark <- FindMarkers(gderm1_4, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm1_4_mark <- rownames_to_column(gderm1_4_mark, var="gene_symbol")
head(gderm1_4_mark)
```

Gastrodermis 2
```{r}
gderm2 <- subset(oculina_named_new, idents = "Gastrodermis 2") 

gderm2 <- FindVariableFeatures(gderm2)
gderm2 <- ScaleData(gderm2, features = features_to_scale)
gderm2 <- RunPCA(gderm2)
gderm2 <- FindNeighbors(gderm2, dims = 1:30, reduction = "pca")
gderm2 <- FindClusters(gderm2, resolution = 0.25)
#3 cluster
gderm2 <- RunUMAP(gderm2, reduction="pca", dims=1:30, return.model=TRUE)

#gderm2 <- RunUMAP(gderm2, dims = 1:30, reduction = "integrated.cca")

gderm2@meta.data$seurat_clusters <- as.numeric(as.character(gderm2@meta.data$seurat_clusters))+1
Idents(gderm2)<- "seurat_clusters"
gderm2$seurat_clusters
gderm2@active.ident <- factor(x = gderm2@active.ident, levels = 1:3)
gderm2@active.ident

levels(gderm2) <- c("1", "2", "3")

plot7 <- DimPlot(gderm2, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5)
#3 clusters


plot8 <- DimPlot(gderm2, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot8b <- DimPlot(gderm2, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99", "#3333FF", "#FF9900"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gderm2_plot7plot8_new 9x4
#umap_gderm2_plot8b_new 8x4 landscape

#do the intersection between gderm2 cells and algal cells then dimplot them
######## MAKE THE ALGAL CELLS ONLY THE ONES THAT EXPRESS SYMBIONT READS
gderm2_cell_ids <- rownames(gderm2@meta.data)
sym_gderm2_intersect <- intersect(gderm2_cell_ids, alga_cells)

symcell_gderm2_plot <- DimPlot(gderm2, reduction = "umap", cells.highlight = sym_gderm2_intersect, cols.highlight=c("black"), split.by = "symstate")
#saved as sym_cells_umap_091624 as 6x6
```

```{r}
gderm2_topfeat <- TopFeatures(object = gderm2[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gderm2[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gd2 <- FetchData(gderm2, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gd2 <- FetchData(gderm2, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gd2, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gd2, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as gderm2_pcplots_umap_091824

DimPlot(gderm2, reduction = "pca", label=F, order=c("sym", "apo"), group.by = "symstate", cols=alpha(c("gray40", "#CC6600"), 0.5))

stdev_gd2 <- Stdev(gderm2, reduction = "pca")

pca_gd2 <- pc_data_gd2
pca_gd2$sym_state <- gderm2@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gd2)) #2.32e-06 ***
summary(aov(PC_2 ~ sym_state, data = pca_gd2)) #<2e-16 ***
summary(aov(PC_3 ~ sym_state, data = pca_gd2)) #0.891
summary(aov(PC_4 ~ sym_state, data = pca_gd2)) #0.178
summary(aov(PC_5 ~ sym_state, data = pca_gd2)) #8.21e-07 ***                        
summary(aov(PC_6 ~ sym_state, data = pca_gd2)) #0.0166 *
summary(aov(PC_7 ~ sym_state, data = pca_gd2)) #0.0148 *
summary(aov(PC_8 ~ sym_state, data = pca_gd2)) #1.78e-09 ***
summary(aov(PC_9 ~ sym_state, data = pca_gd2)) #0.86
summary(aov(PC_10 ~ sym_state, data = pca_gd2)) #0.428

print(gderm2[["pca"]], dims = 1:5, nfeatures = 10)

DefaultAssay(gderm2) <- "RNA"
gderm2[["RNA"]]$counts<-as.matrix(gderm2[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gderm2_mark <- FindMarkers(gderm2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gderm2_mark)
#neg lfc means down in sym; pos lfc means up in sym


gderm2_mark <- rownames_to_column(gderm2_mark, var="gene_symbol")

#look at subcluster marker genes
gderm2_markers_subclust <- FindAllMarkers(gderm2, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gderm2_markers_subclust
gderm2_markers_subclust <- rownames_to_column(gderm2_markers_subclust, var="gene_symbol")

# also do some feature plots with the immunity and nutrition genes
catl.9_feat_plot_g2 <- FeaturePlot(gderm2, features=c("CATL.9"), split.by = "symstate", order = TRUE, reduction="umap",  min.cutoff = 2) + theme(legend.position = c(0.98,0.98))
catl.7_feat_plot_g2 <- FeaturePlot(gderm2, features=c("CATL.7"), split.by = "symstate", order = TRUE, reduction="umap",  min.cutoff = 2) + theme(legend.position = c(0.98,0.98))
catb.1_feat_plot_g2 <- FeaturePlot(gderm2, features=c("CATB.1"), split.by = "symstate", order = TRUE, reduction="umap",  min.cutoff = 2) + theme(legend.position = c(0.98,0.98))
h90_feat_plot_g2 <- FeaturePlot(gderm2, features=c("H90A1"), split.by = "symstate", order = TRUE, reduction="umap",  min.cutoff = 2) + theme(legend.position = c(0.98,0.98))

glsn_feat_plot_g2 <- FeaturePlot(gderm2, features=c("GLSN"), split.by = "symstate", order = TRUE, reduction="umap",  min.cutoff = 1) + theme(legend.position = c(0.98,0.98))
aplp_feat_plot_g2 <- FeaturePlot(gderm2, features=c("APLP"), split.by = "symstate", order = TRUE, reduction="umap",  min.cutoff = 3) + theme(legend.position = c(0.98,0.98))
apoB1_feat_plot_g2 <- FeaturePlot(gderm2, features=c("APOB.1"), split.by = "symstate", order = TRUE, reduction="umap",  min.cutoff = 3) + theme(legend.position = c(0.98,0.98))
dhe4_feat_plot_g2 <- FeaturePlot(gderm2, features=c("DHE4"), split.by = "symstate", order = TRUE, reduction="umap",  min.cutoff = 2) + theme(legend.position = c(0.98,0.98))




#get the top10 from the gderm2
gd2_markers_lvld_new<- FindAllMarkers(gderm2, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
nrow(gd2_markers_lvld_new)
#454

gd2_markers_lvld_new %>% group_by(cluster) %>% top_n(10, avg_log2FC) -> top10_gd2
top10_gd2
nrow(top10_gd2)
#34 bc cluster 4 only has 4

DoHeatmap(gderm2, features = top10_gd2$gene,label=F, group.colors = c("#CC6699", "#33CC99", "#3333FF", "#FF9900")) + theme(axis.text.y= element_text(size=0)) + scale_fill_gradientn(colors=color)
#cluster 4 has most similar expression to cluster 2, which are entirely ribosomal proteins (RLs), then the tbbs in cluster 4 are super specific to cluster 4

rl10_feat_plot_g2 <- FeaturePlot(gderm2, features=c("RL10"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) # some expression in algal-hosters

rl18a_feat_plot_g2 <- FeaturePlot(gderm2, features=c("RL18A"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) # some expression in algal-hosters

entp5_feat_plot_g2 <- FeaturePlot(gderm2, features=c("ENTP5"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) # ectonucleoside triphosphate diphosphohydrolase 5; some expression


#look at features that define gderm2 from all the other clusters and see what is highly expressed across all subclusters
nfat5_feat_plot_g2 <- FeaturePlot(gderm2, features=c("NFAT5.4"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) #cluster 4 excluded

evm1_feat_plot_g2 <- FeaturePlot(gderm2, features=c("evm.TU.Ap3.1250.evm.TU.Ap3.1253"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) #nope

tie2_feat_plot_g2 <- FeaturePlot(gderm2, features=c("TIE2"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98))

evm2_feat_plot_g2 <- FeaturePlot(gderm2, features=c("evm.TU.Ap3.1276.evm.TU.Ap3.1281"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) # naur

wdr41_feat_plot_g2 <- FeaturePlot(gderm2, features=c("WDR41"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) # naur

aplp_feat_plot_g2 <- FeaturePlot(gderm2, features=c("APLP"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) #nah

s6a13_feat_plot_g2 <- FeaturePlot(gderm2, features=c("S6A13"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) #nah

s6a13_feat_plot_g2 <- FeaturePlot(gderm2, features=c("S6A13"), split.by = "symstate", order = TRUE, reduction="umap") + theme(legend.position = c(0.98,0.98)) 

##there's gottta be a better way to do this
# let's take the list of markers for gderm2, then heatmap them within gderm2
markers_strong_2 <- markers_strong[717:1414, ]

DoHeatmap(gderm2, features = markers_strong_2$gene,label=T, group.colors = c("#CC6699", "#33CC99", "#3333FF", "#FF9900")) + theme(axis.text.y= element_text(size=0)) + scale_fill_gradientn(colors=color) # not really any clearer

gderm2_clustertree <- BuildClusterTree(object = gderm2)
PlotClusterTree(object = gderm2_clustertree)

### have to think of a technical justification for why I removed cells with high symbiont:host count ratios (or a way to better say that the symbiont reads were fucking with the data)

#at this stage, have a list of cells assigned as "algal-hosting" and that are in gderm2
#in gderm2, I'm going to "calculate the percentage of all the counts belonging to a subset [host genes] of the possible features for each cell"
gderm_host_percent <- PercentageFeatureSet(gderm2, col.name = "host_percent", features = rownames(counts_lim))
#okay this is going to have to be done earlier before I remove the symbiont reads

################# HAVEN'T DONE BELOW YET
#get apo vs sym differences for subcluster 1 of gderm1
gderm3_1 <- subset(gderm3, idents="1")
DefaultAssay(gderm3_1) <- "RNA"

gderm3_1_mark <- FindMarkers(gderm3_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm3_1_mark <- rownames_to_column(gderm3_1_mark, var="gene_symbol")
head(gderm3_1_mark)

#get apo vs sym differences for subcluster 2 of gderm1
gderm3_2 <- subset(gderm3, idents="2")
DefaultAssay(gderm3_2) <- "RNA"
#no apos

#for 3
gderm3_3 <- subset(gderm3, idents="3")
DefaultAssay(gderm3_3) <- "RNA"

gderm3_3_mark <- FindMarkers(gderm3_3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm3_3_mark <- rownames_to_column(gderm3_3_mark, var="gene_symbol")
head(gderm3_3_mark)

```


Gastrodermis 3
```{r}
gderm3 <- subset(oculina_named_h, idents = "Gastrodermis 3") 

gderm3 <- FindVariableFeatures(gderm3)
gderm3 <- ScaleData(gderm3, features = rownames(gderm3@assays$RNA@features))
gderm3 <- RunPCA(gderm3)
gderm3 <- FindNeighbors(gderm3, dims = 1:30, reduction = "pca")
gderm3 <- FindClusters(gderm3, resolution = 0.25)
#3 clusters
gderm3 <- RunUMAP(gderm3, reduction="pca", dims=1:30, return.model=TRUE)

#gderm3 <- RunUMAP(gderm3, dims = 1:30, reduction = "integrated.cca")

gderm3@meta.data$seurat_clusters <- as.numeric(as.character(gderm3@meta.data$seurat_clusters))+1
Idents(gderm3)<- "seurat_clusters"
gderm3$seurat_clusters
gderm3@active.ident <- factor(x = gderm3@active.ident, levels = 1:3)
gderm3@active.ident

levels(gderm3) <- c("1", "2", "3")

plot9 <- DimPlot(gderm3, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5)
#3 cluster


plot10 <- DimPlot(gderm3, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot10b <- DimPlot(gderm3, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gderm3_plot9plot10_new 9x4
#umap_gderm3_plot10b_new 8x4 landscape

###### Cluster 2 is entirely sym
```

```{r}
gderm3_topfeat <- TopFeatures(object = gderm3[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gderm3[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gd3 <- FetchData(gderm3, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gd3 <- FetchData(gderm3, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gd3, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gd3, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as gderm3_pcplots_umap_091624

DimPlot(gderm3, reduction = "pca", label=F, order=c("sym", "apo"), group.by = "symstate", cols=alpha(c("gray40", "#CC6600"), 0.5))

stdev_gd3 <- Stdev(gderm3, reduction = "pca")

pca_gd3 <- pc_data_gd3
pca_gd3$sym_state <- gderm3@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gd3)) #1.35e-14 ***
summary(aov(PC_2 ~ sym_state, data = pca_gd3)) #6.3e-11 ***
summary(aov(PC_3 ~ sym_state, data = pca_gd3)) #<2e-16 ***
summary(aov(PC_4 ~ sym_state, data = pca_gd3)) #0.345
summary(aov(PC_5 ~ sym_state, data = pca_gd3)) #8.66e-12 ***                           
summary(aov(PC_6 ~ sym_state, data = pca_gd3)) #0.141
summary(aov(PC_7 ~ sym_state, data = pca_gd3)) #4.97e-13 ***
summary(aov(PC_8 ~ sym_state, data = pca_gd3)) #0.329
summary(aov(PC_9 ~ sym_state, data = pca_gd3)) #0.019 *
summary(aov(PC_10 ~ sym_state, data = pca_gd3)) #0.94

print(gderm3[["pca"]], dims = 1:5, nfeatures = 10)

DefaultAssay(gderm3) <- "RNA"
gderm3[["RNA"]]$counts<-as.matrix(gderm3[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gderm3_mark <- FindMarkers(gderm3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gderm3_mark)
#pretty sure in this case neg lfc means down in sym; pos lfc means up in sym
gderm3_mark <- rownames_to_column(gderm3_mark, var="gene_symbol")

#look at subcluster marker genes
gderm3_markers_subclust <- FindAllMarkers(gderm3, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gderm3_markers_subclust

#get apo vs sym differences for subcluster 1 of gderm1
gderm3_1 <- subset(gderm3, idents="1")
DefaultAssay(gderm3_1) <- "RNA"

gderm3_1_mark <- FindMarkers(gderm3_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm3_1_mark <- rownames_to_column(gderm3_1_mark, var="gene_symbol")
head(gderm3_1_mark)

#get apo vs sym differences for subcluster 2 of gderm1
gderm3_2 <- subset(gderm3, idents="2")
DefaultAssay(gderm3_2) <- "RNA"
#no apos

#for 3
gderm3_3 <- subset(gderm3, idents="3")
DefaultAssay(gderm3_3) <- "RNA"

gderm3_3_mark <- FindMarkers(gderm3_3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm3_3_mark <- rownames_to_column(gderm3_3_mark, var="gene_symbol")
head(gderm3_3_mark)

```



Gastrodermis 4
```{r}
gderm4 <- subset(oculina_named_h, idents = "Gastrodermis 4") 

gderm4 <- FindVariableFeatures(gderm4)
gderm4 <- ScaleData(gderm4, features = rownames(gderm4@assays$RNA@features))
gderm4 <- RunPCA(gderm4)
gderm4 <- FindNeighbors(gderm4, dims = 1:30, reduction = "pca")
gderm4 <- FindClusters(gderm4, resolution = 0.25)
#2 clusters
gderm4 <- RunUMAP(gderm4, reduction="pca", dims=1:30, return.model=TRUE)

#gderm4 <- RunUMAP(gderm4, dims = 1:30, reduction = "integrated.cca")

gderm4@meta.data$seurat_clusters <- as.numeric(as.character(gderm4@meta.data$seurat_clusters))+1
Idents(gderm4)<- "seurat_clusters"
gderm4$seurat_clusters
gderm4@active.ident <- factor(x = gderm4@active.ident, levels = 1:2)
gderm4@active.ident

levels(gderm4) <- c("1", "2")

plot11 <- DimPlot(gderm4, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 cluster


plot12 <- DimPlot(gderm4, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot12b <- DimPlot(gderm4, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gderm4_plot11plot12_new 9x4
#umap_gderm4_plot12b_new 8x4 landscape
```

Algal-Hosting
```{r}
alga <- subset(oculina_named_h, idents = "Algal-Hosting") 

alga <- FindVariableFeatures(alga)
alga <- ScaleData(alga, features = rownames(alga@assays$RNA@features))
alga <- RunPCA(alga)
alga <- FindNeighbors(alga, dims = 1:30, reduction = "pca")
alga <- FindClusters(alga, resolution = 0.25)
#1 cluster
alga <- RunUMAP(alga, reduction="pca", dims=1:30, return.model=TRUE)

#gderm1 <- RunUMAP(gderm1, dims = 1:30, reduction = "integrated.cca")

alga@meta.data$seurat_clusters <- as.numeric(as.character(alga@meta.data$seurat_clusters))+1
Idents(alga)<- "seurat_clusters"
alga$seurat_clusters
alga@active.ident <- factor(x = alga@active.ident, levels = 1)
alga@active.ident

#levels(gderm1) <- c("1","2","3","4")

plot13 <- DimPlot(alga, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot14 <- DimPlot(alga, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot14b <- DimPlot(alga, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_algalhost_plot13plot14_new 9x4
#umap_algalhost_plot14b_new 8x4 landscape
```

```{r}
ah_topfeat <- TopFeatures(object = alga[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
alga[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_ah <- FetchData(alga, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_ah <- FetchData(alga, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_ah, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_ah, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as algalhost_pcplots_umap_091524

DimPlot(alga, reduction = "pca", label=F, order=c("sym", "apo"), group.by = "symstate", cols=alpha(c("gray40", "#CC6600"), 0.5))

stdev_ah <- Stdev(alga, reduction = "pca")

pca_ah <- pc_data_ah
pca_ah$sym_state <- alga@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_ah)) #8.19e-09 ***
summary(aov(PC_2 ~ sym_state, data = pca_ah)) #0.191
summary(aov(PC_3 ~ sym_state, data = pca_ah)) #0.142
summary(aov(PC_4 ~ sym_state, data = pca_ah)) #0.891
summary(aov(PC_5 ~ sym_state, data = pca_ah)) #0.382

summary(aov(PC_6 ~ sym_state, data = pca_ah)) #0.0578 .
summary(aov(PC_7 ~ sym_state, data = pca_ah)) #0.761
summary(aov(PC_8 ~ sym_state, data = pca_ah)) #0.0969 .
summary(aov(PC_9 ~ sym_state, data = pca_ah)) #0.399
summary(aov(PC_10 ~ sym_state, data = pca_ah)) #0.926

print(alga[["pca"]], dims = 1:5, nfeatures = 10)

DefaultAssay(alga) <- "RNA"
alga[["RNA"]]$counts<-as.matrix(alga[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
alga_mark <- FindMarkers(alga, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(alga_mark)
#pretty sure in this case neg lfc means down in sym; pos lfc means up in sym
alga_mark <- rownames_to_column(alga_mark, var="gene_symbol")
```

What defines algal-hosting from the gastroderm1_subclust 3 (which is almost entirely sym cells)? and gastroderm3_subclust 2

What differentiates gderm1_3 from the rest of the gderm1 subclusters? What differentiates gderm3_2 from the rest of the gderm3 subclusters? What differentiates gderm1_3 from gderm3_2

And, if gderm1_3 and gderm3_2 aren't algal-hosting, what makes the sym vs apo separation so strong in the gastrodermal cells?

To parse out these questions, the first thing I'm going to do is compare all of the algal-hosting cluster to all of gderm1 then to all of gderm3
```{r}
mark_g1_ah <- FindMarkers(oculina_named_h, ident.1 = "Gastrodermis 1", ident.2 = "Algal-Hosting") #one caveat to this code is that i didn't down sample so there are many more cells in gderm1 than in ah
mark_g1_ah <- rownames_to_column(mark_g1_ah, var="gene_symbol")
#neg is down in gderm1
#higher in algal-hosting other than sym reads: tubulin beta chains, s-antigen protein, Transmembrane protease serine 11D, Plasma kallikrein; trypsin; Major facilitator superfamily domain-containing protein, some others
#no clear patterns in the genes that are upregulated in gderm1 compared to algal-hosting
#I think what I have to do is match the genes to the gastro vs algal in the spist paper
#and def compare the gderm1_3 and gderm3_2

# get list of cell identifiers for gderm1_3
gderm1_3_cells <- rownames(gderm1_3@meta.data)
# get list of cell identifiers for alga
alga_cells <- rownames(alga@meta.data)
mark_g1_3_ah <- FindMarkers(oculina_named_h, ident.1 = gderm1_3_cells, ident.2 = alga_cells) 
mark_g1_3_ah <- rownames_to_column(mark_g1_3_ah, var="gene_symbol")
#cathepsins totally higher in gderm1_3_cells compared to alga_cells

# get list of cell identifiers for gderm3_2
gderm3_2_cells <- rownames(gderm3_2@meta.data)
mark_g3_2_ah <- FindMarkers(oculina_named_h, ident.1 = gderm3_2_cells, ident.2 = alga_cells) 
mark_g3_2_ah <- rownames_to_column(mark_g3_2_ah, var="gene_symbol")
#aplp higher in gderm but cathepsins not higher

# compare gderm1_3_cells and gderm3_2_cells
mark_g1_3_g3_2 <- FindMarkers(oculina_named_h, ident.1 = gderm1_3_cells, ident.2 = gderm3_2_cells) 
mark_g1_3_g3_2 <- rownames_to_column(mark_g1_3_g3_2, var="gene_symbol")
```

#########################

####
Take your list of algal-hosting cells from the sym, because the apo ones don't even express the symbiont gene markers

Read in your cat_host objects (where refgenome was the concatenated genomes but only the host genes)
```{r}
F8.host.data.only <- Read10X(data.dir <- "/projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/A3_host_cat_new/outs/filtered_feature_bc_matrix/")
F1.host.data.only <- Read10X(data.dir <- "/projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/S3_host_cat_new/outs/filtered_feature_bc_matrix/")

# check for duplicated barcodes(column names) within a dataset
any(duplicated(x = colnames(x = F8.host.data.only)))
#FALSE
any(duplicated(x = colnames(x = F1.host.data.only)))
#FALSE

# Rename columns to avoid trouble when merging data sets
colnames(x = F8.host.data.only) <- paste('F8', colnames(x = F8.host.data.only), sep = '_')
colnames(x = F1.host.data.only) <- paste('F1', colnames(x = F1.host.data.only), sep = '_')
rownames(x = F8.host.data.only) <- gsub("_", "-", rownames(F8.host.data.only))
head(rownames(F8.host.data.only))
rownames(x = F1.host.data.only) <- gsub("_", "-", rownames(F1.host.data.only))
head(rownames(F1.host.data.only))

#rename rownames
length(gene2gene_symbol$gene_symbol)
#45867
length(rownames(F8.host.data.only))
#43610
length(rownames(F1.host.data.only))
#43610

#replace the "merged" from F8_rownames with "" 
#------->>>>>>> THIS MIGHT CHANGE
####---->>>> no longer doing this, now instead of the "merged" the isoforms are in the format: "evm.TU.Ap11.1003-evm.TU.Ap11.1005"
F8.host.data_names.only <- F8.host.data.only
F8_host_rownames.only <- as.data.frame(rownames(F8.host.data_names.only))
colnames(F8_host_rownames.only) <-paste("X")
#F8_host_rownames$X <- gsub("MERGED%3A%20", "", F8_host_rownames$X)
F8_host_gene2gene_symbol.only <- F8_host_rownames.only %>%
  left_join(gene2gene_symbol, by=c("X"="gene"))
F8_host_gene2gene_symbol.only <- 
    transform(
        F8_host_gene2gene_symbol.only,
        gene_symbol =
            ifelse( is.na(gene_symbol), X, gene_symbol)) %>%
  mutate(gene_symbol = make.names(gene_symbol, unique = TRUE))
rownames(x=F8.host.data_names.only) <- F8_host_gene2gene_symbol.only$gene_symbol
head(rownames(F8.host.data_names.only))
head(rownames(F8.host.data.only))

#do the same for F1
F1.host.data_names.only <- F1.host.data.only
F1_host_rownames.only <- as.data.frame(rownames(F1.host.data_names.only))
colnames(F1_host_rownames.only) <-paste("X")
#F1_host_rownames$X <- gsub("MERGED%3A%20", "", F1_host_rownames$X)
F1_host_gene2gene_symbol.only <- F1_host_rownames.only %>%
  left_join(gene2gene_symbol, by=c("X"="gene"))
F1_host_gene2gene_symbol.only <- 
    transform(
        F1_host_gene2gene_symbol.only,
        gene_symbol =
            ifelse( is.na(gene_symbol), X, gene_symbol)) %>%
  mutate(gene_symbol = make.names(gene_symbol, unique = TRUE))
rownames(x=F1.host.data_names.only) <- F1_host_gene2gene_symbol.only$gene_symbol
head(rownames(F1.host.data_names.only))
head(rownames(F1.host.data.only))

### Initialize the Seurat object with the raw, non-normalized data

F8.only <- CreateSeuratObject(counts = F8.host.data_names.only, 
                             min.cells = 3, 
                             min.features = 200, 
                             project = "F8_host")
F1.only <- CreateSeuratObject(counts = F1.host.data_names.only, 
                             min.cells = 3, 
                             min.features = 200, 
                             project = "F1_host")

### add metadata to objects

# add individual
F8.only[["individual"]] <- "A_2F8"
F8.only@meta.data$individual <- as.factor(F8.only@meta.data$individual)
F1.only[["individual"]] <- "S_2F1"
F1.only@meta.data$individual <- as.factor(F1.only@meta.data$individual)

# add symstate
F8.only[["symstate"]] <- "apo"
F8.only@meta.data$symstate <- as.factor(F8.only@meta.data$symstate)
F1.only[["symstate"]] <- "sym"
F1.only@meta.data$symstate <- as.factor(F1.only@meta.data$symstate)

# add genet
F8.only[["genet"]] <- "F"
F8.only@meta.data$genet <- as.factor(F8.only@meta.data$genet)
F1.only[["genet"]] <- "F"
F1.only@meta.data$genet <- as.factor(F1.only@meta.data$genet)

str(F8.only@meta.data)
str(F1.only@meta.data)

F8.only
#24706 features across 3813 samples within 1 assay 
#Active assay: RNA (24706 features, 0 variable features)

F1.only
#24771 features across 3129 samples within 1 assay 
#Active assay: RNA (24771 features, 0 variable features)

F1_host_cell_ids <- rownames(F1.only@meta.data)
sym_host_intersect <- intersect(F1_host_cell_ids, alga_cells)
#9 of them, so there are 45 that get lost

F8_host_cell_ids <- rownames(F8.only@meta.data)
sym_host_intersect_apo <- intersect(F8_host_cell_ids, alga_cells)
#14 of them (which is all of them)
```

```{r}
obj.list.only <- list(F8.only, F1.only)

rownames(obj.list.only[[1]]) %>% as.data.frame() %>% View()

for (i in 1:length(obj.list.only)) {
  obj.list.only[[i]] <- PercentageFeatureSet(obj.list.only[[i]], pattern = "^MT-", col.name = 'percent.mt')
  obj.list.only[[i]][["per.mito.bin"]] <- ifelse(obj.list.only[[i]][["percent.mt"]] > 10, "> 10%", ifelse(obj.list.only[[i]][["percent.mt"]] > 5, "5%-10%", ifelse(obj.list.only[[i]][["percent.mt"]] > 2.5, "2.5%-5%", "< 2.5%")))
}

plts_qc.only <- lapply(obj.list.only, function(x) VlnPlot(x, 
                                             features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), 
                                             group.by = "individual",
                                             pt.size = .25,
                                             ncol = 3))

patchwork::wrap_plots(plts_qc.only, ncol = 1) 
```

```{r}
F8_new.only <- obj.list.only[[1]]
F1_new.only <- obj.list.only[[2]]
obj.list_merge.only <- merge(F8_new.only, F1_new.only)
obj.list_merge.only

#following the "Perform analysis without integration" first:
# run standard anlaysis workflow
merge.only <- NormalizeData(obj.list_merge.only)
merge.only <- FindVariableFeatures(merge.only)
merge.only <- ScaleData(merge.only, features = rownames(obj.list_merge.only@assays$RNA@features))
merge.only <- RunPCA(merge.only)
merge.only <- FindNeighbors(merge.only, dims = 1:30, reduction = "pca")
merge.only <- FindClusters(merge.only, resolution = 2, cluster.name = "unintegrated_clusters")
#42 is way too many clusters but that is unintegrated and with a high res

merge.only <- RunUMAP(merge.only, dims = 1:30, reduction = "pca", reduction.name = "umap.unintegrated")
DimPlot(merge.only, reduction = "umap.unintegrated", group.by = c("individual", "seurat_clusters"))
#the individuals do overlap really well

#"Perform integration"
merge.only <- IntegrateLayers(object = merge.only, method = CCAIntegration, orig.reduction = "pca", new.reduction = "integrated.cca",
    verbose = TRUE)
# re-join layers after integration
merge.only[["RNA"]] <- JoinLayers(merge.only[["RNA"]])

merge.only <- FindNeighbors(merge.only, reduction = "integrated.cca", dims = 1:30)
merge.only <- FindClusters(merge.only, resolution = 0.5)
#28 clusters at res of 0.5

merge.only <- RunUMAP(merge.only, dims = 1:30, reduction = "integrated.cca")

# Visualization
plt1merge.only <- DimPlot(merge.only, reduction = "umap", label = T, shuffle=TRUE)
plt2merge.only <- DimPlot(merge.only, reduction = "umap", group.by = "individual", shuffle=TRUE)
plt3merge.only <- DimPlot(merge.only, reduction = "umap", group.by = "symstate", shuffle=TRUE)

sym_cell_feat_plot <- DimPlot(merge.only, reduction = "umap", cells.highlight = sym_host_intersect, cols.highlight=c("black"))

sym_cell_feat_plot_apo <- DimPlot(merge.only, reduction = "umap", cells.highlight = sym_host_intersect_apo, cols.highlight=c("black"))
```

Okay that didn't work as well as I'd hoped, I think the cells with high amounts of symbiont reads are getting tossed, so I'm going to try to identify these cells in just a host genome, host gene background


```{r}
F8.host.host.data <- Read10X(data.dir <- "/projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/A3_host_only_new/outs/filtered_feature_bc_matrix/")
F1.host.host.data <- Read10X(data.dir <- "/projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/S3_host_only_new/outs/filtered_feature_bc_matrix/")

# check for duplicated barcodes(column names) within a dataset
any(duplicated(x = colnames(x = F8.host.host.data)))
#FALSE
any(duplicated(x = colnames(x = F1.host.host.data)))
#FALSE

# Rename columns to avoid trouble when merging data sets
colnames(x = F8.host.host.data) <- paste('F8', colnames(x = F8.host.host.data), sep = '_')
colnames(x = F1.host.host.data) <- paste('F1', colnames(x = F1.host.host.data), sep = '_')
rownames(x = F8.host.host.data) <- gsub("_", "-", rownames(F8.host.host.data))
head(rownames(F8.host.host.data))
rownames(x = F1.host.host.data) <- gsub("_", "-", rownames(F1.host.host.data))
head(rownames(F1.host.data.data))

#rename rownames
length(gene2gene_symbol$gene_symbol)
#45867
length(rownames(F8.host.host.data))
#43610
length(rownames(F1.host.host.data))
#43610

#replace the "merged" from F8_rownames with "" 
#------->>>>>>> THIS MIGHT CHANGE
####---->>>> no longer doing this, now instead of the "merged" the isoforms are in the format: "evm.TU.Ap11.1003-evm.TU.Ap11.1005"
F8.host.host.data_names <- F8.host.host.data
F8_host.host_rownames <- as.data.frame(rownames(F8.host.host.data_names))
colnames(F8_host.host_rownames) <-paste("X")
#F8_host_rownames$X <- gsub("MERGED%3A%20", "", F8_host_rownames$X)
F8_host.host_gene2gene_symbol <- F8_host.host_rownames %>%
  left_join(gene2gene_symbol, by=c("X"="gene"))
F8_host.host_gene2gene_symbol <- 
    transform(
        F8_host.host_gene2gene_symbol,
        gene_symbol =
            ifelse( is.na(gene_symbol), X, gene_symbol)) %>%
  mutate(gene_symbol = make.names(gene_symbol, unique = TRUE))
rownames(x=F8.host.host.data_names) <- F8_host.host_gene2gene_symbol$gene_symbol
head(rownames(F8.host.host.data_names))
head(rownames(F8.host.host.data))

#do the same for F1
F1.host.host.data_names <- F1.host.host.data
F1_host.host_rownames <- as.data.frame(rownames(F1.host.host.data_names))
colnames(F1_host.host_rownames) <-paste("X")
#F1_host_rownames$X <- gsub("MERGED%3A%20", "", F1_host_rownames$X)
F1_host.host_gene2gene_symbol <- F1_host.host_rownames %>%
  left_join(gene2gene_symbol, by=c("X"="gene"))
F1_host.host_gene2gene_symbol <- 
    transform(
        F1_host.host_gene2gene_symbol,
        gene_symbol =
            ifelse( is.na(gene_symbol), X, gene_symbol)) %>%
  mutate(gene_symbol = make.names(gene_symbol, unique = TRUE))
rownames(x=F1.host.host.data_names) <- F1_host.host_gene2gene_symbol$gene_symbol
head(rownames(F1.host.host.data_names))
head(rownames(F1.host.host.data))

### Initialize the Seurat object with the raw, non-normalized data

F8.final <- CreateSeuratObject(counts = F8.host.host.data_names, 
                             min.cells = 3, 
                             min.features = 200, 
                             project = "F8_host")
F1.final <- CreateSeuratObject(counts = F1.host.host.data_names, 
                             min.cells = 3, 
                             min.features = 200, 
                             project = "F1_host")

### add metadata to objects

# add individual
F8.final[["individual"]] <- "A_2F8"
F8.final@meta.data$individual <- as.factor(F8.final@meta.data$individual)
F1.final[["individual"]] <- "S_2F1"
F1.final@meta.data$individual <- as.factor(F1.final@meta.data$individual)

# add symstate
F8.final[["symstate"]] <- "apo"
F8.final@meta.data$symstate <- as.factor(F8.final@meta.data$symstate)
F1.final[["symstate"]] <- "sym"
F1.final@meta.data$symstate <- as.factor(F1.final@meta.data$symstate)

# add genet
F8.final[["genet"]] <- "F"
F8.final@meta.data$genet <- as.factor(F8.final@meta.data$genet)
F1.final[["genet"]] <- "F"
F1.final@meta.data$genet <- as.factor(F1.final@meta.data$genet)

str(F8.final@meta.data)
str(F1.final@meta.data)

F8.final
#24702 features across 3811 samples within 1 assay
#Active assay: RNA (24702 features, 0 variable features)

F1.final
#24767 features across 3135 samples within 1 assay 
#Active assay: RNA (24767 features, 0 variable features)

F1_final_cell_ids <- rownames(F1.final@meta.data)
sym_host_intersect_final <- intersect(F1_final_cell_ids, alga_cells)
#9 of them, so there are 45 that get lost

F8_final_cell_ids <- rownames(F8.final@meta.data)
sym_host_intersect_apo_final <- intersect(F8_final_cell_ids, alga_cells)
#14 of them (which is all of them)
```

Okay well darn that didn't work either

I think I have to keep the cat-cat situation to identify those algal cells, but then do the umap stuff and variable gene stuff just with the Astrangia host genes and no sym genes

Define features to scale for the real dealio
```{r}
features_to_scale = rownames(obj.list_merge.only@assays$RNA@features)
```


