---
title: "2025_Oculina_scRNAseq"
author: "Maria Valadez Ingersoll"
date: "2025-03-28"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Set working directory
```{r}
setwd("/projectnb/coral/MVI_Oculina/Oculina_sc_manuscript/SingleCell")
```

load packages
```{r}
library(stringr)
library(deldir)
library(Seurat)
library(tidyverse)
library(openxlsx)
library(MAST)
library(dplyr)
#install.packages("microseq")
library(microseq)
library(sctransform)
library(RColorBrewer)
library(ggplot2)
#BiocManager::install("glmGamPoi")
library(glmGamPoi)
library(gplots)
library(ggplot2)
library(pheatmap)
#install.packages("superheat")
library(superheat)
library(purrr)
library(tibble)
library(DESeq2)
library(ggrepel)
library(cowplot)
library(devtools)
#install.packages("simplecolors")
library(simplecolors)
#install_github('satijalab/seurat-wrappers')
library("SeuratWrappers")
#devtools::install_github('cole-trapnell-lab/monocle3')
library("monocle3")
#BiocManager::install("scDblFinder")
library("scDblFinder")
#BiocManager::install("slingshot")
library(slingshot)
```

```{r}
# load gene annotations
annotations <- read.delim("/projectnb/coral/Oculina_Genome_2025/Oculina_arbuscula/jaOcuArbu1.emapper.annotations_edit", sep="\t", header=TRUE)
head(annotations)
#filter annotations to only contain one entry for each gene (the original anntoation file contains all the gene versions)
annotations_filt <- filter(annotations, grepl(".t1", query)) %>%
  mutate(query = gsub(".t1", "", query))
write.table(annotations_filt, file="./files/jaOcuArbu1.emapper.annotations_edit_filt.txt", quote=F, sep="\t", row.names=FALSE)

gene = annotations_filt %>%
  select(query, Description, Preferred_name)
head(gene)
gene <-
    transform(gene ,
        Preferred_name =
            ifelse(Preferred_name=="-", query , Preferred_name)) %>%
  mutate(Preferred_name = make.names(Preferred_name, unique = TRUE))

annotations_filt_go <- annotations_filt[,c(1,10)]
annotations_filt_cog <- annotations_filt[,c(1,7)]
annotations_filt_pfam <- annotations_filt[,c(1,21)]
gene2go <- gene %>%
  left_join(annotations_filt_go) %>%
  select(Preferred_name, GOs)
write.table(gene2go, file="./files/jaOcuArbu1.gene2go.txt", quote=F, sep="\t", row.names=FALSE)  
gene2cog <- gene %>%
  left_join(annotations_filt_cog)%>%
  select(Preferred_name, COG_category)
gene2pfam <- gene %>%
  left_join(annotations_filt_pfam)%>%
  select(Preferred_name, PFAMs)

gene2gene_symbol <- gene[,c(1,3)]
gene2gene_symbol <- 
    transform(
        gene2gene_symbol ,
        Preferred_name =
            ifelse(Preferred_name=="-", query , Preferred_name)) %>%
  mutate(Preferred_name = make.names(Preferred_name, unique = TRUE))
write.table(gene2gene_symbol, file="./files/oculina2025_gene2genesymbol.txt", quote=F, sep="\t", row.names=FALSE)
```

# Concatonated analysis with symbiont and host reads to identify cells with symbionts
1. Read in your concatonated results data from cellranger
```{r}
#Set up your experiment
results_dir <- "catcat_analysis2025"
proj_name_lead <- "Oculina_catcat2025"

dir.create(results_dir, showWarnings = F)
proj_name <- paste0(results_dir, "/", proj_name_lead)

#these are not on the github or in the Rproject
F8.host.data <- Read10X(data.dir <- "/projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/A3_catcat_Oa/outs/filtered_feature_bc_matrix/")
F1.host.data <- Read10X(data.dir <- "/projectnb/coral/MVI_Oculina/10X_22_ApV2/outs/fastq_path/S3_catcat_Oa/outs/filtered_feature_bc_matrix/")

# check for duplicated barcodes(column names) within a dataset
any(duplicated(x = colnames(x = F8.host.data)))
#FALSE
any(duplicated(x = colnames(x = F1.host.data)))
#FALSE

# Rename columns to avoid trouble when merging data sets
colnames(x = F8.host.data) <- paste('F8', colnames(x = F8.host.data), sep = '_')
colnames(x = F1.host.data) <- paste('F1', colnames(x = F1.host.data), sep = '_')
rownames(x = F8.host.data) <- gsub("_", "-", rownames(F8.host.data))
head(rownames(F8.host.data))
rownames(x = F1.host.data) <- gsub("_", "-", rownames(F1.host.data))
head(rownames(F1.host.data))


#rename rownames
length(gene2gene_symbol$query)
#24130
length(rownames(F8.host.data))
#62323 bc of symbiont gtf input
length(rownames(F1.host.data))
#62323 bc of symbiont gtf input

F8.host.data_names <- F8.host.data
F8_host_rownames <- as.data.frame(rownames(F8.host.data_names))
colnames(F8_host_rownames) <-paste("X")
#F8_host_rownames$X <- gsub("MERGED%3A%20", "", F8_host_rownames$X)
F8_host_gene2gene_symbol <- F8_host_rownames %>%
  left_join(gene2gene_symbol, by=c("X"="query"))
F8_host_gene2gene_symbol <- 
    transform(
        F8_host_gene2gene_symbol ,
        Preferred_name =
            ifelse( is.na(Preferred_name), X, Preferred_name)) %>%
  mutate(Preferred_name = make.names(Preferred_name, unique = TRUE))
rownames(x=F8.host.data_names) <- F8_host_gene2gene_symbol$Preferred_name
head(rownames(F8.host.data_names))
head(rownames(F8.host.data))

#do the same for F1
F1.host.data_names <- F1.host.data
F1_host_rownames <- as.data.frame(rownames(F1.host.data_names))
colnames(F1_host_rownames) <-paste("X")
#F1_host_rownames$X <- gsub("MERGED%3A%20", "", F1_host_rownames$X)
F1_host_gene2gene_symbol <- F1_host_rownames %>%
  left_join(gene2gene_symbol, by=c("X"="query"))
F1_host_gene2gene_symbol <- 
    transform(
        F1_host_gene2gene_symbol ,
        Preferred_name =
            ifelse( is.na(Preferred_name), X, Preferred_name)) %>%
  mutate(Preferred_name = make.names(Preferred_name, unique = TRUE))
rownames(x=F1.host.data_names) <- F1_host_gene2gene_symbol$Preferred_name
head(rownames(F1.host.data_names))
head(rownames(F1.host.data))

### Initialize the Seurat objects with the raw, non-normalized data

F8 <- CreateSeuratObject(counts = F8.host.data_names, 
                             min.cells = 3, 
                             min.features = 200, 
                             project = "F8_host")
F1 <- CreateSeuratObject(counts = F1.host.data_names, 
                             min.cells = 3, 
                             min.features = 200, 
                             project = "F1_host")
### add metadata to objects

# add individual
F8[["individual"]] <- "A_2F8"
F8@meta.data$individual <- as.factor(F8@meta.data$individual)
F1[["individual"]] <- "S_2F1"
F1@meta.data$individual <- as.factor(F1@meta.data$individual)

# add symstate
F8[["symstate"]] <- "apo"
F8@meta.data$symstate <- as.factor(F8@meta.data$symstate)
F1[["symstate"]] <- "sym"
F1@meta.data$symstate <- as.factor(F1@meta.data$symstate)

# add genet
F8[["genet"]] <- "F"
F8@meta.data$genet <- as.factor(F8@meta.data$genet)
F1[["genet"]] <- "F"
F1@meta.data$genet <- as.factor(F1@meta.data$genet)

str(F8@meta.data)
str(F1@meta.data)

F8
#20848 features across 3824 samples within 1 assay 
#Active assay: RNA (20848 features, 0 variable features)

F1
#25112 features across 3113 samples within 1 assay 
#Active assay: RNA (25112 features, 0 variable features)
```

2. Doublet detection 
tutorials here: https://www.youtube.com/watch?v=ReXr4F1hhlo and https://biostatsquid.com/scdblfinder-tutorial/
using scDblFinder: https://plger.github.io/scDblFinder/articles/scDblFinder.html
```{r}
# Relevant paths
project_path <- results_dir #catcat_analysis2025
out_path <- file.path(project_path, 'results') # for all scRNAseq preprocessing results
doublet_folder <- file.path(out_path, 'Doublet_detection') # subfolder for Doublet detection results
#dir.create(doublet_folder, recursive = TRUE, showWarnings = FALSE) 

# Read in your Seurat objects
seuF8 <- F8
head(seuF8@meta.data)
seuF1 <- F1
head(seuF1@meta.data)

### Doublet detection of each sample
sceF8 <- as.SingleCellExperiment(seuF8)
sceF1 <- as.SingleCellExperiment(seuF1)

# scDblFinder
sceF8 <- scDblFinder(sceF8) #dbr = multiplet_rate)
table(sceF8$scDblFinder.class)
#singlet doublet 
#3549     275
sceF8@colData@listData %>% as.data.frame() %>% head()

sceF1 <- scDblFinder(sceF1) #dbr = multiplet_rate)
table(sceF1$scDblFinder.class)
#singlet doublet 
#2923     190 
sceF1@colData@listData %>% as.data.frame() %>% head()

# Explore results and add to seurat object
meta_scdblfinderF8 <- sceF8@colData@listData %>% as.data.frame() %>% 
  dplyr::select(starts_with('scDblFinder')) # 'scDblFinder.class')
head(meta_scdblfinderF8)
rownames(meta_scdblfinderF8) <- sceF8@colData@rownames
head(meta_scdblfinderF8)
seuF8 <- AddMetaData(object = seuF8, metadata = meta_scdblfinderF8 %>% dplyr::select('scDblFinder.class'))
head(seuF8@meta.data)
table(seuF8$scDblFinder.class)
#singlet doublet 
#3549     275
rm(list = c('meta_scdblfinderF8', 'sceF8'))

meta_scdblfinderF1 <- sceF1@colData@listData %>% as.data.frame() %>% 
  dplyr::select(starts_with('scDblFinder')) # 'scDblFinder.class')
head(meta_scdblfinderF1)
rownames(meta_scdblfinderF1) <- sceF1@colData@rownames
head(meta_scdblfinderF1)
seuF1 <- AddMetaData(object = seuF1, metadata = meta_scdblfinderF1 %>% dplyr::select('scDblFinder.class'))
head(seuF1@meta.data)
table(seuF1$scDblFinder.class)
#singlet doublet 
#2923     190 
rm(list = c('meta_scdblfinderF1', 'sceF1'))

# Doublet stats
# Check how doublets singlets differ in QC measures per sample.
VlnPlot(seuF8, split.by = "scDblFinder.class",
        features = c("nFeature_RNA", "nCount_RNA"), 
        ncol = 2, pt.size = 0) + theme(legend.position = 'right')
#saved as doublet_apoF8_qc
VlnPlot(seuF1, split.by = "scDblFinder.class",
        features = c("nFeature_RNA", "nCount_RNA"), 
        ncol = 2, pt.size = 0) + theme(legend.position = 'right')
#saved as doublet_symF1_qc

doublets_summaryF8 <- seuF8@meta.data %>% 
  group_by(scDblFinder.class) %>% 
  summarise(total_count = n(),.groups = 'drop') %>% as.data.frame() %>% ungroup() %>%
  mutate(countT = sum(total_count)) %>%
  group_by(scDblFinder.class, .add = TRUE) %>%
  mutate(percent = paste0(round(100 * total_count/countT, 2),'%')) %>%
  dplyr::select(-countT)
doublets_summaryF8
write.table(doublets_summaryF8, file = file.path(doublet_folder, paste0('scDblFinder_doublets_summaryF8.txt')), quote = FALSE, row.names = FALSE, sep = '\t')

doublets_summaryF1 <- seuF1@meta.data %>% 
  group_by(scDblFinder.class) %>% 
  summarise(total_count = n(),.groups = 'drop') %>% as.data.frame() %>% ungroup() %>%
  mutate(countT = sum(total_count)) %>%
  group_by(scDblFinder.class, .add = TRUE) %>%
  mutate(percent = paste0(round(100 * total_count/countT, 2),'%')) %>%
  dplyr::select(-countT)
doublets_summaryF1
write.table(doublets_summaryF1, file = file.path(doublet_folder, paste0('scDblFinder_doublets_summaryF1.txt')), quote = FALSE, row.names = FALSE, sep = '\t')

# Removing doublets 
seu_dbltF8 <- subset(seuF8, scDblFinder.class == "singlet")
table(seu_dbltF8$scDblFinder.class)
#singlet doublet 
#   3549       0 

seu_dbltF1 <- subset(seuF1, scDblFinder.class == "singlet")
table(seu_dbltF1$scDblFinder.class)
#singlet doublet 
#   2923       0 
```


3. Quality control and filtering of cat data
```{r}
obj.list <- list(seu_dbltF8, seu_dbltF1)

rownames(obj.list[[1]]) %>% as.data.frame() %>% View()

for (i in 1:length(obj.list)) {
  obj.list[[i]] <- PercentageFeatureSet(obj.list[[i]], pattern = "^MT-", col.name = 'percent.mt')
  obj.list[[i]][["per.mito.bin"]] <- ifelse(obj.list[[i]][["percent.mt"]] > 10, "> 10%", ifelse(obj.list[[i]][["percent.mt"]] > 5, "5%-10%", ifelse(obj.list[[i]][["percent.mt"]] > 2.5, "2.5%-5%", "< 2.5%")))
}

plts_qc <- lapply(obj.list, function(x) VlnPlot(x, 
                                             features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), 
                                             group.by = "individual",
                                             pt.size = .25,
                                             ncol = 3))

patchwork::wrap_plots(plts_qc, ncol = 1)
```

4. Perform integration of the two seurat objects, according to seurat v5
```{r}
F8_new <- obj.list[[1]]
F1_new <- obj.list[[2]]
obj.list_merge <- merge(F8_new, F1_new)
obj.list_merge
#An object of class Seurat 
#26343 features across 6472 samples within 1 assay 
#Active assay: RNA (26343 features, 0 variable features)
# 2 layers present: counts.F8_host, counts.F1_host


#following the "Perform analysis without integration" first:
# run standard anlaysis workflow
merge <- NormalizeData(obj.list_merge)
merge <- FindVariableFeatures(merge)
#merge <- ScaleData(merge, features = rownames(obj.list_merge@assays$RNA@features))

#scale with only host genes
features_to_scale <- rownames(obj.list_merge@assays$RNA@features)
#just scaling the first 20,719 features
features_to_scale <- features_to_scale[1:20719]
merge<- ScaleData(merge, features = features_to_scale)
merge <- RunPCA(merge)
merge <- FindNeighbors(merge, dims = 1:30, reduction = "pca")
merge <- FindClusters(merge, resolution = 2, cluster.name = "unintegrated_clusters")
#43 clusters; unintegrated and with a high res

merge <- RunUMAP(merge, dims = 1:30, reduction = "pca", reduction.name = "umap.unintegrated")
DimPlot(merge, reduction = "umap.unintegrated", group.by = c("individual", "seurat_clusters"))
#the individuals overlap really well

#"Perform integration"
merge <- IntegrateLayers(object = merge, method = CCAIntegration, orig.reduction = "pca", new.reduction = "integrated.cca",
    verbose = TRUE)
# re-join layers after integration
merge[["RNA"]] <- JoinLayers(merge[["RNA"]])

merge <- FindNeighbors(merge, reduction = "integrated.cca", dims = 1:30)
merge <- FindClusters(merge, resolution = 0.55, algorithm=1)
#28 at 0.55

merge <- RunUMAP(merge, dims = 1:30, reduction = "integrated.cca")

# Visualization
plt1merge <- DimPlot(merge, reduction = "umap", label = T, shuffle=TRUE)
plt2merge <- DimPlot(merge, reduction = "umap", group.by = "individual", shuffle=TRUE)
plt3merge <- DimPlot(merge, reduction = "umap", group.by = "symstate", shuffle=TRUE)
```

5. Identify cells with high percentage of symbiont reads
```{r}
counts <- GetAssayData(merge, assay = "RNA")
length(rownames(counts))
#26343
tail(counts)
counts_lim <- counts[1:20719,] #these are all the host genes

#get list of sym genes
counts_sym <-rownames(counts[20720:26343,])

#visualize cells with high percentage of sym genes
merge_sym_percent <- PercentageFeatureSet(merge, features = counts_sym, col.name = "sym_percent")
merge_sym_percent[["sym_perc_bin"]] <- ifelse(merge_sym_percent[["sym_percent"]] > 50, ">50%", "<50%")
sym_perc_bin <- DimPlot(merge_sym_percent, reduction = "umap", group.by = "sym_perc_bin", cols=c("grey80", "blue"), split.by = "symstate", order=c(">50%","<50%"))
#filter out the 45 cells (only in sym) that have over 50% symbiont reads
#saved as sym_perc_bin as landscape 8x4
#merge_nosym <- subset(merge_sym_percent, subset = sym_percent < 50)
# now remove sym genes all together
ultimate_merge <- subset(merge_sym_percent, features = rownames(counts_lim))
tail(rownames(ultimate_merge@assays$RNA))

#remove 13 cells with over 3000 features
ultimate_merge <- subset(ultimate_merge, subset = nFeature_RNA > 200 & nFeature_RNA < 3000)
```

# Cell type ID
1. Identify marker genes
```{r}
DefaultAssay(ultimate_merge) <- "RNA"
ultimate_merge@meta.data$seurat_clusters <- as.numeric(as.character(ultimate_merge@meta.data$seurat_clusters))+1
Idents(ultimate_merge)<- "seurat_clusters"
ultimate_merge$seurat_clusters
ultimate_merge@active.ident <- factor(x = ultimate_merge@active.ident, levels = 1:28)
ultimate_merge@active.ident

saveRDS(ultimate_merge, paste0(proj_name, '_ultimate_033025.rds'))
ultimate_merge <- readRDS(file.choose())

markers <- FindAllMarkers(ultimate_merge,
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          return.thresh = 0.01,
                          latent.vars = c('percent.mt', 'nFeature_RNA'))



markers <- markers %>%
  filter(avg_log2FC > 0)
nrow(markers)
#22993

markers_fullnames <- markers
rownames(markers_fullnames) =NULL
head(markers_fullnames)
head(gene)
col_names <- c("p_val", "avg_log2FC", "pct.1", "pct.2", "p_val_adj", "cluster", "Preferred_name")
colnames(markers_fullnames) <- col_names
markers_fullnames <- markers_fullnames %>%
  left_join(gene) %>%
  write_csv('./files/Oculina_ultimate_markers_033025_logFC.5.csv')

markers_fullnames <- read.csv(file.choose())
head(markers_fullnames)

markers_strong <- markers_fullnames %>%
  filter(p_val_adj<0.05) %>%
  filter(pct.1 > 0.1)

nrow(markers_strong)
#7376
write_csv(markers_strong, "./files/SuppDataset_oculina_markers_strong_033025.csv")
```

2. Use genes of interest identified in previous cnidarian datasets
```{r}
oculina_new <- ultimate_merge
plt1oc <- DimPlot(oculina_new, reduction = "umap", label = T, shuffle=TRUE)

####TF highly expressed in spist cnidocytes
pax6 <- c("PAX6", "PAX6.1", "PAX6.2") #PAX6 marker for 3, 13; PAX6.2 for 3, 10, 13, 14
six1 <- c("g3271") #ns
six2 <- c("SIX2") #ns

####TF highly expressed in spist gastrodermis etc
#foxc2 <- c("FXC2B")
deaf1 <- c("DEAF1", "DEAF1.1", "DEAF1.2") #spist gastro muscle; ns
#erg <- c("ERG") #all spist gastros
#fli1 <- c("FLII") #same spist as erg; ns #####RECHECK IF FLII OR FLI1
#etv6 <- c("ETV6","ETV6.1") #gastro algal hosting in spist
elf2 <- c("ELF2", "ELF2.1") #gastro algal hosting in spist, --> elf2 marker for 2, 13, 20
#elf4 <- c("ELF4", "ELF4.1", "ETS1A.3") #gastro algal hosting in spist --> 
spdef <- c("SPDEF") #gastro algal hosting in spist --> marker for 5, 10, 13
#usf2 <- c("USF2", "USF2.1") #in gastrodermis in spist
otx1 <-c("OTX1", "OTX1.1", "OTX1.2", "otx1", "otx1.1", "otx1.2") #gastro in spist, markers for 4 and 5
six4_6 <- c("SIX4", "SIX6") #ns

###TFs for neuron
#gata2 <- c("GATA2") #
pax7 <- c("PAX7", "PAX7.1") #ns
asc <- c("ASCC1", "ASCC2", "ASCC3") #ns
pou4 <- c("POU4F2") # marker for 13, 16, 28 (especially 28)
islet <- c("ICA1") #ns
tbx20 <- c("TBX20", "TBX20.1", "TBX20.2") #20 is marker 13, 20, 24, especially 20

#####TFs for Immune
irf1 <- c("IRF1", "IRF1,1", "IRF1.2") #1.2 marker for 15
irf2 <- c("IRF2BP2") #ns
irf4 <- c("IRF4", "IRF4.1") #markers for 15, added 04/23/25
nfat5 <- c("NFAT5") #marker gene for 2 but high in everything
#atf3 <- c("ATF3") #ns
atf5 <- c("ATF5") #marker for 11, 12, and 24 but in all cells

#####TFs for spist epidermis 
#pax2 <- c("PAX2", "PAX2A", "PAX2A.1", "PAX2A.2") 
#pax8 <- c("PAX8", "PAX8.1", "PAX8.2", "PAX8.3", "PAX8.4", "PAX8.5", "PAX8.6", "PAX8.7")
#bach <- c("BACH", "BACH.1") #ns
nfe2 <- c("NFE2L1") #marker for 4 but high in everything

####TFs for spist gland
rfx <- c("RFX3", "RFX6", "RFX7", "RFX4") #RFX6 marker for 11 and 22

#####TFs for calicoblast
dnajc21 <- c("DNAJC21") #ns
prdm9 <- c("PRDM9") #marker for 7 and 25; other versions of prdm9 that aren't markers; mostly 25
myb <- c("MYBBP1A", "MYBL1") #MYBL1 marker for 7
ncor <- c("NCOR1", "NCOR2", "NCOR2.1", "NCOR2.2", "NCOR2.3") #ncor1 marker for 2 but high in many clusters
ssrp1 <- c("SSRP1") #marker for 7
dmrt <- c("DMRT1", "DMRT3", "DMRTA1", "DMRTA2", "DMRTA2.1", "DMRTC2") #2.1 marker for 3 and 4 but weak
#csd <- c("CSD") #i think this is supposed to be a cysteine desulfarase
atf <- c("crebzf", "CREBL2") #not sure about the orthos here, supposed to be orthos to ZHANG (CREB/ATF bZIP transcription factor) and CREM (cAMP-responsive element modulator) resp, ns
dbp <- c("HLF", "HLF.1") #HLF.1 marker for 6 and 16; also in others
##check the dbp for further orthos of "D site-binding protein" above and for tef (transcription factor vbp)
atf2 <- c("ATF2") #marker for 28
###forkhead (fox) 

###random genes of interest from spist paper
#cnidocyte (minicollagen but we don't have that)
tal2 <- c("g25145") #tachylectins or galactosamine binding lectins, there are other tachylectins as well, but this is marker for 8 and 18 (and a bit in 23)

#####immune
#alps <- c("ALPS", "ALPS.1") #Anti-lipopolysaccharide factor, immune cell linked in spistillata; also 4
#ikka <- c("IKKA") #
#ikbp1 <- c("IKBP1") #
ilf2 <- c("ILF2") #marker for 25
hsp70 <- c("g30303") #marker for 15, but there are other heat shock 70 proteins in the gene list that aren't markers
hsp90 <- c("HSP90B1") #marker for 1 and 5
#alphaA <- c("L2EFL.1") #alpha-crystallin A
alphaB <- c("CRYAB", "CRYAB.1", "CRYAB.2") #alpha-crystallin B, .1 is marker for 1, 11, 12
cathepsin <- c("CTSZ") #cathepsin z marker for 1, 5, 15; missing some cathepsins for sure
bcl2 <- c( "BCL2") #bcl2 is marker for 2 and 4
#additional genes from spist used on 04/23/25
perforin <- c("MPEG1", "MPEG1.3") #there are many MAC/Perforin domain genes, but these are markers for 15; MPEG1 is a good one
caspase <- c("g19217") #marker for 15, this is peptidase C14 caspase catalytic subunit p20 --> nah
hsps <- c("Hsp68", "g14732", "HSPA4L") #these are other hsp markers for 15; hsp68 is a good one
rab <- c("RAB30.2") #marker for 15, not rab43 as in spist but still a ras oncogene --> nah
interferon <- c("g6378")#Interferon induced with helicase C domain 1, marker for 15 --> yeah lets keep this one
nacht <- c("NLRP3.7")# is a NACHT, LRR and PYD domains-containing protein 3-like, marker for 15; keep for now
duf4773 <- c("g10019")#Domain of unknown function (DUF4773) g10019 is marker for 15 --> keep
####other random immune genes
nfkb1 <- c("NFKB1", "NFKB1.1", "NFKB1.2") #1.1 is marker for 4; description is weird but pfams are correct
my881 <- c("MYD88", "MYD88.1", "MYD88.2") #MYD88.2 is marker for 2, 4, and 7
bcl3 <- c("BCL3", "BCL3.1") #ns
lbp <- c("LBP") #marker for 4 and 15
#prosaposin <- c("SAP") # marker for 1, 4
endop <- c("K123") #endonuclease domain containing protein; idk if TF; there are 11 possible genes, but this one is the only marker and it's a marker for 15
cd36 <- c("SCARB2", "SCARB2.1", "SCARB2.2") #ns but maybe there are others
#sting <- c("STING", "STING.1", "STING.2", "STING.3", "STING.4", "STING.5") #ns
ifih1 <- c("IFIH1") #ns interferon-induced helicase c domain-containing protein
dock <- c("DOCK1", "DOCK1.1", "DOCK11", "DOCK6") #dedicator of cytokinesis; markers for 2
#gbp6 <- c("GBP6.6", "GBP6", "GBP6.2") #guanylate binding protein 6; I don't have 6 but I do have GBP5 and it's a marker for 1, 4, and 16
rab43 <- c("RAB43", "RAB43.1") #ns
bag <- c("BAG3", "BAG4.1", "BAG4") #BAG family molecular chaperone regulator 3 and 4.1 (ns) BAG4 is described as something else but PFAM is right, marker for 4, 5, and 15
cflar <- c("CFLAR", "CFLAR.1", "CFLAR.2", "CFLAR.3", "CFLAR.4") #apoptosis regulators; markers for 2, 3, and 15
apoptosis <- c("BFAR", "BAX", "SIVA1") #other apoptosis regulators; AR1 included in apoculata but not here; ns
#casp3 <- c("CASP3.2") #no caspase 3s, odd
mvp <- c("MVP") #major vault protein; marker for 1, 3, 5 and 15
ttc28 <- c("TTC28", "TTC28.1", "TTC28.2", "TTC28.3", "TTC28.4") #Tetratricopeptide repeat protein 28; ns

####gastro
tropomyosin <- c("Tm1", "g8969", "g21117", "g24753") #markers for many (1, 4, 5, 6, 12, 15)
apoB <- c("g8256") #apolipoprotein, there are many, but this is marker for 1, 2, 4, and 5
#aplp <- c("APLP") #apolipophorin
galk <- c("GALK1") #galactokinase; marker for 1
carb <- c("CA13", "CA13.2", "CA9.1", "CA9.2", "CA9.3", "CA9.4", "Car15") #carbonic anhydrase; high exp in algal-hosting cells in spist; many; these are markers for 1, 2, 3, 9, 10, 14, 17, 18
npc2 <- c("NPC2", "NPC2.1", "NPC2.2", "NPC2.3", "NPC2.4") #markers for 1, 5, and 15
#others to note some of which are markers: granulins (1, 2, 4, 5)
#####S2611 (transporter, didn't find here)
#lysosome associated things (LAMP1 in 1, 5, 15)
#####GRP78 (glucose-reg, didn't find here)
#####PLCL.5 and .1 (didn't find plcl here)
#glutamic acid-rich proteins Sh3bgrl3 and Sh3bgrl3.1 (1, 6, and 15)
#CMLO2.3 and NAT8L (acetyltransf., couldnt find CMLO2 but NAT8L in 4)
#acid phosphatase (genes for this in 1, 4, 5, 17, and 23)

#####mitotic/germline
#look for cyclins and tubulins
sy65 <- c("SYT16.3", "SYT16") #synaptotagmin, mitotic marker in spist; in 3, 10, 14, 16, other synapt genes present in genome

#####epidermis
pxdn <- c("PXDN", "PXDN.14") #there are many pxdns, these are markers for 1, 2 (PXDN) and 10 (PXDN.14)
moxd1 <- c("MOXD1.20", "MOXD1.14") #DBH-like monooxygenase protein, markers for many; lots of these genes; including markers for 1, 2, 3, 4, 5, 6, 11, 13, 15, 16, 22
gst1 <- c("MGST1.2", "MGST3") #maybe some other GST possibilities too, but these are mgst markers for 1, 5, 6, and 11
cadh <- c("FAT4", "FAT4.2", "FAT4.5") #FAT4s in 1, 2, 5, 13, 16; no fat2s; no pcdl
#more genes: CYP2U1 (monooxygenase activity, marker for 3, 10, 14); GstS1 (Glutathione S-transferase, N-terminal domain; marker for 1, 9, 12, 15, 17); SPON1 (spondin, marker for 3, 10, 14, 18); TSPAN4 (one example of tetraspanin; this is marker for 9 and 17)

#####neuron
pkd1 <- c("PKD1L2.10", ) #many pkd1 genes; PKD1L2.10 is marker for 12, 20, 24 (there's also PKD1L1 but it seems like maybe different thing and PKD1L is the correct description - polycystic kidney disease protein 1)
dclk2 <- c("DCLK2") #there's also DCLK1 and 3, maybe different? DCLK2 is marker for 11
cel3b <- c("CELA3B.1", "CELA2A.1", "CELA2A") #chymotrypsin-like elastase family member 3B, no "cel3b" but these seem right (thought CELA2A isn't described the same); markers for 6, 16, 20, 23, 24
cabp5 <- c("g29794") #calcium binding protein 5, calcium gated channel (neuron-y); g29794 might be cabp5; there are also other other calcium binding proteins that are markers for 1, 2, 3, 5, 9, 10, 16, 17, 28
trpa <- c("TRPA1.22", "TRPA1.11", "TRPA1.9") #trpas are markers 10, 11, 14, 20, 23, 25
atoh8 <- c("ATOH8", "ATOH8.1") #neuron marker in spist, ns
kcn <- c("KCNA2.6") #potassium voltage gated channel thing, many KCNAs, no KCNFs; KCNA2.6 marker for 13 and 16
cbpd <- c("CPB2.2", "CPB1", "CPB2.1", "CPB2.4", "CPB2.3") #carboxypeptidase, there are many carboxypeptidases, though not annotated as cbp... these are markers for 6 (first) and 12 (all others)
#lwa <- c("LWA", "LWA.1") #ns
chymo <- c("g26650", "g206") #all could be chymptrypsin homologs; ns
#cabo <- c("CABO.1") #no squidulin
calm <- c("g12975", "g12976", "g12979", "g1979") #many calmodulins, and many markers, including for 3, 6, 7, 8, 11, 12, 13, 16, 18, 19, 20, 23
cac1 <- c("GABBR2") #marker for many; voltage-dep calcium channels; marker for 20; there are also many other voltage-gated or voltage-sensitive calcium channels that are markers for 6, 8, 12, 13, 16, 20
#scn <- c("SCN2A") #sodium channel possibilities

##gland
olm2 <- c("OLFML2B.1", "OLFML2B") #olfactomedin, markers 22
#muc5a <- c("MUC5A") #no mucin 5
muc4 <- c("g18665", "g18731", "g18665") #but there are mucin 4s that are markers for 3, 10, 11, 14
mlp <- c("EMR1") #no mlps but there are mucin-like hormone receptors, this one is marker for 20
#no mlrp integumentary mucins
#lots of fibrinogen in 22 especially (also some in other clusters)
#creb3l2s and creb3l3s in 8, 12, 18, 22, 24
#FKBP-type peptidyl-prolyl cis-trans isomerases (FKBP and others) in 1, 2, 3, 5, 8, 18, 19, 23, 24, 28
#carbohydrate sulfotransferase: GAL3ST1.5 is a marker for 26 and is a galactose-3-O-sulfotransferase 1
#Astacin is marker for 22, 26 and others
#collagen marker for 19, 22, and others
#tyrosinase marker for 19, 22, and others

####calicoblast
hemopexin <- c("eprA1") #not 100% sure this is hemopexin, but also ns 
#myof <- c("MYOF","MYOF.5") #no myoferlines/dysferlines
adam <- c("ADAM8.2") #many ADAMs; including for 8 and 27; metallopeptidase
#hemicentin is in many cells (1, 2, 4, 5, 15, 25) and there are many orthologs
#sodium bicarbonate transporter (SLC4A10) in 9
```

FINISH THIS AFTER WED
Predictions: 
Gastro:
- 1 (tropomyosin, apoB, galk, carb, npc2, granulins, LAMP1, glutamic acid-rich proteins, acid phosphatase)
- 2 (elf2, apoB, carb, granulins)
- 3 (carb)
- 4 (otx1, tropomyosin, apoB, granulins, NAT8L, acid phosphatase)
- 5 (spdef, otx1, tropomyosin, apoB, npc2, granulins, LAMP1, acid phosphatase)
    - 6 (tropomyosin, glutamic acid-rich proteins)
- 7
    - 9 (carb)
    - 10 (spdef, carb)
    - 12 (tropomyosin)
    - 13 (elf2, spdef)
    - 14 (carb)
    - 15 (tropomyosin, npc2, LAMP1, glutamic acid-rich proteins)
    - 17 (carb, acid phosphatase)
    - 18 (carb)
    - 20 (elf2)
    - 23 (acid phosphatase)

Epidermis:
    - 1 (pxdn, moxd1, gst1, cadh, GstS1)
    - 2 (pxdn, moxd1, cadh)
- 3 (moxd1, CYP2U1, SPON1)
    - 4 (nfe2, moxd1)
    - 5 (moxd1, gst1, cadh)
    - 6 (moxd1, gst1)
- 9 (GstS1, TSPAN4)
- 10 (pxdn, CYP2U1, SPON1)
    - 11 (moxd1, gst1)
    - 12 (GstS1)
    - 13 (moxd1, cadh)
- 14 (CYP2U1, SPON1)
    - 15 (moxd1, GstS1)
    - 16 (moxd1, cadh)
- 17 (GstS1, TSPAN4)
    - 18 (SPON1)
    - 22 (moxd1)

Neuron
    - 1 (cabp5)
    - 2 (cabp5)
    - 3 (cabp5, calm)
    - 5 (cabp5)
- 6 (cel3b, cbpd, calm)
    - 7 (calm)
    - 8 (calm)
    - 9 (cabp5)
    - 10 (cabp5, trpa)
- 11 (dclk2, trpa, calm)
- 12 (pkd1, cbpd, calm)
- 13 (pou4, tbx20, kcn, calm)
    - 14 (trpa)
- 16 (pou4, cel3b, cabp5, kcn, calm)
    - 17 (cabp5)
    - 18 (calm)
    - 19 (calm)
- 20 (tbx20, pkd1, cel3b, trpa, calm, cac1)
- 23 (cel3b, trpa, calm)
- 24 (tbx20, pkd1, cel3b)
    - 25 (trpa)
- 27
- 28 (pou4, cabp5)
   

Immune
- 15 (irf1, irf4, hsp70, cathepsin, LBP, endop, BAG4, cflar, mvp)
04/23/25: MPEG1, Hsp68, g6378 (interferon guy), NLRP3.7, IRF4 (good), IRF4.1

Gland
    - 1 (FKBP)
    - 2 (FKBP)
    - 3 (muc4, FKBP)
    - 4 (fibrinogen)
    - 5 (fibrinogen, FKBP)
    - 8 (CREB3L3, CREB3L2, FKBP)
    - 10 (muc4)
    - 11 (rfx, fibrinogen)
    - 12 (CREB3L2)
    - 13 (fibrinogen)
    - 14 (muc4)
    - 15 (fibrinogen)
    - 16 (fibrinogen)
    - 18 (CREB3L3, CREB3L2, FKBP)
- 19 (FKBP, collagen, tyrosinase)
    - 20 (mlp)
- 22 (rfx, olm2, lots of fibrinogen, CREB3L2, Astacin, collagen, tyrosinase)
    - 23 (FKBP)
    - 24 (fibrinogen, CREB3L3, CREB3L2, FKBP)
- 26 (GAL3ST1.5, Astacin)
    - 28 (FKBP)
   
Calicoblast
    - 8 (adam)
    - 9 (SLC4A10)
- 25 (hemicentin, CUB domain-containing protein)
    - 27 (adam)
  
Cnidocyte:
- 8 (tal2)
- 18 (tal2)


1 [Gastrodermis]
2 [Gastrodermis]
3 [Epidermis] 
4 [Gastrodermis]
5 [Gastrodermis]
6 [Neuron] 
7 [Gastrodermis]
8 [Cnidocyte]
9 [Epidermis] 
10 [Epidermis]
11 [Neuron]
12 [Neuron]
13 [Neuron]
14 [Epidermis]
15 [Immune Cell]
16 [Neuron]
17 [Epidermis]
18 [Cnidocyte]
19 [Gland]
20 [Neuron]
21 [Algal-Hosting]
22 [Gland]
23 [Neuron]
24 [Neuron]
25 [Calicoblast]
26 [Gland]
27 [Neuron]
28 [Neuron]

3. look at gene expression heatmap to verify
```{r}
oc_markers_pre_new<- FindAllMarkers(oculina_new, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
nrow(oc_markers_pre_new)
#6091

oc_markers_pre_new %>% group_by(cluster) %>% top_n(20, avg_log2FC) -> top20_pre_new
top20_pre_new
nrow(top20_pre_new)
#543

color<-colorRampPalette(brewer.pal(n=9, name="Reds"),bias=0.5)(20)

DoHeatmap(oculina_new, features = top20_pre_new$gene, label=T) + theme(axis.text.y= element_text(size=0)) + scale_fill_gradientn(colors=color)
#looks good but some of the neurons and glands might switch, I could consider changing the ones I'm not sure about to just "secretory cell" or all the glands + the weirdos to "secretory cell"
```

4. Rename cell clusters
```{r}
oculina_named_new <- RenameIdents(oculina_new, "1" = "Gastrodermis 1")
oculina_named_new <- RenameIdents(oculina_named_new, "2" = "Gastrodermis 2")
oculina_named_new <- RenameIdents(oculina_named_new, "3" = "Epidermis 1")
oculina_named_new <- RenameIdents(oculina_named_new, "4" = "Gastrodermis 3")
oculina_named_new <- RenameIdents(oculina_named_new, "5" = "Gastrodermis 4")
oculina_named_new <- RenameIdents(oculina_named_new, "6" = "Neuron 1")
oculina_named_new <- RenameIdents(oculina_named_new, "7" = "Gastrodermis 5")
oculina_named_new <- RenameIdents(oculina_named_new, "8" = "Cnidocyte 1")
oculina_named_new <- RenameIdents(oculina_named_new, "9" = "Epidermis 2")
oculina_named_new <- RenameIdents(oculina_named_new, "10" = "Epidermis 3")
oculina_named_new <- RenameIdents(oculina_named_new, "11" = "Neuron 2")
oculina_named_new <- RenameIdents(oculina_named_new, "12" = "Neuron 3")
oculina_named_new <- RenameIdents(oculina_named_new, "13" = "Neuron 4")
oculina_named_new <- RenameIdents(oculina_named_new, "14" = "Epidermis 4")
oculina_named_new <- RenameIdents(oculina_named_new, "15" = "Immune Cell")
oculina_named_new <- RenameIdents(oculina_named_new, "16" = "Neuron 5")
oculina_named_new <- RenameIdents(oculina_named_new, "17" = "Epidermis 5")
oculina_named_new <- RenameIdents(oculina_named_new, "18" = "Cnidocyte 2")
oculina_named_new <- RenameIdents(oculina_named_new, "19" = "Gland 1")
oculina_named_new <- RenameIdents(oculina_named_new, "20" = "Neuron 6")
oculina_named_new <- RenameIdents(oculina_named_new, "21" = "Algal-Hosting")
oculina_named_new <- RenameIdents(oculina_named_new, "22" = "Gland 2")
oculina_named_new <- RenameIdents(oculina_named_new, "23" = "Neuron 7")
oculina_named_new <- RenameIdents(oculina_named_new, "24" = "Neuron 8")
oculina_named_new <- RenameIdents(oculina_named_new, "25" = "Calicoblast")
oculina_named_new <- RenameIdents(oculina_named_new, "26" = "Gland 3")
oculina_named_new <- RenameIdents(oculina_named_new, "27" = "Neuron 9")
oculina_named_new <- RenameIdents(oculina_named_new, "28" = "Neuron 10")

levels(oculina_named_new)
levels(oculina_named_new) <- c("Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gastrodermis 4", "Gastrodermis 5", "Algal-Hosting", "Epidermis 1", "Epidermis 2", "Epidermis 3", "Epidermis 4", "Epidermis 5", "Neuron 1", "Neuron 2", "Neuron 3", "Neuron 4", "Neuron 5", "Neuron 6", "Neuron 7", "Neuron 8", "Neuron 9", "Neuron 10", "Gland 1", "Gland 2", "Gland 3", "Calicoblast", "Cnidocyte 1", "Cnidocyte 2", "Immune Cell")
levels(oculina_named_new)

DefaultAssay(oculina_named_new) <- "RNA"
saveRDS(oculina_named_new, paste0(proj_name, '_oculinanamed_2025.rds'))
oculina_named_new <- readRDS(file.choose())

#install.packages("simplecolors")
library(simplecolors)

gastro_col <- colorRampPalette(c("#C7B8E6", "#330099"))
gdc <- gastro_col(5) 
gdc6 <- c("#C7B8E6", "#A28AD2", "#7D5CBF", "#582EAC", "#330099", "#000066")
epi_col <- colorRampPalette(c("#B6D8B6", "#006600"))
ec <- epi_col(5) 
ec5 <- c("#B6D8B6", "#88BB88", "#5B9F5B", "#2D822D", "#006600")
neuro_col <- colorRampPalette(c("#B8DEE6", "#004499")) #006699 was original
nc <- neuro_col(10)
nc10 <- c("#B8DEE6", "#A3CCDD", "#8FBBD4", "#7AAACC", "#6699C3", "#5188BB", "#3D77B2", "#2866AA", "#1455A1", "#004499")
gland_col <- colorRampPalette(c("#E6B8BF","#9C1616"))
gc <- gland_col(3)
gc3 <- c("#E6B8BF", "#C1676A", "#9C1616")

cols <- c("#C7B8E6", "#A28AD2", "#7D5CBF", "#582EAC", "#330099", "#000066", "#B6D8B6", "#88BB88", "#5B9F5B", "#2D822D", "#006600", "#B8DEE6", "#A3CCDD", "#8FBBD4", "#7AAACC", "#6699C3", "#5188BB", "#3D77B2", "#2866AA", "#1455A1", "#004499", "#E6B8BF", "#C1676A", "#9C1616", "#996633", "#FF9900", "#FFCC66", "#333333")

plot1 <- DimPlot(oculina_named_new, reduction = "umap", label=F, label.size = 3, cols=cols, alpha=0.5) 
LabelClusters(plot1, id = "ident", size = 3, repel = T,  box.padding = 0.25, fontface="bold") + theme(legend.text=element_text(size=10))
#umap_oculina_2025 as landscape 8.5x6

plot2 <- DimPlot(oculina_named_new, reduction = "umap", label=F,  shuffle=TRUE, group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))
#umap_oculina_aposym_2025 as landscape 6.5x5.5
```

5. Look at DEGs between apo and sym at whole dataset level
```{r}
#make a new seurat with 1 added to the counts
oc_diff <- oculina_named_new
DefaultAssay(oc_diff) <- "RNA"
oc_diff[["RNA"]]$counts<-as.matrix(oc_diff[["RNA"]]$counts)+1

oc_full_aposym_mark <- FindMarkers(oc_diff, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(oc_full_aposym_mark)

oc_full_aposym_mark <- rownames_to_column(oc_full_aposym_mark, var="Preferred_name")
head(oc_full_aposym_mark)

oc_DE_05 = subset(oc_full_aposym_mark, p_val_adj < 0.05)
nrow(oc_DE_05)
#407

oc_DE_10 = subset(oc_full_aposym_mark, p_val_adj < 0.1)
nrow(oc_DE_10)
#431

go_input_full = oc_full_aposym_mark %>%
  mutate(mutated_p = -log(p_val)) %>%
  mutate(mutated_p_updown = ifelse(avg_log2FC < 0, mutated_p*-1, mutated_p*1)) %>%
  na.omit() %>%
  select(Preferred_name, mutated_p_updown)
head(go_input_full)
nrow(go_input_full)
#11721

#go_input_full2 <- go_input_full %>%
#  left_join(seq2iso2gene) %>%
#  select(mutated_p_updown, Protein_ID) %>%
#  na.omit()
#nrow(go_input_full2)
#12619
colnames(go_input_full) <- c("Preferred_name", "pval")
head(go_input_full)
#go_input_full2 <- go_input_full2[, c(2, 1)]
write.csv(go_input_full, file="./files/go_input_full2025.csv", quote=F, row.names=FALSE)

head(annotations_filt)
```

GO TO SCRIPT GO_MWU_for_SC_Oculina2025 TO RUN GO MWU ANALYSES

6. Explore GO terms of interest

Reformat GO BP output tables
```{r}
go_bp_all <- read.csv("./files/MWU_BP_go_input_full2025.csv", header=TRUE, sep=" ")
head(go_bp_all)
write.table(go_bp_all, "./files/MWU_BP_go_input_full2025.txt", quote=F, sep="\t", row.names=FALSE)
```

```{r}
bp_all = read.csv("./files/BP_go_input_full2025.csv", sep="\t")
head(bp_all)
colnames(bp_all)[4]="Preferred_name"
```


Now filter your GO BP files for the GO terms of interest

Pull out all immunity signals from the GO BP (total 845 significant BP terms)
- pulling out any that say "immune" "antigen" or anything related to nfkb (TLRs)
- not doing "stress", ros, or "dna damage" stuff (including a cellular response to heat term), not including wound healing
- apoptotic: GO:2001236; GO:1902041 and GO:1902042 (apop via death domain) and there are a couple other apoptotic terms
- GO:0038093 is Fc receptor signaling ?
- defense response GO:0045089;GO:0031349
- cell killing GO:0001906;GO:0001909
- leukocyte proliferation GO:0032944;GO:0070663;GO:0050670 and migration GO:0050900 and chemotaxis GO:0002688;GO:0002690

yes interleukin, yes toll-like receptor, tgf beta, nfkb, "immune," "antigen," "cytokine"

Antigen: GO:0002474|GO:0042590|GO:0002479
General "Immune": GO:0050778|GO:0045087|GO:0002253|GO:0002757|GO:0002764|GO:0002682|GO:0002683|GO:0002768|GO:0002251|GO:0002385|GO:0002821|GO:0002824
More specific "Immune" things: GO:0002431|GO:0002433|GO:0038094|GO:0038096|GO:0002429

Toll-like receptor: GO:0034123|GO:0034121
Interleukin: GO:0032717|GO:0032677|GO:0032757|GO:0038111|GO:0032663|GO:0070670|GO:0071353
NF-kappaB: GO:0043123|GO:1901222
TGFbeta: GO:0071560|GO:0071559|GO:0030511

GO:0002474|GO:0050778|GO:0045087|GO:0002253|GO:0002757|GO:0002764|GO:0002431|GO:0002433|GO:0038094|GO:0038096|GO:0034123|GO:0002682|GO:0032717|GO:0002683|GO:0002429|GO:0002768|GO:0032677|GO:0032757|GO:0038111|GO:0034121|GO:0043123|GO:0071560|GO:0071559|GO:0042590|GO:0002479|GO:1901222|GO:0032663|GO:0070670|GO:0071353|GO:0002251|GO:0002385|GO:0002821|GO:0002824|GO:0030511
```{r}
rownames_oc_new <- rownames(GetAssayData(oculina_named_new))
rownames_oc_df <- as.data.frame(rownames_oc_new)

immunebroad_all = bp_all %>%
  filter(str_detect(term, "GO:0034123|GO:0032717|GO:0032677|GO:0032757|GO:0038111|GO:0034121|GO:0043123|GO:0071560|GO:0071559|GO:1901222|GO:0032663|GO:0070670|GO:0071353|GO:0030511")) %>% # select desired GO terms (immune)
  left_join(oc_full_aposym_mark) %>% 
  drop_na() %>%
  distinct(Preferred_name, .keep_all = TRUE) %>%
  #left_join(gene) %>%
  column_to_rownames(var = "Preferred_name") %>%
  dplyr::select(-name, -term, -lev, -value) %>%
  dplyr::select(sort(tidyselect::peek_vars()))
head(immunebroad_all)
nrow(immunebroad_all)
#255 --> using only EXPLICITLY Toll-like receptor, Interleukin, NF-kappaB, and TGFbeta GO terms

symapo_immbroad_all <- rownames(immunebroad_all)

symapo_immbroad_df <- as.data.frame(symapo_immbroad_all)

oc_new_overlap_immbroad <- subset(symapo_immbroad_df, symapo_immbroad_all %in% rownames_oc_df$rownames_oc_new)

oc_diff <- PercentageFeatureSet(oculina_named_new, features = oc_new_overlap_immbroad$symapo_immbroad_all, col.name = "immunebroad_percent")
oc_diff[["immunebroad_perc_bin"]] <- ifelse(oc_diff[["immunebroad_percent"]] > 4.5, ">4.5%", "<4.5%")

immbroad_perc_bin <- DimPlot(oc_diff, reduction = "umap", group.by = "immunebroad_perc_bin", cols=c("grey80", "blue"), split.by = "symstate", order=c(">4.5%","<4.5%"))

immbroad_perc_cont <- FeaturePlot(oc_diff, reduction = "umap", features="immunebroad_percent", order=TRUE, min.cutoff = 4.5, max.cutoff = 7, split.by = "symstate") + theme(legend.position = c(0.98,0.98)) 
#saved as immbroad_perc_cont_all2025 as 8x4

immbroad_perc_vln <- VlnPlot(oc_diff, features = "immunebroad_percent", split.by = "symstate")
###percent of cells that are above cutoff in apo and sym
```

7. Top 10 and top 20 genes per cluster
```{r}
#get the top10 from the re-leveled oculina_named seurat object
oc_markers_lvld_new<- FindAllMarkers(oculina_named_new, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
nrow(oc_markers_lvld_new)
#6091

oc_markers_lvld_new %>% group_by(cluster) %>% top_n(10, avg_log2FC) -> top10_lev_new
top10_lev_new
nrow(top10_lev_new)
#273
#gives the top 10 genes for each cluster; 28 clusters
saveRDS(top10_lev_new, paste0(proj_name, '_top10_releveled_named_040325.rds'))

#do top20
oc_markers_lvld_new %>% group_by(cluster) %>% top_n(20, avg_log2FC) -> top20_lev_new
top20_lev_new
nrow(top20_lev_new)
#543
saveRDS(top20_lev_new, paste0(proj_name, '_top20_releveled_named_040325.rds'))

DoHeatmap(oculina_named_new, features = top10_lev_new$gene,label=F, group.colors = cols) + theme(axis.text.y= element_text(size=0)) + scale_fill_gradientn(colors=color)
#saved as top10_releveled_named_2025 as landscape 9x6

top20_hm <- DoHeatmap(oculina_named_new, features = top20_lev_new$gene,label=F, group.colors = cols) + theme(axis.text.y= element_text(size=0)) + scale_fill_gradientn(colors=color)
#saved as top20_releveled_named_2025 as landscape 9x6
```

# Analyze specific cell clusters

1. Gastrodermis 1 0.25 res
```{r}
gderm1 <- subset(oculina_named_new, idents = "Gastrodermis 1") 

gderm1 <- FindVariableFeatures(gderm1)
gderm1 <- ScaleData(gderm1, features = features_to_scale)
gderm1 <- RunPCA(gderm1)
gderm1 <- FindNeighbors(gderm1, dims = 1:30, reduction = "pca")
gderm1 <- FindClusters(gderm1, resolution = 0.25)
# 3 clusters
gderm1 <- RunUMAP(gderm1, reduction="pca", dims=1:30, return.model=TRUE)

#gderm1 <- RunUMAP(gderm1, dims = 1:30, reduction = "integrated.cca")

gderm1@meta.data$seurat_clusters <- as.numeric(as.character(gderm1@meta.data$seurat_clusters))+1
Idents(gderm1)<- "seurat_clusters"
gderm1$seurat_clusters
gderm1@active.ident <- factor(x = gderm1@active.ident, levels = 1:3)
gderm1@active.ident

levels(gderm1) <- c("1", "2", "3")

plot5 <- DimPlot(gderm1, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5)
# 3 clusters

plot6 <- DimPlot(gderm1, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot6b <- DimPlot(gderm1, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gderm1_plot5plot6_2025 9x4
#umap_gderm1_plot6b_2025 8x4 landscape
```

```{r}
gderm1_topfeat <- TopFeatures(object = gderm1[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gderm1[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gd1 <- FetchData(gderm1, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gd1 <- FetchData(gderm1, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gd1, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gd1, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as gderm1_pcplots_umap_2025

stdev_gd1 <- Stdev(gderm1, reduction = "pca")

pca_gd1 <- pc_data_gd1
pca_gd1$sym_state <- gderm1@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gd1)) #1.04e-14 ***
summary(aov(PC_2 ~ sym_state, data = pca_gd1)) #<2e-16 ***
summary(aov(PC_3 ~ sym_state, data = pca_gd1)) #0.0261 *
summary(aov(PC_4 ~ sym_state, data = pca_gd1)) #<2e-16 ***
summary(aov(PC_5 ~ sym_state, data = pca_gd1)) #0.381

summary(aov(PC_6 ~ sym_state, data = pca_gd1)) #0.0421 *
summary(aov(PC_7 ~ sym_state, data = pca_gd1)) #2.86e-05 ***
summary(aov(PC_8 ~ sym_state, data = pca_gd1)) #0.239
summary(aov(PC_9 ~ sym_state, data = pca_gd1)) #0.443
summary(aov(PC_10 ~ sym_state, data = pca_gd1)) #0.000365 ***

DefaultAssay(gderm1) <- "RNA"
gderm1[["RNA"]]$counts<-as.matrix(gderm1[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gderm1_mark <- FindMarkers(gderm1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gderm1_mark)
#neg lfc means down in sym; pos lfc means up in sym

gderm1_mark <- rownames_to_column(gderm1_mark, var="Preferred_name")

#look at subcluster marker genes
gderm1_markers_subclust <- FindAllMarkers(gderm1, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gderm1_markers_subclust
gderm1_markers_subclust <- rownames_to_column(gderm1_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of gderm1
gderm1_1 <- subset(gderm1, idents="1")
DefaultAssay(gderm1_1) <- "RNA"

gderm1_1_mark <- FindMarkers(gderm1_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm1_1_mark <- rownames_to_column(gderm1_1_mark, var="Preferred_name")
head(gderm1_1_mark)

#get apo vs sym differences for subcluster 2
gderm1_2 <- subset(gderm1, idents="2")
DefaultAssay(gderm1_2) <- "RNA"

gderm1_2_mark <- FindMarkers(gderm1_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm1_2_mark <- rownames_to_column(gderm1_2_mark, var="Preferred_name")
head(gderm1_2_mark)

#for 3
gderm1_3 <- subset(gderm1, idents="3")
DefaultAssay(gderm1_3) <- "RNA"
#less than 3 apos
```

```{r}
#get genes from signficant PCs (1,2,3,4,6,7,10)
print(gderm1[["pca"]], dims = 1:10, nfeatures = 10)

gderm1_feats <- c("COL2A1", "g21999", "g20264", "g20254", "g22237", "TAGLN3", "CSRP1.1", "COL4A1", "g24753", "CALML3", "g11768", "g24461", "BSG.1", "ACOT4.2", "ASL", "HNF4A", "g24200", "ACSBG1", "GBA", "g17216", "g11768", "g24461", "BSG.1", "ACOT4.2", "ASL", "HNF4A", "g24200", "ACSBG1", "GBA", "g17216", "g1806", "CRYAB.1", "g17871", "IFI30.1", "g804", "CTSZ", "g26750", "g14743", "BAAT", "g3423", "EDIL3.9", "g4358", "WNT4", "ARHGEF3", "HECTD1.1", "ACAN", "g21831", "g28270", "g2313", "otx1", "g22615", "g23828", "ATP5G2", "OST4", "RPL35A", "g24422", "HEBP2", "g1985", "g11700", "CSTB", "g1806", "CTSZ", "g25838", "PRCP", "Mmp1.16", "HSPA5", "g5331", "IFI30.1", "CTSE", "GRN.2", "CSRP1.1", "g15701", "MYL12A.1", "g30147", "TAGLN3", "g21999", "unc.54", "g12979", "MYL6B.1", "TMSB10", "HMCN1.13", "ADAM29.3", "g3990", "HMCN1.14", "PRSS8", "WNT7A", "g23828", "g24760", "g5318", "g5331", "AHCY", "g18453", "NASP", "TIMELESS", "g12827", "MCM4", "g30147", "NOLC1", "MAT1A", "HELLS", "g24753", "g24422", "TBX10", "FK506.bp2", "AHCY", "g20240", "CALML3", "ZC2HC1C", "g22237", "ACTG1.2", "MYL12A.1", "MYL6B.1", "g12979", "g29138", "g17929", "g26348", "g20247", "GGT6", "g17926", "g7536", "g7536", "NOG", "g28014", "g29504", "APCDD1", "CRIP2", "NKX1.1", "ZNRF3", "g5149", "g650", "g17063", "TRAF3.3", "FOSL2.2", "SIK2", "DNAJB5", "TRAFD1", "HSF1", "g26821", "CHRDL1", "g5331") 
#for now not removing the unnannotated genes
#two trafs

gderm1_feats_mark <- subset(gderm1_mark, Preferred_name %in% gderm1_feats)
# add a column of NAs
gderm1_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
gderm1_feats_mark$diffexpressed[gderm1_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
gderm1_feats_mark$diffexpressed[gderm1_feats_mark$avg_log2FC < 0] <- "apo"
gderm1_feats_bar <- ggplot(data=gderm1_feats_mark, aes(y=Preferred_name, x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
####

gderm1_feats2go <- gderm1_feats_mark %>%
  left_join(gene2go)
write.table(gderm1_feats2go, file="./files/gderm1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

gderm1_feats2go2cog <- gderm1_feats2go %>%
  left_join(gene2cog)
head(gderm1_feats2go2cog)
gderm1_feats2go2cog$COG_category[gderm1_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
gderm1_feats2go2cog2pfam <- gderm1_feats2go2cog %>%
  left_join(gene2pfam)
gderm1_feats2go2cog2pfam <- subset(gderm1_feats2go2cog2pfam, gderm1_feats2go2cog2pfam$PFAMs!="-")

gderm1_feats2go2cog2pfam <- gderm1_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

gderm1_feats_bar_sort <- ggplot(data=gderm1_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### gderm1_feats_barsort_2025 as 11X4
###use this to identify GO terms: https://www.ebi.ac.uk/QuickGO/annotations
```

2. Gastrodermis 2 res of 0.25
```{r}
gderm2 <- subset(oculina_named_new, idents = "Gastrodermis 2") 

gderm2 <- FindVariableFeatures(gderm2)
gderm2 <- ScaleData(gderm2, features = features_to_scale)
gderm2 <- RunPCA(gderm2)
gderm2 <- FindNeighbors(gderm2, dims = 1:30, reduction = "pca")
gderm2 <- FindClusters(gderm2, resolution = 0.25)
#3 cluster
gderm2 <- RunUMAP(gderm2, reduction="pca", dims=1:30, return.model=TRUE)
#gderm2 <- RunUMAP(gderm2, dims = 1:30, reduction = "integrated.cca")

gderm2@meta.data$seurat_clusters <- as.numeric(as.character(gderm2@meta.data$seurat_clusters))+1
Idents(gderm2)<- "seurat_clusters"
gderm2$seurat_clusters
gderm2@active.ident <- factor(x = gderm2@active.ident, levels = 1:3)
gderm2@active.ident

levels(gderm2) <- c("1", "2", "3")

plot7 <- DimPlot(gderm2, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5)
#3 clusters

plot8 <- DimPlot(gderm2, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot8b <- DimPlot(gderm2, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gderm2_plot7plot8_2025 9x4
#umap_gderm2_plot8b_2025 as 8x4
```

```{r}
gderm2_topfeat <- TopFeatures(object = gderm2[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gderm2[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gd2 <- FetchData(gderm2, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gd2 <- FetchData(gderm2, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gd2, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gd2, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as gderm2_pcplots_umap_2025

stdev_gd2 <- Stdev(gderm2, reduction = "pca")

pca_gd2 <- pc_data_gd2
pca_gd2$sym_state <- gderm2@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gd2)) #<2e-16 ***
summary(aov(PC_2 ~ sym_state, data = pca_gd2)) #<2e-16 ***
summary(aov(PC_3 ~ sym_state, data = pca_gd2)) #5.71e-06 ***
summary(aov(PC_4 ~ sym_state, data = pca_gd2)) #6.97e-10 ***
summary(aov(PC_5 ~ sym_state, data = pca_gd2)) #<2e-16 ***

summary(aov(PC_6 ~ sym_state, data = pca_gd2)) #0.879
summary(aov(PC_7 ~ sym_state, data = pca_gd2)) #0.669
summary(aov(PC_8 ~ sym_state, data = pca_gd2)) #0.011 *
summary(aov(PC_9 ~ sym_state, data = pca_gd2)) #1.83e-07 ***
summary(aov(PC_10 ~ sym_state, data = pca_gd2)) #4.41e-07 ***

print(gderm2[["pca"]], dims = 1:5, nfeatures = 10)

DefaultAssay(gderm2) <- "RNA"
gderm2[["RNA"]]$counts<-as.matrix(gderm2[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gderm2_mark <- FindMarkers(gderm2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gderm2_mark)
#neg lfc means down in sym; pos lfc means up in sym

gderm2_mark <- rownames_to_column(gderm2_mark, var="Preferred_name")

#look at subcluster marker genes
gderm2_markers_subclust <- FindAllMarkers(gderm2, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gderm2_markers_subclust
gderm2_markers_subclust <- rownames_to_column(gderm2_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of gderm2
gderm2_1 <- subset(gderm2, idents="1")
DefaultAssay(gderm2_1) <- "RNA"

gderm2_1_mark <- FindMarkers(gderm2_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm2_1_mark <- rownames_to_column(gderm2_1_mark, var="Preferred_name")
head(gderm2_1_mark)

#get apo vs sym differences for subcluster 2 of gderm2
gderm2_2 <- subset(gderm2, idents="2")
DefaultAssay(gderm2_2) <- "RNA"

gderm2_2_mark <- FindMarkers(gderm2_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm2_2_mark <- rownames_to_column(gderm2_2_mark, var="Preferred_name")
head(gderm2_2_mark)

#for 3
gderm2_3 <- subset(gderm2, idents="3")
DefaultAssay(gderm2_3) <- "RNA"
gderm2_3_mark <- FindMarkers(gderm2_3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
#no apo cells
```

```{r}
#get genes from signficant PCs (1-5, 8, 9, 10)
print(gderm2[["pca"]], dims = 1:10, nfeatures = 10)

gderm2_feats <- c("g16218", "ANKS3", "g2185", "ANO8", "ASL", "HNF4A", "g26521", "GBA", "g24461", "ICT1", "RPL18A", "RPL15", "rps29", "RPLP1", "rpl26l1", "RPS3", "RpL10", "AHCY", "RPLP0", "RpL37A", "ANKS3", "ASL", "g24461", "HNF4A", "ANO8", "g2185", "g16218", "GBA", "ICT1", "SKP2.1", "ZC2HC1C", "g10482", "FOXJ1", "AK9", "CFAP45", "g24668", "g4255", "ccdc147", "FAM184A", "TEKT2", "adt.1.1", "g2313", "g14286", "g570", "g19746", "g16526", "MFI2.3", "g5748", "g20242", "pplD2", "ZC2HC1C", "g10482", "TEKT2", "CFAP45", "FOXJ1", "g24668", "AK9", "CCDC39", "g4255", "ccdc147", "DNAJB13", "ITGAM", "KIF26B", "COL6A3", "DDX4.1", "g1806", "g8101", "CFAP45", "ARSB.6", "g17929", "g4358", "g25521", "CNKSR1", "c5", "MFI2.3", "g6923", "PKHD1L1", "g20242", "g19746", "g16526", "g1806", "LAMP1.1", "MOXD1.12", "g13114", "CALR", "HSP90B1", "g25521", "HSPA5", "FOSL2.2", "g8101", "FST", "g17927", "g28257", "g9360", "g23653", "VKORC1.1", "pplD2", "g26166", "g22781", "TAGLN3", "g570", "Mmp1.16", "g22939", "g12662", "TRPA1.12", "g24943", "mup.4.1", "g7312", "g6517", "MYD88.2", "g20910", "HSF1", "TRAF6.1", "MYD88.2", "g18947", "g21831", "g24727", "Syt15", "SHB", "SPATA17", "DNMT1", "DNAJC21", "g29236", "NFXL1", "SMC4", "PSIP1", "EVA1A.2", "g26412", "BRSK2", "RDH5.3", "HMCN1.46", "GBA", "g7532", "g25521", "ABCC4.44", "COL6A3", "g21756", "g27326", "g26166", "ATAD5.1", "NAPEPLD", "g29918", "MUC5B", "DGCR2", "g2313", "g3472", "g6161", "g1271", "g14813", "DNALI1") 
#for now not removing the unnannotated genes
#MYD88!!

gderm2_feats_mark <- subset(gderm2_mark, Preferred_name %in% gderm2_feats)
# add a column of NAs
gderm2_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
gderm2_feats_mark$diffexpressed[gderm2_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
gderm2_feats_mark$diffexpressed[gderm2_feats_mark$avg_log2FC < 0] <- "apo"
gderm2_feats_bar <- ggplot(data=gderm2_feats_mark, aes(y=Preferred_name, x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()

gderm2_feats2go <- gderm2_feats_mark %>%
  left_join(gene2go)
write.table(gderm2_feats2go, file="./files/gderm2_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

gderm2_feats2go2cog <- gderm2_feats2go %>%
  left_join(gene2cog)
head(gderm2_feats2go2cog)
gderm2_feats2go2cog$COG_category[gderm2_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
gderm2_feats2go2cog2pfam <- gderm2_feats2go2cog %>%
  left_join(gene2pfam)
gderm2_feats2go2cog2pfam <- subset(gderm2_feats2go2cog2pfam, gderm2_feats2go2cog2pfam$PFAMs!="-")

gderm2_feats2go2cog2pfam <- gderm2_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

gderm2_feats_bar_sort <- ggplot(data=gderm2_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### gderm2_feats_barsort_2025 as 12x4
```

3. Gastrodermis 3 RES OF 0.3
```{r}
gderm3 <- subset(oculina_named_new, idents = "Gastrodermis 3") 

gderm3 <- FindVariableFeatures(gderm3)
gderm3 <- ScaleData(gderm3, features = features_to_scale)
gderm3 <- RunPCA(gderm3)
gderm3 <- FindNeighbors(gderm3, dims = 1:30, reduction = "pca")
gderm3 <- FindClusters(gderm3, resolution = 0.3)
#2 cluster
gderm3 <- RunUMAP(gderm3, reduction="pca", dims=1:30, return.model=TRUE)

gderm3@meta.data$seurat_clusters <- as.numeric(as.character(gderm3@meta.data$seurat_clusters))+1
Idents(gderm3)<- "seurat_clusters"
gderm3$seurat_clusters
gderm3@active.ident <- factor(x = gderm3@active.ident, levels = 1:2)
gderm3@active.ident

levels(gderm3) <- c("1", "2")

plot9 <- DimPlot(gderm3, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 clusters

plot10 <- DimPlot(gderm3, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot10b <- DimPlot(gderm3, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gderm3_plot9plot10_2025 9x4
#umap_gderm3_plot10b_2025 8x4 landscape
```

```{r}
gderm3_topfeat <- TopFeatures(object = gderm3[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gderm3[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gd3 <- FetchData(gderm3, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gd3 <- FetchData(gderm3, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gd3, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gd3, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as gderm3_pcplots_umap_2025

stdev_gd3 <- Stdev(gderm3, reduction = "pca")

pca_gd3 <- pc_data_gd3
pca_gd3$sym_state <- gderm3@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gd3)) #0.000183 ***
summary(aov(PC_2 ~ sym_state, data = pca_gd3)) #8.86e-05 ***
summary(aov(PC_3 ~ sym_state, data = pca_gd3)) #<2e-16 ***
summary(aov(PC_4 ~ sym_state, data = pca_gd3)) #7.59e-08 ***
summary(aov(PC_5 ~ sym_state, data = pca_gd3)) #6.66e-07 ***

summary(aov(PC_6 ~ sym_state, data = pca_gd3)) #0.118
summary(aov(PC_7 ~ sym_state, data = pca_gd3)) #0.0723 .
summary(aov(PC_8 ~ sym_state, data = pca_gd3)) #0.022 *
summary(aov(PC_9 ~ sym_state, data = pca_gd3)) #9.5e-10 ***
summary(aov(PC_10 ~ sym_state, data = pca_gd3)) #0.114


DefaultAssay(gderm3) <- "RNA"
gderm3[["RNA"]]$counts<-as.matrix(gderm3[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gderm3_mark <- FindMarkers(gderm3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gderm3_mark)
#neg lfc means down in sym; pos lfc means up in sym

gderm3_mark <- rownames_to_column(gderm3_mark, var="Preferred_name")

#look at subcluster marker genes
gderm3_markers_subclust <- FindAllMarkers(gderm3, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gderm3_markers_subclust
gderm3_markers_subclust <- rownames_to_column(gderm3_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of gderm3
gderm3_1 <- subset(gderm3, idents="1")
DefaultAssay(gderm3_1) <- "RNA"

gderm3_1_mark <- FindMarkers(gderm3_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm3_1_mark <- rownames_to_column(gderm3_1_mark, var="Preferred_name")
head(gderm3_1_mark)

#get apo vs sym differences for subcluster 2 of gderm3
gderm3_2 <- subset(gderm3, idents="2")
DefaultAssay(gderm3_2) <- "RNA"

gderm3_2_mark <- FindMarkers(gderm3_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm3_2_mark <- rownames_to_column(gderm3_2_mark, var="Preferred_name")
head(gderm3_2_mark)
```

```{r}
#get genes from signficant PCs (1-5, 8, 9)
print(gderm3[["pca"]], dims = 1:10, nfeatures = 10)

gderm3_feats <- c("RPL38", "RPS28", "RPS19", "RPS21", "RPL27A", "RPL31", "RpL35", "rps29", "rpl30", "RPL24", "DLC1", "DMRTA2.1", "CHL1.1", "g29016", "g9858", "g17202", "ik2", "g537", "g11876", "MRC1.3", "FOSL2.2", "BAG4", "DNAJB5", "g24422", "g18453", "LHFPL2", "g4909", "HEBP2", "HMCN1.14", "ITGAM", "g6517", "DAG1", "g26348", "g6925", "RYK", "AGRN.1", "MFI2.3", "otx1", "c5", "LTA", "g20264", "COL2A1", "TSTD1", "DLC1", "ADAMTS7", "g8814", "g16924", "g24770", "g20257", "DMRTA2.1", "g24461", "g24460", "g11768", "BSG.1", "g20747", "ASL", "ANKS3", "GBA", "ICT1", "SLC37A1", "HMCN1.14", "HMCN1.13", "g23832", "WNT7A", "g24760", "g3990", "ADAM29.3", "g23828", "PRSS8", "g14814", "TEKT2", "g24668", "CFAP45", "CCDC11", "ENKUR", "LCA5", "WDR65", "EFHC2", "ccdc147", "AK9", "HMCN1.14", "g23832", "UNC5C.1", "g23828", "g16199", "g25603", "g3990", "HMCN1.13", "WNT7A", "g27910", "g20257", "APOB", "g20256", "SVEP1.1", "g8256", "g20264", "COL2A1", "g9828", "g8814", "ADAMTS17", "ITGA9.4", "g13473", "TAGLN3", "TBX10", "g22237", "g8181", "ITGAM", "MCC", "APCDD1", "ADAMTS17", "HMCN1.14", "WNT7A", "g23832", "HMCN1.13", "TEKT2", "g23828", "g24668", "ADAM29.3", "WDR65", "g24760", "g3279", "COL6A3", "TBX10", "FRK.2", "CALB1.1", "COL2A1", "g20264", "g18489", "g24759", "CSRP1.1", "g3541", "CHL1.1", "ALOXE3.2", "ITGA9.5", "TIMP1.1", "Mrpl52", "g16080", "ITGA9.4", "DLC1", "g22917") 
#for now not removing the unnannotated genes
#MYD88!!

gderm3_feats_mark <- subset(gderm3_mark, Preferred_name %in% gderm3_feats)
# add a column of NAs
gderm3_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
gderm3_feats_mark$diffexpressed[gderm3_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
gderm3_feats_mark$diffexpressed[gderm3_feats_mark$avg_log2FC < 0] <- "apo"
gderm3_feats_bar <- ggplot(data=gderm3_feats_mark, aes(y=Preferred_name, x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()

gderm3_feats2go <- gderm3_feats_mark %>%
  left_join(gene2go)
write.table(gderm3_feats2go, file="./files/gderm3_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

gderm3_feats2go2cog <- gderm3_feats2go %>%
  left_join(gene2cog)
head(gderm3_feats2go2cog)
gderm3_feats2go2cog$COG_category[gderm3_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
gderm3_feats2go2cog2pfam <- gderm3_feats2go2cog %>%
  left_join(gene2pfam)
gderm3_feats2go2cog2pfam <- subset(gderm3_feats2go2cog2pfam, gderm3_feats2go2cog2pfam$PFAMs!="-")

gderm3_feats2go2cog2pfam <- gderm3_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

gderm3_feats_bar_sort <- ggplot(data=gderm3_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### gderm3_feats_barsort_2025 as 12x4
```

4. Gastrodermis 4 THIS HAS RES OF 0.3 AS WELL
```{r}
gderm4 <- subset(oculina_named_new, idents = "Gastrodermis 4") 

gderm4 <- FindVariableFeatures(gderm4)
gderm4 <- ScaleData(gderm4, features = features_to_scale)
gderm4 <- RunPCA(gderm4)
gderm4 <- FindNeighbors(gderm4, dims = 1:30, reduction = "pca")
gderm4 <- FindClusters(gderm4, resolution = 0.3)
#2 cluster
gderm4 <- RunUMAP(gderm4, reduction="pca", dims=1:30, return.model=TRUE)

gderm4@meta.data$seurat_clusters <- as.numeric(as.character(gderm4@meta.data$seurat_clusters))+1
Idents(gderm4)<- "seurat_clusters"
gderm4$seurat_clusters
gderm4@active.ident <- factor(x = gderm4@active.ident, levels = 1:2)
gderm4@active.ident

levels(gderm4) <- c("1", "2")

plot11 <- DimPlot(gderm4, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 clusters

plot12 <- DimPlot(gderm4, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot12b <- DimPlot(gderm4, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gderm4_plot11plot12_2025 9x4
#umap_gderm4_plot12b_2025 8x4 landscape
```

```{r}
gderm4_topfeat <- TopFeatures(object = gderm4[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gderm4[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gd4 <- FetchData(gderm4, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gd4 <- FetchData(gderm4, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gd4, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gd4, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as gderm4_pcplots_umap_2025

stdev_gd4 <- Stdev(gderm4, reduction = "pca")

pca_gd4 <- pc_data_gd4
pca_gd4$sym_state <- gderm4@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gd4)) #0.83
summary(aov(PC_2 ~ sym_state, data = pca_gd4)) #8.52e-10 ***
summary(aov(PC_3 ~ sym_state, data = pca_gd4)) #<2e-16 ***
summary(aov(PC_4 ~ sym_state, data = pca_gd4)) #1.35e-09 ***
summary(aov(PC_5 ~ sym_state, data = pca_gd4)) #0.103

summary(aov(PC_6 ~ sym_state, data = pca_gd4)) #0.00692 **
summary(aov(PC_7 ~ sym_state, data = pca_gd4)) #0.407
summary(aov(PC_8 ~ sym_state, data = pca_gd4)) #0.692
summary(aov(PC_9 ~ sym_state, data = pca_gd4)) #0.0302 *
summary(aov(PC_10 ~ sym_state, data = pca_gd4)) #0.00168 **


DefaultAssay(gderm4) <- "RNA"
gderm4[["RNA"]]$counts<-as.matrix(gderm4[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gderm4_mark <- FindMarkers(gderm4, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gderm4_mark)
#neg lfc means down in sym; pos lfc means up in sym

gderm4_mark <- rownames_to_column(gderm4_mark, var="Preferred_name")

#look at subcluster marker genes
gderm4_markers_subclust <- FindAllMarkers(gderm4, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gderm4_markers_subclust
gderm4_markers_subclust <- rownames_to_column(gderm4_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of gderm4
gderm4_1 <- subset(gderm4, idents="1")
DefaultAssay(gderm4_1) <- "RNA"

gderm4_1_mark <- FindMarkers(gderm4_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm4_1_mark <- rownames_to_column(gderm4_1_mark, var="Preferred_name")
head(gderm4_1_mark)

#get apo vs sym differences for subcluster 2 of gderm4
gderm4_2 <- subset(gderm4, idents="2")
DefaultAssay(gderm4_2) <- "RNA"

gderm4_2_mark <- FindMarkers(gderm4_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm4_2_mark <- rownames_to_column(gderm4_2_mark, var="Preferred_name")
head(gderm4_2_mark)
```

```{r}
#get genes from signficant PCs (2,3,4,6,9,10)
print(gderm4[["pca"]], dims = 1:10, nfeatures = 10)

gderm4_feats <- c("g25687", "GIF", "g27381", "g816", "g21288", "MAPK7", "ELK1", "CD59.2", "g7576", "GLIPR2.4", "g7221", "g13322", "g12149", "g25112", "EDIL3.9", "CENPC", "PXDN", "PSAT1", "PHGDH", "g12927", "GIF", "c5", "g12420", "g9400", "g5461", "hisC", "LRP6", "g26814", "IQCA1", "Unc.89", "g1806", "IFI30.1", "g6767", "HEBP2", "PRCP", "CYP3A4", "g14259", "HSPA5", "g26821", "PHGDH", "ANKS3", "LRP6", "g16526", "g9400", "OLFML2B.1", "g10733", "g18298", "g20242", "g5320", "APOB", "ZC2HC1C", "g24668", "g12149", "TEKT2", "FOXJ1", "ODF3L2", "g312", "g12927", "WDR65", "g2145", "RIT1", "VWA5A.2", "g6041", "g25991", "HSPA12A.3", "RAB8B.1", "g2185", "g26521", "g19095", "g25685", "g25599", "g20257", "g20264", "g14956", "MFI2.3", "g14286", "g17566", "g20242", "FLI1.1", "g22939", "TSPAN5", "OLFML2B.1", "g7576", "g10742", "engB", "g22832", "g4224", "PLEKHG7", "g27451", "USB1", "g9962", "g3646", "g6041", "g9899", "g30342", "SMC4", "g8968", "g15636", "LIG1", "g15488", "RPA2", "DCUN1D4", "g8968", "g6825", "FIGF", "FAM65A", "g18695", "g8475", "TEX9", "RIT1", "g18620", "g6517", "GRIN2B", "g5331", "WNT4", "g23979", "GRN.2", "g25426", "IL16", "g28309") 
#for now not removing the unnannotated genes

gderm4_feats_mark <- subset(gderm4_mark, Preferred_name %in% gderm4_feats)
# add a column of NAs
gderm4_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
gderm4_feats_mark$diffexpressed[gderm4_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
gderm4_feats_mark$diffexpressed[gderm4_feats_mark$avg_log2FC < 0] <- "apo"

gderm4_feats2go <- gderm4_feats_mark %>%
  left_join(gene2go)
write.table(gderm4_feats2go, file="./files/gderm4_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

gderm4_feats2go2cog <- gderm4_feats2go %>%
  left_join(gene2cog)
head(gderm4_feats2go2cog)
gderm4_feats2go2cog$COG_category[gderm4_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
gderm4_feats2go2cog2pfam <- gderm4_feats2go2cog %>%
  left_join(gene2pfam)
gderm4_feats2go2cog2pfam <- subset(gderm4_feats2go2cog2pfam, gderm4_feats2go2cog2pfam$PFAMs!="-")

gderm4_feats2go2cog2pfam <- gderm4_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

gderm4_feats_bar_sort <- ggplot(data=gderm4_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### gderm4_feats_barsort_2025 as 12x4
```

5. Gastrodermis 5 RES OF 0.3 
```{r}
gderm5 <- subset(oculina_named_new, idents = "Gastrodermis 5") 

gderm5 <- FindVariableFeatures(gderm5)
gderm5 <- ScaleData(gderm5, features = features_to_scale)
gderm5 <- RunPCA(gderm5)
gderm5 <- FindNeighbors(gderm5, dims = 1:30, reduction = "pca")
gderm5 <- FindClusters(gderm5, resolution = 0.3)
#2 cluster
gderm5 <- RunUMAP(gderm5, reduction="pca", dims=1:30, return.model=TRUE)

gderm5@meta.data$seurat_clusters <- as.numeric(as.character(gderm5@meta.data$seurat_clusters))+1
Idents(gderm5)<- "seurat_clusters"
gderm5$seurat_clusters
gderm5@active.ident <- factor(x = gderm5@active.ident, levels = 1:2)
gderm5@active.ident

levels(gderm5) <- c("1", "2")

plot13 <- DimPlot(gderm5, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 clusters

plot14 <- DimPlot(gderm5, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot14b <- DimPlot(gderm5, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gderm5_plot13plot14_2025 9x4
#umap_gderm5_plot14b_2025 8x4 landscape
```

```{r}
gderm5_topfeat <- TopFeatures(object = gderm5[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gderm5[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gd5 <- FetchData(gderm5, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gd5 <- FetchData(gderm5, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gd5, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gd5, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as gderm5_pcplots_umap_2025

stdev_gd5 <- Stdev(gderm5, reduction = "pca")

pca_gd5 <- pc_data_gd5
pca_gd5$sym_state <- gderm5@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gd5)) #0.000126 ***
summary(aov(PC_2 ~ sym_state, data = pca_gd5)) #0.00225 **
summary(aov(PC_3 ~ sym_state, data = pca_gd5)) #0.0736 .
summary(aov(PC_4 ~ sym_state, data = pca_gd5)) #0.00644 **
summary(aov(PC_5 ~ sym_state, data = pca_gd5)) #0.55

summary(aov(PC_6 ~ sym_state, data = pca_gd5)) #0.00328 **
summary(aov(PC_7 ~ sym_state, data = pca_gd5)) #0.121
summary(aov(PC_8 ~ sym_state, data = pca_gd5)) #0.00702 **
summary(aov(PC_9 ~ sym_state, data = pca_gd5)) #0.183
summary(aov(PC_10 ~ sym_state, data = pca_gd5)) #0.359


DefaultAssay(gderm5) <- "RNA"
gderm5[["RNA"]]$counts<-as.matrix(gderm5[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gderm5_mark <- FindMarkers(gderm5, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gderm5_mark)
#neg lfc means down in sym; pos lfc means up in sym

gderm5_mark <- rownames_to_column(gderm5_mark, var="Preferred_name")

#look at subcluster marker genes
gderm5_markers_subclust <- FindAllMarkers(gderm5, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gderm5_markers_subclust
gderm5_markers_subclust <- rownames_to_column(gderm5_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of gderm5
gderm5_1 <- subset(gderm5, idents="1")
DefaultAssay(gderm5_1) <- "RNA"

gderm5_1_mark <- FindMarkers(gderm5_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm5_1_mark <- rownames_to_column(gderm5_1_mark, var="Preferred_name")
head(gderm5_1_mark)

#get apo vs sym differences for subcluster 2 of gderm5
gderm5_2 <- subset(gderm5, idents="2")
DefaultAssay(gderm5_2) <- "RNA"

gderm5_2_mark <- FindMarkers(gderm5_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gderm5_2_mark <- rownames_to_column(gderm5_2_mark, var="Preferred_name")
head(gderm5_2_mark)
```

```{r}
#get genes from signficant PCs (1,2,4,6,8)
print(gderm5[["pca"]], dims = 1:10, nfeatures = 10)

gderm5_feats <- c("g10389", "g20955", "g19115", "HIST3H3.6", "g4247", "COF1.6", "HIST1H1A.2", "CUZD1.5", "HIST1H1A.4", "g10149", "g20264", "APOB", "g20254", "g25521", "COL2A1", "COL4A1", "AGRN.1", "EPDR1.1", "g1804", "CTSB", "XRCC1.1", "PRDM9", "g12921", "FANCB", "CDK14", "g29107", "g20242", "g16526", "CA9.2", "g28155", "AHCY", "RPL22", "RPL32", "TUBA3D.8", "ADK", "OST4", "g30147", "LAMP1.1", "ANLN", "NPC2", "g23943", "ARMC1", "MICAL3", "CDK1", "kinesin.A", "g11818", "ANKRD7", "FAM161A", "HMCN1.26", "MEIG1", "H1F0.1", "HIST3H3.1", "g18113", "g27354", "g9927", "g27438", "g27514", "g27357", "HIST1H3D.1", "g16536", "g16526", "g20245", "PAX1", "TTLL1", "g23646", "ADAMTS12", "FREM3", "TBX20.2", "IRF4.1", "RAB30", "g16088", "g252", "g5282", "LHFPL2", "g4567", "g16799", "g14381", "g1315", "g16080", "MEP1B.1", "PXDN.14", "g6148", "SEMA5A.2", "CFP.5", "g28594", "g30214", "g13592", "SPRYD3", "g18360", "SDK2", "RAB30", "Ly6i", "DGKE.1", "MAFB", "NEUROD4", "g16683", "g29508", "NANOS2", "MYC.1", "g1238") 
#for now not removing the unnannotated genes

gderm5_feats_mark <- subset(gderm5_mark, Preferred_name %in% gderm5_feats)
# add a column of NAs
gderm5_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
gderm5_feats_mark$diffexpressed[gderm5_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
gderm5_feats_mark$diffexpressed[gderm5_feats_mark$avg_log2FC < 0] <- "apo"

gderm5_feats2go <- gderm5_feats_mark %>%
  left_join(gene2go)
write.table(gderm5_feats2go, file="./files/gderm5_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

gderm5_feats2go2cog <- gderm5_feats2go %>%
  left_join(gene2cog)
head(gderm5_feats2go2cog)
gderm5_feats2go2cog$COG_category[gderm5_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
gderm5_feats2go2cog2pfam <- gderm5_feats2go2cog %>%
  left_join(gene2pfam)
gderm5_feats2go2cog2pfam <- subset(gderm5_feats2go2cog2pfam, gderm5_feats2go2cog2pfam$PFAMs!="-")

gderm5_feats2go2cog2pfam <- gderm5_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

gderm5_feats_bar_sort <- ggplot(data=gderm5_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### gderm5_feats_barsort_2025 as 11x4
```

6. Algal-Hosting
```{r}
agh <- subset(oculina_named_new, idents = "Algal-Hosting") 

agh <- FindVariableFeatures(agh)
agh <- ScaleData(agh, features = features_to_scale)
agh <- RunPCA(agh)
agh <- FindNeighbors(agh, dims = 1:30, reduction = "pca")
agh <- FindClusters(agh, resolution = 0.3)
#2 cluster
agh <- RunUMAP(agh, reduction="pca", dims=1:30, return.model=TRUE)

agh@meta.data$seurat_clusters <- as.numeric(as.character(agh@meta.data$seurat_clusters))+1
Idents(agh)<- "seurat_clusters"
agh$seurat_clusters
agh@active.ident <- factor(x = agh@active.ident, levels = 1:2)
agh@active.ident

levels(agh) <- c("1", "2")

plot15 <- DimPlot(agh, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 clusters

plot16 <- DimPlot(agh, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot16b <- DimPlot(agh, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_algalhost_plot15plot16_2025 9x4
#umap_algalhost_plot16b_2025 8x4 landscape
```

```{r}
agh_topfeat <- TopFeatures(object = agh[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
agh[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_agh <- FetchData(agh, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_agh <- FetchData(agh, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_agh, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_agh, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)
#saved as agh_pcplots_umap_2025

stdev_agh <- Stdev(agh, reduction = "pca")

pca_agh <- pc_data_agh
pca_agh$sym_state <- agh@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_agh)) #6.87e-06 ***
summary(aov(PC_2 ~ sym_state, data = pca_agh)) #0.0143 *
summary(aov(PC_3 ~ sym_state, data = pca_agh)) #0.209
summary(aov(PC_4 ~ sym_state, data = pca_agh)) #0.0782 .
summary(aov(PC_5 ~ sym_state, data = pca_agh)) #0.428

summary(aov(PC_6 ~ sym_state, data = pca_agh)) #0.0038 **
summary(aov(PC_7 ~ sym_state, data = pca_agh)) #0.802
summary(aov(PC_8 ~ sym_state, data = pca_agh)) #0.946
summary(aov(PC_9 ~ sym_state, data = pca_agh)) #0.894
summary(aov(PC_10 ~ sym_state, data = pca_agh)) #0.144


DefaultAssay(agh) <- "RNA"
agh[["RNA"]]$counts<-as.matrix(agh[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
agh_mark <- FindMarkers(agh, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(agh_mark)
#neg lfc means down in sym; pos lfc means up in sym

agh_mark <- rownames_to_column(agh_mark, var="Preferred_name")

#look at subcluster marker genes
agh_markers_subclust <- FindAllMarkers(agh, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

agh_markers_subclust
agh_markers_subclust <- rownames_to_column(agh_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of gderm5
agh_1 <- subset(agh, idents="1")
DefaultAssay(agh_1) <- "RNA"

agh_1_mark <- FindMarkers(agh_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
#only 1 apo

#get apo vs sym differences for subcluster 2 of gderm5
agh_2 <- subset(agh, idents="2")
DefaultAssay(agh_2) <- "RNA"

agh_2_mark <- FindMarkers(agh_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
agh_2_mark <- rownames_to_column(agh_2_mark, var="Preferred_name")
head(agh_2_mark)
```

```{r}
#get genes from signficant PCs (1,2,6)
print(agh[["pca"]], dims = 1:10, nfeatures = 10)

agh_feats <- c("CFAP44", "TTC14", "PPP1R27.1", "SPTBN5.1", "g17428", "GAPVD1", "g4338", "g19161", "WDR74", "g22413", "g28613", "hbn", "COX6C.1", "PDK2", "MAT1A", "COL7A1", "HEXIM1", "g30141", "add.1", "PDGFRB.3", "RpL37A", "RPL24", "RPL9", "RPL13", "rps18", "RPL28", "RPS19", "RPL38", "RpS20", "RPL7", "USP9X", "HEATR5B", "MRPL38", "g9584", "g5630", "C12orf55", "g19135", "NUB1", "SPTBN5.1", "HIAT1", "AP1AR", "Pk17E", "UBA5", "g12853", "g20764", "g25863", "EFCAB12", "AGRN.1", "CELSR2.2", "POLR3K", "CFAP44", "NOP58", "PPP1R27.1", "SEN34", "PDIA4.2", "ARHGEF3", "SMC4", "g4802", "ATIC", "BAZ1A") 
#for now not removing the unnannotated genes

agh_feats_mark <- subset(agh_mark, Preferred_name %in% agh_feats)
# add a column of NAs
agh_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
agh_feats_mark$diffexpressed[agh_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
agh_feats_mark$diffexpressed[agh_feats_mark$avg_log2FC < 0] <- "apo"

agh_feats2go <- agh_feats_mark %>%
  left_join(gene2go)
write.table(agh_feats2go, file="./files/agh_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

agh_feats2go2cog <- agh_feats2go %>%
  left_join(gene2cog)
head(agh_feats2go2cog)
agh_feats2go2cog$COG_category[agh_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
agh_feats2go2cog2pfam <- agh_feats2go2cog %>%
  left_join(gene2pfam)
agh_feats2go2cog2pfam <- subset(agh_feats2go2cog2pfam, agh_feats2go2cog2pfam$PFAMs!="-")

agh_feats2go2cog2pfam <- agh_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

agh_feats_bar_sort <- ggplot(data=agh_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### agh_feats_barsort_2025 as 8x4
```

7. Immune Cell RES OF 0.4
```{r}
IC <- subset(oculina_named_new, idents = "Immune Cell") 

IC <- FindVariableFeatures(IC)
IC <- ScaleData(IC, features = features_to_scale)
IC <- RunPCA(IC)
IC <- FindNeighbors(IC, dims = 1:30, reduction = "pca")
IC <- FindClusters(IC, resolution = 0.4)
#3 cluster
IC <- RunUMAP(IC, reduction="pca", dims=1:30, return.model=TRUE)

IC@meta.data$seurat_clusters <- as.numeric(as.character(IC@meta.data$seurat_clusters))+1
Idents(IC)<- "seurat_clusters"
IC$seurat_clusters
IC@active.ident <- factor(x = IC@active.ident, levels = 1:3)
IC@active.ident

levels(IC) <- c("1", "2", "3")

plot17 <- DimPlot(IC, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5)
#3 clusters

plot18 <- DimPlot(IC, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot18b <- DimPlot(IC, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99", "#3333FF"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_immune_plot17plot18_2025 8x7 portrait
#umap_immune_plot18b_2025 8x4 landscape
```

```{r}
IC_topfeat <- TopFeatures(object = IC[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
IC[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_IC <- FetchData(IC, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_IC <- FetchData(IC, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_IC, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_IC, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_IC <- Stdev(IC, reduction = "pca")

pca_IC <- pc_data_IC
pca_IC$sym_state <- IC@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_IC)) #0.101
summary(aov(PC_2 ~ sym_state, data = pca_IC)) #0.265
summary(aov(PC_3 ~ sym_state, data = pca_IC)) #0.985
summary(aov(PC_4 ~ sym_state, data = pca_IC)) #0.495
summary(aov(PC_5 ~ sym_state, data = pca_IC)) #0.162

summary(aov(PC_6 ~ sym_state, data = pca_IC)) #0.143
summary(aov(PC_7 ~ sym_state, data = pca_IC)) #0.173
summary(aov(PC_8 ~ sym_state, data = pca_IC)) #0.886
summary(aov(PC_9 ~ sym_state, data = pca_IC)) #0.732
summary(aov(PC_10 ~ sym_state, data = pca_IC)) #0.163


DefaultAssay(IC) <- "RNA"
IC[["RNA"]]$counts<-as.matrix(IC[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
IC_mark <- FindMarkers(IC, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(IC_mark)
#neg lfc means down in sym; pos lfc means up in sym

IC_mark <- rownames_to_column(IC_mark, var="Preferred_name")

#look at subcluster marker genes
IC_markers_subclust <- FindAllMarkers(IC, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

IC_markers_subclust
IC_markers_subclust <- rownames_to_column(IC_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of IC
IC_1 <- subset(IC, idents="1")
DefaultAssay(IC_1) <- "RNA"

IC_1_mark <- FindMarkers(IC_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
IC_1_mark <- rownames_to_column(IC_1_mark, var="Preferred_name")
head(IC_1_mark)

#get apo vs sym differences for subcluster 2 of IC
IC_2 <- subset(IC, idents="2")
DefaultAssay(IC_2) <- "RNA"

IC_2_mark <- FindMarkers(IC_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
IC_2_mark <- rownames_to_column(IC_2_mark, var="Preferred_name")
head(IC_2_mark)

#get apo vs sym differences for subcluster 3 of IC
IC_3 <- subset(IC, idents="3")
DefaultAssay(IC_3) <- "RNA"

IC_3_mark <- FindMarkers(IC_3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
IC_3_mark <- rownames_to_column(IC_3_mark, var="Preferred_name")
head(IC_3_mark)
```

```{r}
#NO significant PCs
```

Immune Cell Cluster Dot Plot and Heatmap
```{r}
oculina_sym <- subset(oculina_named_new, subset = symstate == "sym")
oculina_apo <- subset(oculina_named_new, subset = symstate == "apo")
#these are marker genes for the spist immune cell that are also in ours
#irf1, irf1.1, and irf1.2 are TFs
#g30303 is hsp70
#K123 is endopeptidase
#identified these genes 04/23/25 MPEG1, Hsp68, g6378 (interferon guy), NLRP3.7, IRF4 (good), IRF4.1
immune_features <- c("IRF1", "IRF1.1", "IRF1.2", "g30303", "CTSZ", "LBP", "K123", "BAG3", "BAG4.1", "BAG4", "CFLAR", "CFLAR.1", "CFLAR.3", "CFLAR.4", "MVP", "MPEG1", "Hsp68", "g6378", "NLRP3.7", "IRF4", "IRF4.1")

immune_features_small <- c("IRF1", "IRF1.2", "IRF4", "IRF4.1", "g30303", "CTSZ", "LBP", "K123", "BAG4", "CFLAR.4", "MVP", "MPEG1", "Hsp68", "g6378", "NLRP3.7")

# Dot plots - the size of the dot corresponds to the percentage of cells expressing the feature in each cluster. The color represents the average expression level
imm_dot1 <- DotPlot(oculina_sym, features = immune_features_small) + RotatedAxis()
imm_dot2 <- DotPlot(oculina_apo, features = immune_features_small) + RotatedAxis()

#immune_dotplot_small as 14x6 landscape

#get the top20 from the Immune Cell
immune_markers_lvld<- FindAllMarkers(IC, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
nrow(immune_markers_lvld)
#980

immune_markers_lvld %>% group_by(cluster) %>% top_n(10, avg_log2FC) -> top10_lev_im
top10_lev_im
nrow(top10_lev_im)
#30
#gives the top 10 genes for each cluster; 3 clusters

#get top20
immune_markers_lvld %>% group_by(cluster) %>% top_n(20, avg_log2FC) -> top20_lev_im

colnames(top20_lev_im) <- paste(c("p_val", "avg_log2FC", "pct.1", "pct.2", "p_val_adj", "cluster", "Preferred_name"))
top20_lev_im_pfam <- top20_lev_im %>%
  left_join(gene2pfam)
#Only including those with pfams
top20_named_imm <- c("g10019", "unc.22.4", "CD59.2", "g12921", "g23618", "g5302", "CSTB", "ACTG1.2", "g17493", "EPDR1.1", "g5301", "EEF1D", "ST13", "TXNDC8", "CTSB", "g16890", "c5", "g9620", "ACTG1", "TNR.17", "EPHA5", "ABCB4", "spn.E.25", "g26924", "g6298", "MICALL1.1", "c5.2", "COL26A1", "EFNB2.2", "g9076", "MCOLN2", "g1169", "g28737", "g4914", "g11436", "g13091", "g29974", "B3GNT5", "IFIH1", "g14045")

DoHeatmap(IC, features = top20_named_imm,label=T, group.colors = c("#CC6699", "#33CC99", "#3333FF")) + theme(axis.text.y= element_text(size=7)) + scale_fill_gradientn(colors=color)
#saved as top20_immune_named_2025 as landscape 6x6

```

8. Epidermis 1 RES OF 0.5
```{r}
epi1 <- subset(oculina_named_new, idents = "Epidermis 1") 

epi1 <- FindVariableFeatures(epi1)
epi1 <- ScaleData(epi1, features = features_to_scale)
epi1 <- RunPCA(epi1)
epi1 <- FindNeighbors(epi1, dims = 1:30, reduction = "pca")
epi1 <- FindClusters(epi1, resolution = 0.5)
#4 cluster
epi1 <- RunUMAP(epi1, reduction="pca", dims=1:30, return.model=TRUE)

epi1@meta.data$seurat_clusters <- as.numeric(as.character(epi1@meta.data$seurat_clusters))+1
Idents(epi1)<- "seurat_clusters"
epi1$seurat_clusters
epi1@active.ident <- factor(x = epi1@active.ident, levels = 1:4)
epi1@active.ident

levels(epi1) <- c("1", "2", "3", "4")

plot19 <- DimPlot(epi1, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99", "#3333FF", "#FF9900"), alpha=0.5)
#4 clusters

plot20 <- DimPlot(epi1, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot20b <- DimPlot(epi1, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99", "#3333FF", "#FF9900"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_epi1_plot19plot20_2025 9x4
#umap_epi1_plot20b_2025 8x4 landscape
```

```{r}
epi1_topfeat <- TopFeatures(object = epi1[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
epi1[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_epi1 <- FetchData(epi1, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_epi1 <- FetchData(epi1, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_epi1, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_epi1, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_epi1 <- Stdev(epi1, reduction = "pca")

pca_epi1 <- pc_data_epi1
pca_epi1$sym_state <- epi1@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_epi1)) #8.85e-09 ***
summary(aov(PC_2 ~ sym_state, data = pca_epi1)) #1.24e-12 ***
summary(aov(PC_3 ~ sym_state, data = pca_epi1)) #3.73e-09 ***
summary(aov(PC_4 ~ sym_state, data = pca_epi1)) #0.155
summary(aov(PC_5 ~ sym_state, data = pca_epi1)) #0.000121 ***

summary(aov(PC_6 ~ sym_state, data = pca_epi1)) #0.734
summary(aov(PC_7 ~ sym_state, data = pca_epi1)) #0.00339 **
summary(aov(PC_8 ~ sym_state, data = pca_epi1)) #0.359
summary(aov(PC_9 ~ sym_state, data = pca_epi1)) #1.25e-11 ***
summary(aov(PC_10 ~ sym_state, data = pca_epi1)) #0.854


DefaultAssay(epi1) <- "RNA"
epi1[["RNA"]]$counts<-as.matrix(epi1[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
epi1_mark <- FindMarkers(epi1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(epi1_mark)
#neg lfc means down in sym; pos lfc means up in sym

epi1_mark <- rownames_to_column(epi1_mark, var="Preferred_name")

#look at subcluster marker genes
epi1_markers_subclust <- FindAllMarkers(epi1, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

epi1_markers_subclust
epi1_markers_subclust <- rownames_to_column(epi1_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of epi1
epi1_1 <- subset(epi1, idents="1")
DefaultAssay(epi1_1) <- "RNA"

epi1_1_mark <- FindMarkers(epi1_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
epi1_1_mark <- rownames_to_column(epi1_1_mark, var="Preferred_name")
head(epi1_1_mark)

#get apo vs sym differences for subcluster 2 of epi1
epi1_2 <- subset(epi1, idents="2")
DefaultAssay(epi1_2) <- "RNA"

epi1_2_mark <- FindMarkers(epi1_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
epi1_2_mark <- rownames_to_column(epi1_2_mark, var="Preferred_name")
head(epi1_2_mark)

#get apo vs sym differences for subcluster 3 of epi1
epi1_3 <- subset(epi1, idents="3")
DefaultAssay(epi1_3) <- "RNA"

epi1_3_mark <- FindMarkers(epi1_3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
epi1_3_mark <- rownames_to_column(epi1_3_mark, var="Preferred_name")
head(epi1_3_mark)

#get apo vs sym differences for subcluster 4 of epi1
epi1_4 <- subset(epi1, idents="4")
DefaultAssay(epi1_4) <- "RNA"

epi1_4_mark <- FindMarkers(epi1_4, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
epi1_4_mark <- rownames_to_column(epi1_4_mark, var="Preferred_name")
head(epi1_4_mark)
```

```{r}
#get genes from signficant PCs (1, 2, 3, 5, 7, 9)
print(epi1[["pca"]], dims = 1:10, nfeatures = 10)

epi1_feats <- c("TRAF3.3", "Unc.89", "SIK2", "g9710", "g18665", "MFI2.4", "ELF2", "g20910", "HSF1", "g18093", "NHP2L1", "g24835", "Cyp1", "ATP5G2", "awd", "TMA7", "PA2G4", "RPS28", "g12827", "SERBP1", "g8803", "STX17", "g25627", "g17270", "g9629", "g12705", "SOX8", "COL4A1", "g19708", "g27569", "FOSL2.2", "JUN.1", "TRAF3.3", "g20910", "DNAJB5", "g14761", "FLI1.1", "HSF1", "HSPA8", "Hsp68.1", "OFD1", "BLM", "HSF1", "g18093", "LRRIQ1.1", "DSTYK", "APOB", "KATNBL1", "g18665", "SIK2", "g30147", "RPS28", "g20955", "g12980", "Cyp1", "TFF1", "g26349", "CRIP2", "SPON1", "g12976", "UNC5C.1", "g4979", "g13041", "UNC5C", "ELK1", "g9711", "g29618", "IRF1.1", "g9629", "g8173", "g4247", "TFF1", "HMCN1.26", "g21303", "g20195", "ADAMTS16", "g9196", "g4387", "g4859", "g20274", "g22757", "g23088", "g18430", "g29516", "g30337", "g29508", "INSM2", "g18434", "NAALAD2.1", "FOSL2.2", "HMCN1.26", "TAGLN3.1", "TFF1", "g4247", "g4387", "g20195", "g20837", "MYL12A.1", "g9196", "g1192", "INSM2", "g25251", "g20608", "MYC", "g9970", "SMARCB1", "CD59.2", "HIGD1A", "g14761", "g13439", "MFI2.4", "g14973", "STK31", "CNKSR1", "g29618", "Unc.89", "g4859", "FAM21A", "g16396", "CRIP2") 
#for now not removing the unnannotated genes

epi1_feats_mark <- subset(epi1_mark, Preferred_name %in% epi1_feats)
# add a column of NAs
epi1_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
epi1_feats_mark$diffexpressed[epi1_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
epi1_feats_mark$diffexpressed[epi1_feats_mark$avg_log2FC < 0] <- "apo"

epi1_feats2go <- epi1_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

epi1_feats2go2cog <- epi1_feats2go %>%
  left_join(gene2cog)
head(epi1_feats2go2cog)
epi1_feats2go2cog$COG_category[epi1_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
epi1_feats2go2cog2pfam <- epi1_feats2go2cog %>%
  left_join(gene2pfam)
epi1_feats2go2cog2pfam <- subset(epi1_feats2go2cog2pfam, epi1_feats2go2cog2pfam$PFAMs!="-")

epi1_feats2go2cog2pfam <- epi1_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

epi1_feats_bar_sort <- ggplot(data=epi1_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### epi1_feats_barsort_2025 as 8x4
```

9. Epidermis 2 RES OF 0.3
```{r}
epi2 <- subset(oculina_named_new, idents = "Epidermis 2") 

epi2 <- FindVariableFeatures(epi2)
epi2 <- ScaleData(epi2, features = features_to_scale)
epi2 <- RunPCA(epi2)
epi2 <- FindNeighbors(epi2, dims = 1:30, reduction = "pca")
epi2 <- FindClusters(epi2, resolution = 0.3)
#2 cluster
epi2 <- RunUMAP(epi2, reduction="pca", dims=1:30, return.model=TRUE)

epi2@meta.data$seurat_clusters <- as.numeric(as.character(epi2@meta.data$seurat_clusters))+1
Idents(epi2)<- "seurat_clusters"
epi2$seurat_clusters
epi2@active.ident <- factor(x = epi2@active.ident, levels = 1:2)
epi2@active.ident

levels(epi2) <- c("1", "2")

plot21 <- DimPlot(epi2, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 clusters

plot22 <- DimPlot(epi2, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot22b <- DimPlot(epi2, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_epi2_plot21plot22_2025 9x4
#umap_epi2_plot22b_2025 8x4 landscape
```

```{r}
epi2_topfeat <- TopFeatures(object = epi2[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
epi2[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_epi2 <- FetchData(epi2, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_epi2 <- FetchData(epi2, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_epi2, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_epi2, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_epi2 <- Stdev(epi2, reduction = "pca")

pca_epi2 <- pc_data_epi2
pca_epi2$sym_state <- epi2@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_epi2)) #0.0949 .
summary(aov(PC_2 ~ sym_state, data = pca_epi2)) #0.034 *
summary(aov(PC_3 ~ sym_state, data = pca_epi2)) #0.215
summary(aov(PC_4 ~ sym_state, data = pca_epi2)) #0.467
summary(aov(PC_5 ~ sym_state, data = pca_epi2)) #0.0177 *

summary(aov(PC_6 ~ sym_state, data = pca_epi2)) #0.265
summary(aov(PC_7 ~ sym_state, data = pca_epi2)) #0.522
summary(aov(PC_8 ~ sym_state, data = pca_epi2)) #0.185
summary(aov(PC_9 ~ sym_state, data = pca_epi2)) #0.549
summary(aov(PC_10 ~ sym_state, data = pca_epi2)) #0.974


DefaultAssay(epi2) <- "RNA"
epi2[["RNA"]]$counts<-as.matrix(epi2[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
epi2_mark <- FindMarkers(epi2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(epi2_mark)
#neg lfc means down in sym; pos lfc means up in sym

epi2_mark <- rownames_to_column(epi2_mark, var="Preferred_name")

#look at subcluster marker genes
epi2_markers_subclust <- FindAllMarkers(epi2, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

epi2_markers_subclust
epi2_markers_subclust <- rownames_to_column(epi2_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of epi2
epi2_1 <- subset(epi2, idents="1")
DefaultAssay(epi2_1) <- "RNA"

epi2_1_mark <- FindMarkers(epi2_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
epi2_1_mark <- rownames_to_column(epi2_1_mark, var="Preferred_name")
head(epi2_1_mark)

#get apo vs sym differences for subcluster 2 of epi2
epi2_2 <- subset(epi2, idents="2")
DefaultAssay(epi2_2) <- "RNA"

epi2_2_mark <- FindMarkers(epi2_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
epi2_2_mark <- rownames_to_column(epi2_2_mark, var="Preferred_name")
head(epi2_2_mark)
```

```{r}
#get genes from signficant PCs (2, 5)
print(epi2[["pca"]], dims = 1:10, nfeatures = 10)

epi2_feats <- c("g12705", "g12579", "g22247", "g12702", "HPGDS.1", "g30177", "g19484", "g24497", "g21082", "SDC2", "HSF1", "TRAF3.3", "DLC1", "g20910", "XIRP2", "SLC28A3.1", "g27537", "SZT2", "ARHGEF4", "RET.5", "TNC.11", "ROBO1", "g9177", "g22528", "g12501", "g18001", "g15463", "MYO1", "PFKFB2.1", "g3348", "E2F7", "ATXN7", "SYF2", "ATAD5.1", "NIPBL", "g3775", "NCAPG2", "SLC28A3.1", "g17234", "g25629") 
#for now not removing the unnannotated genes

epi2_feats_mark <- subset(epi2_mark, Preferred_name %in% epi2_feats)
# add a column of NAs
epi2_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
epi2_feats_mark$diffexpressed[epi2_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
epi2_feats_mark$diffexpressed[epi2_feats_mark$avg_log2FC < 0] <- "apo"

epi2_feats2go <- epi2_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

epi2_feats2go2cog <- epi2_feats2go %>%
  left_join(gene2cog)
head(epi2_feats2go2cog)
epi2_feats2go2cog$COG_category[epi2_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
epi2_feats2go2cog2pfam <- epi2_feats2go2cog %>%
  left_join(gene2pfam)
epi2_feats2go2cog2pfam <- subset(epi2_feats2go2cog2pfam, epi2_feats2go2cog2pfam$PFAMs!="-")

epi2_feats2go2cog2pfam <- epi2_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

epi2_feats_bar_sort <- ggplot(data=epi2_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### epi2_feats_barsort_2025 as 6x4
```

10. Epidermis 3 RES OF 0.3
```{r}
epi3 <- subset(oculina_named_new, idents = "Epidermis 3") 

epi3 <- FindVariableFeatures(epi3)
epi3 <- ScaleData(epi3, features = features_to_scale)
epi3 <- RunPCA(epi3)
epi3 <- FindNeighbors(epi3, dims = 1:30, reduction = "pca")
epi3 <- FindClusters(epi3, resolution = 0.3)
#1 cluster
epi3 <- RunUMAP(epi3, reduction="pca", dims=1:30, return.model=TRUE)

epi3@meta.data$seurat_clusters <- as.numeric(as.character(epi3@meta.data$seurat_clusters))+1
Idents(epi3)<- "seurat_clusters"
epi3$seurat_clusters
epi3@active.ident <- factor(x = epi3@active.ident, levels = 1)
epi3@active.ident

levels(epi3) <- c("1")

plot23 <- DimPlot(epi3, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot24 <- DimPlot(epi3, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot24b <- DimPlot(epi3, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_epi3_plot23plot24_2025 9x4
#umap_epi3_plot24b_2025 8x4 landscape
```

```{r}
epi3_topfeat <- TopFeatures(object = epi3[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
epi3[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_epi3 <- FetchData(epi3, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_epi3 <- FetchData(epi3, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_epi3, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_epi3, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_epi3 <- Stdev(epi3, reduction = "pca")

pca_epi3 <- pc_data_epi3
pca_epi3$sym_state <- epi3@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_epi3)) #0.208
summary(aov(PC_2 ~ sym_state, data = pca_epi3)) #0.000145 ***
summary(aov(PC_3 ~ sym_state, data = pca_epi3)) #0.298
summary(aov(PC_4 ~ sym_state, data = pca_epi3)) #0.116
summary(aov(PC_5 ~ sym_state, data = pca_epi3)) #0.247

summary(aov(PC_6 ~ sym_state, data = pca_epi3)) #0.0889 .
summary(aov(PC_7 ~ sym_state, data = pca_epi3)) #0.0371 *
summary(aov(PC_8 ~ sym_state, data = pca_epi3)) #0.462
summary(aov(PC_9 ~ sym_state, data = pca_epi3)) #0.561
summary(aov(PC_10 ~ sym_state, data = pca_epi3)) #0.423


DefaultAssay(epi3) <- "RNA"
epi3[["RNA"]]$counts<-as.matrix(epi3[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
epi3_mark <- FindMarkers(epi3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(epi3_mark)
#neg lfc means down in sym; pos lfc means up in sym

epi3_mark <- rownames_to_column(epi3_mark, var="Preferred_name")
```

```{r}
#only one cluster
```

```{r}
#get genes from signficant PCs (2, 7)
print(epi3[["pca"]], dims = 1:10, nfeatures = 10)

epi3_feats <- c("PA2G4", "HNRNPA3", "g12827", "DKC1", "ANP32B", "CRELD2", "CCT8L2", "RPS23", "PDIA3", "ESCO1", "ZFAND6", "g8941", "PRDM6", "g3550", "g5543", "g21922", "g9635", "g15539", "g22681", "NOXO1", "CAMK4", "HIP1", "g21922", "CRB1.2", "ABCB6", "g12049", "MRPS14", "Mcm7", "EOGT", "g16553", "g11730", "P4HA2", "ISYNA1", "HPN", "g13446", "CCT8L2", "g23646", "SUGP1", "RNF25", "g25957") 
#for now not removing the unnannotated genes

epi3_feats_mark <- subset(epi3_mark, Preferred_name %in% epi3_feats)
# add a column of NAs
epi3_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
epi3_feats_mark$diffexpressed[epi3_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
epi3_feats_mark$diffexpressed[epi3_feats_mark$avg_log2FC < 0] <- "apo"

epi3_feats2go <- epi3_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

epi3_feats2go2cog <- epi3_feats2go %>%
  left_join(gene2cog)
head(epi3_feats2go2cog)
epi3_feats2go2cog$COG_category[epi3_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
epi3_feats2go2cog2pfam <- epi3_feats2go2cog %>%
  left_join(gene2pfam)
epi3_feats2go2cog2pfam <- subset(epi3_feats2go2cog2pfam, epi3_feats2go2cog2pfam$PFAMs!="-")

epi3_feats2go2cog2pfam <- epi3_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

epi3_feats_bar_sort <- ggplot(data=epi3_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### epi3_feats_barsort_2025 as 6x4
```

11. Epidermis 4 RES OF 0.3
```{r}
epi4 <- subset(oculina_named_new, idents = "Epidermis 4") 

epi4 <- FindVariableFeatures(epi4)
epi4 <- ScaleData(epi4, features = features_to_scale)
epi4 <- RunPCA(epi4)
epi4 <- FindNeighbors(epi4, dims = 1:30, reduction = "pca")
epi4 <- FindClusters(epi4, resolution = 0.3)
#1 cluster
epi4 <- RunUMAP(epi4, reduction="pca", dims=1:30, return.model=TRUE)

epi4@meta.data$seurat_clusters <- as.numeric(as.character(epi4@meta.data$seurat_clusters))+1
Idents(epi4)<- "seurat_clusters"
epi4$seurat_clusters
epi4@active.ident <- factor(x = epi4@active.ident, levels = 1)
epi4@active.ident

levels(epi4) <- c("1")

plot25 <- DimPlot(epi4, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot26 <- DimPlot(epi4, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot26b <- DimPlot(epi4, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_epi4_plot25plot26_2025 9x4
#umap_epi4_plot26b_2025 8x4 landscape
```

```{r}
epi4_topfeat <- TopFeatures(object = epi4[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
epi4[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_epi4 <- FetchData(epi4, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_epi4 <- FetchData(epi4, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_epi4, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_epi4, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_epi4 <- Stdev(epi4, reduction = "pca")

pca_epi4 <- pc_data_epi4
pca_epi4$sym_state <- epi4@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_epi4)) #3.35e-05 ***
summary(aov(PC_2 ~ sym_state, data = pca_epi4)) #0.000906 ***
summary(aov(PC_3 ~ sym_state, data = pca_epi4)) #0.0492 *
summary(aov(PC_4 ~ sym_state, data = pca_epi4)) #0.0171 *
summary(aov(PC_5 ~ sym_state, data = pca_epi4)) #0.836

summary(aov(PC_6 ~ sym_state, data = pca_epi4)) #0.00162 **
summary(aov(PC_7 ~ sym_state, data = pca_epi4)) #0.295
summary(aov(PC_8 ~ sym_state, data = pca_epi4)) #0.0131 *
summary(aov(PC_9 ~ sym_state, data = pca_epi4)) #0.305
summary(aov(PC_10 ~ sym_state, data = pca_epi4)) #0.164


DefaultAssay(epi4) <- "RNA"
epi4[["RNA"]]$counts<-as.matrix(epi4[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
epi4_mark <- FindMarkers(epi4, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(epi4_mark)
#neg lfc means down in sym; pos lfc means up in sym

epi4_mark <- rownames_to_column(epi4_mark, var="Preferred_name")

#look at subcluster marker genes
#only one subcluster
```

```{r}
#Only one subcluster
```

```{r}
#get genes from signficant PCs (1-4, 6, 8)
print(epi4[["pca"]], dims = 1:10, nfeatures = 10)

epi4_feats <- c("CEBPA", "g20955", "g23998", "g18649", "VPS41", "g24251", "HECTD1.1", "g28509", "g24496", "g4798", "CCT8L2", "g12582", "WDR3", "HELLS", "RRN3", "VAC14", "PNO1", "DDX21", "URI1", "TNFSF10.1", "RRM2B.1", "g21594", "g12582", "g4584", "g18731", "MCPH1", "PCNA", "RRM1", "g20605", "NCAPD2", "g29643", "g12779", "SMCR8", "g15683", "TMX3", "g26203", "g12008", "DAGLA", "CUL3", "ODF3L2", "LONRF3", "HIAT1", "g1407", "g17566", "EIF2AK1", "MRPS5", "ZKSCAN8", "ANKS6", "LARS", "PNO1", "g12779", "g24335", "g15683", "g10897", "g8388", "SMCR8", "CUL3", "SWT1.1", "g26203", "g11635", "NDUFS8", "PXDN.14", "SLC35E1", "ASZ1", "PIGK", "NFKBIL1", "g16199", "C1orf35", "g10842", "ZRANB2", "g1686", "MAN2A2", "g6932", "g9971", "BCS1L", "g11846", "g1365", "FUCA1", "IFT20", "pyx", "Gdi", "g4235", "g6846", "RRN3", "BCS1L", "MFI2.4", "NELFB", "MAN2A2", "PSMD2", "g22917", "g17063", "EIF2AK1", "g20910", "g21913", "TRAFD1", "BAAT", "g17566", "g22783", "g13247", "g8878", "KIF13B", "BHMT.9", "g8527", "g9356", "PSEN1", "g9245", "CENPC", "TNFSF10.1", "LRP4", "g29989", "g6932", "g9971", "g10842", "g6408", "C1orf194", "g1686", "g8076", "g13386", "g17909", "PXDN.14") 
#for now not removing the unnannotated genes

epi4_feats_mark <- subset(epi4_mark, Preferred_name %in% epi4_feats)
# add a column of NAs
epi4_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
epi4_feats_mark$diffexpressed[epi4_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
epi4_feats_mark$diffexpressed[epi4_feats_mark$avg_log2FC < 0] <- "apo"

epi4_feats2go <- epi4_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

epi4_feats2go2cog <- epi4_feats2go %>%
  left_join(gene2cog)
head(epi4_feats2go2cog)
epi4_feats2go2cog$COG_category[epi4_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
epi4_feats2go2cog2pfam <- epi4_feats2go2cog %>%
  left_join(gene2pfam)
epi4_feats2go2cog2pfam <- subset(epi4_feats2go2cog2pfam, epi4_feats2go2cog2pfam$PFAMs!="-")

epi4_feats2go2cog2pfam <- epi4_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

epi4_feats_bar_sort <- ggplot(data=epi4_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### epi4_feats_barsort_2025 as 12x4
```

12. Epidermis 5 RES OF 0.3
```{r}
epi5 <- subset(oculina_named_new, idents = "Epidermis 5") 

epi5 <- FindVariableFeatures(epi5)
epi5 <- ScaleData(epi5, features = features_to_scale)
epi5 <- RunPCA(epi5)
epi5 <- FindNeighbors(epi5, dims = 1:30, reduction = "pca")
epi5 <- FindClusters(epi5, resolution = 0.3)
#2 clusters
epi5 <- RunUMAP(epi5, reduction="pca", dims=1:30, return.model=TRUE)

epi5@meta.data$seurat_clusters <- as.numeric(as.character(epi5@meta.data$seurat_clusters))+1
Idents(epi5)<- "seurat_clusters"
epi5$seurat_clusters
epi5@active.ident <- factor(x = epi5@active.ident, levels = 1:2)
epi5@active.ident

levels(epi5) <- c("1", "2")

plot27 <- DimPlot(epi5, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 cluster

plot28 <- DimPlot(epi5, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot28b <- DimPlot(epi5, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_epi5_plot27plot28_2025 9x4
#umap_epi5_plot28b_2025 8x4 landscape
```

```{r}
epi5_topfeat <- TopFeatures(object = epi5[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
epi5[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_epi5 <- FetchData(epi5, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_epi5 <- FetchData(epi5, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_epi5, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_epi5, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_epi5 <- Stdev(epi5, reduction = "pca")

pca_epi5 <- pc_data_epi5
pca_epi5$sym_state <- epi5@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_epi5)) #0.11
summary(aov(PC_2 ~ sym_state, data = pca_epi5)) #0.00716 **
summary(aov(PC_3 ~ sym_state, data = pca_epi5)) #0.213
summary(aov(PC_4 ~ sym_state, data = pca_epi5)) #0.00307 **
summary(aov(PC_5 ~ sym_state, data = pca_epi5)) #0.389

summary(aov(PC_6 ~ sym_state, data = pca_epi5)) #0.875
summary(aov(PC_7 ~ sym_state, data = pca_epi5)) #0.86
summary(aov(PC_8 ~ sym_state, data = pca_epi5)) #0.22
summary(aov(PC_9 ~ sym_state, data = pca_epi5)) #0.623
summary(aov(PC_10 ~ sym_state, data = pca_epi5)) #0.48


DefaultAssay(epi5) <- "RNA"
epi5[["RNA"]]$counts<-as.matrix(epi5[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
epi5_mark <- FindMarkers(epi5, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(epi5_mark)
#neg lfc means down in sym; pos lfc means up in sym

epi5_mark <- rownames_to_column(epi5_mark, var="Preferred_name")

#look at subcluster marker genes
epi5_markers_subclust <- FindAllMarkers(epi5, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

epi5_markers_subclust
epi5_markers_subclust <- rownames_to_column(epi5_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of epi5
epi5_1 <- subset(epi5, idents="1")
DefaultAssay(epi5_1) <- "RNA"

epi5_1_mark <- FindMarkers(epi5_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
epi5_1_mark <- rownames_to_column(epi5_1_mark, var="Preferred_name")
head(epi5_1_mark)

#get apo vs sym differences for subcluster 2 of epi5
epi5_2 <- subset(epi5, idents="2")
DefaultAssay(epi5_2) <- "RNA"

epi5_2_mark <- FindMarkers(epi5_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
epi5_2_mark <- rownames_to_column(epi5_2_mark, var="Preferred_name")
head(epi5_2_mark)
```

```{r}
#get genes from signficant PCs (2, 4)
print(epi5[["pca"]], dims = 1:10, nfeatures = 10)

epi5_feats <- c("CA9.2", "g1933", "PPP1R15A", "g29315", "g6497", "g443", "g28357", "g16536", "g24244", "g9048", "KCNC3", "g6592", "g356", "ELMOD3", "SLC28A3.1", "g22804", "CSTF1", "TRPA1.9", "g20831", "PEAK1", "KAZN", "BRF1", "g25300", "g23951", "g6019", "BTBD16", "SLC6A18", "g3731", "DNAJB9", "g25785", "g18057", "g1043", "GCLC", "g257", "g3842", "EHMT1", "SH3RF3", "g12146", "g7413", "g17180") 
#for now not removing the unnannotated genes

epi5_feats_mark <- subset(epi5_mark, Preferred_name %in% epi5_feats)
# add a column of NAs
epi5_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
epi5_feats_mark$diffexpressed[epi5_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
epi5_feats_mark$diffexpressed[epi5_feats_mark$avg_log2FC < 0] <- "apo"

epi5_feats2go <- epi5_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

epi5_feats2go2cog <- epi5_feats2go %>%
  left_join(gene2cog)
head(epi5_feats2go2cog)
epi5_feats2go2cog$COG_category[epi5_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
epi5_feats2go2cog2pfam <- epi5_feats2go2cog %>%
  left_join(gene2pfam)
epi5_feats2go2cog2pfam <- subset(epi5_feats2go2cog2pfam, epi5_feats2go2cog2pfam$PFAMs!="-")

epi5_feats2go2cog2pfam <- epi5_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

epi5_feats_bar_sort <- ggplot(data=epi5_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### epi5_feats_barsort_2025 as 6x4
```

13. Neuron 1 RES OF 0.3
```{r}
n1 <- subset(oculina_named_new, idents = "Neuron 1") 

n1 <- FindVariableFeatures(n1)
n1 <- ScaleData(n1, features = features_to_scale)
n1 <- RunPCA(n1)
n1 <- FindNeighbors(n1, dims = 1:30, reduction = "pca")
n1 <- FindClusters(n1, resolution = 0.3)
#5 clusters
n1 <- RunUMAP(n1, reduction="pca", dims=1:30, return.model=TRUE)

n1@meta.data$seurat_clusters <- as.numeric(as.character(n1@meta.data$seurat_clusters))+1
Idents(n1)<- "seurat_clusters"
n1$seurat_clusters
n1@active.ident <- factor(x = n1@active.ident, levels = 1:5)
n1@active.ident

levels(n1) <- c("1", "2", "3", "4", "5")

plot29 <- DimPlot(n1, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99", "#3333FF", "#FF9900", "gray"), alpha=0.5)
#5 clusters, kinda messy but going up to 6 custers at res of 0.5 doesn't help

plot30 <- DimPlot(n1, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot30b <- DimPlot(n1, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99", "#3333FF", "#FF9900", "gray"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n1_plot29plot30_2025 9x4
#umap_n1_plot30b_2025 8x4 landscape
```

```{r}
n1_topfeat <- TopFeatures(object = n1[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n1[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n1 <- FetchData(n1, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n1 <- FetchData(n1, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n1, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n1, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n1 <- Stdev(n1, reduction = "pca")

pca_n1 <- pc_data_n1
pca_n1$sym_state <- n1@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n1)) #0.452
summary(aov(PC_2 ~ sym_state, data = pca_n1)) #0.567
summary(aov(PC_3 ~ sym_state, data = pca_n1)) #0.000445 ***
summary(aov(PC_4 ~ sym_state, data = pca_n1)) #0.724
summary(aov(PC_5 ~ sym_state, data = pca_n1)) #0.0952 .

summary(aov(PC_6 ~ sym_state, data = pca_n1)) #0.0343 *
summary(aov(PC_7 ~ sym_state, data = pca_n1)) #0.817
summary(aov(PC_8 ~ sym_state, data = pca_n1)) #0.251
summary(aov(PC_9 ~ sym_state, data = pca_n1)) #0.169
summary(aov(PC_10 ~ sym_state, data = pca_n1)) #0.449


DefaultAssay(n1) <- "RNA"
n1[["RNA"]]$counts<-as.matrix(n1[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n1_mark <- FindMarkers(n1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n1_mark)
#neg lfc means down in sym; pos lfc means up in sym

n1_mark <- rownames_to_column(n1_mark, var="Preferred_name")

#look at subcluster marker genes
n1_markers_subclust <- FindAllMarkers(n1, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

n1_markers_subclust
n1_markers_subclust <- rownames_to_column(n1_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of n1
n1_1 <- subset(n1, idents="1")
DefaultAssay(n1_1) <- "RNA"

n1_1_mark <- FindMarkers(n1_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
n1_1_mark <- rownames_to_column(n1_1_mark, var="Preferred_name")
head(n1_1_mark)

#get apo vs sym differences for subcluster 2 of n1
n1_2 <- subset(n1, idents="2")
DefaultAssay(n1_2) <- "RNA"

n1_2_mark <- FindMarkers(n1_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
n1_2_mark <- rownames_to_column(n1_2_mark, var="Preferred_name")
head(n1_2_mark)

#get apo vs sym differences for subcluster 3 of n1
n1_3 <- subset(n1, idents="3")
DefaultAssay(n1_3) <- "RNA"

n1_3_mark <- FindMarkers(n1_3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
n1_3_mark <- rownames_to_column(n1_3_mark, var="Preferred_name")
head(n1_3_mark)

#get apo vs sym differences for subcluster 4 of n1
n1_4 <- subset(n1, idents="4")
DefaultAssay(n1_4) <- "RNA"

n1_4_mark <- FindMarkers(n1_4, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
n1_4_mark <- rownames_to_column(n1_4_mark, var="Preferred_name")
head(n1_4_mark)

#get apo vs sym differences for subcluster 5 of n1
n1_5 <- subset(n1, idents="5")
DefaultAssay(n1_5) <- "RNA"

n1_5_mark <- FindMarkers(n1_5, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
n1_5_mark <- rownames_to_column(n1_5_mark, var="Preferred_name")
head(n1_5_mark)
```

```{r}
#get genes from signficant PCs (3, 6)
print(n1[["pca"]], dims = 1:10, nfeatures = 10)

n1_feats <- c("g3690", "g17427", "g13303", "g4958", "g17610", "CA14.1", "Rdl.7", "g13161", "g8521", "g17349", "g24655", "g12942", "CALM3", "g22246", "g21996", "CBFB", "g29242", "g18361", "g12796", "g26140", "g18297", "g18624", "P4HTM.14", "CALB1.2", "g26610", "g22435", "CSMD2.3", "g18393", "g24576", "g17371", "g466", "g12008", "NOTCH1", "g27152", "PAX1", "g13744", "g22781", "PXR1", "DGKE.1", "g4859") 
#for now not removing the unnannotated genes

n1_feats_mark <- subset(n1_mark, Preferred_name %in% n1_feats)
# add a column of NAs
n1_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n1_feats_mark$diffexpressed[n1_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n1_feats_mark$diffexpressed[n1_feats_mark$avg_log2FC < 0] <- "apo"

n1_feats2go <- n1_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n1_feats2go2cog <- n1_feats2go %>%
  left_join(gene2cog)
head(n1_feats2go2cog)
n1_feats2go2cog$COG_category[n1_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n1_feats2go2cog2pfam <- n1_feats2go2cog %>%
  left_join(gene2pfam)
n1_feats2go2cog2pfam <- subset(n1_feats2go2cog2pfam, n1_feats2go2cog2pfam$PFAMs!="-")

n1_feats2go2cog2pfam <- n1_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n1_feats_bar_sort <- ggplot(data=n1_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n1_feats_barsort_2025 as 5x4
```

14. Neuron 2 RES OF 0.3
```{r}
n2 <- subset(oculina_named_new, idents = "Neuron 2") 

n2 <- FindVariableFeatures(n2)
n2 <- ScaleData(n2, features = features_to_scale)
n2 <- RunPCA(n2)
n2 <- FindNeighbors(n2, dims = 1:30, reduction = "pca")
n2 <- FindClusters(n2, resolution = 0.3)
#1 cluster
n2 <- RunUMAP(n2, reduction="pca", dims=1:30, return.model=TRUE)

n2@meta.data$seurat_clusters <- as.numeric(as.character(n2@meta.data$seurat_clusters))+1
Idents(n2)<- "seurat_clusters"
n2$seurat_clusters
n2@active.ident <- factor(x = n2@active.ident, levels = 1)
n2@active.ident

levels(n2) <- c("1")

plot31 <- DimPlot(n2, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot32 <- DimPlot(n2, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot32b <- DimPlot(n2, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n2_plot31plot32_2025 9x4
#umap_n2_plot32b_2025 8x4 landscape
```

```{r}
n2_topfeat <- TopFeatures(object = n2[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n2[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n2 <- FetchData(n2, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n2 <- FetchData(n2, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n2, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n2, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n2 <- Stdev(n2, reduction = "pca")

pca_n2 <- pc_data_n2
pca_n2$sym_state <- n2@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n2)) #0.585
summary(aov(PC_2 ~ sym_state, data = pca_n2)) #0.0413 *
summary(aov(PC_3 ~ sym_state, data = pca_n2)) #0.0349 *
summary(aov(PC_4 ~ sym_state, data = pca_n2)) #0.213
summary(aov(PC_5 ~ sym_state, data = pca_n2)) #0.407

summary(aov(PC_6 ~ sym_state, data = pca_n2)) #0.0262 *
summary(aov(PC_7 ~ sym_state, data = pca_n2)) #0.322
summary(aov(PC_8 ~ sym_state, data = pca_n2)) #0.00805 **
summary(aov(PC_9 ~ sym_state, data = pca_n2)) #0.414
summary(aov(PC_10 ~ sym_state, data = pca_n2)) #0.101


DefaultAssay(n2) <- "RNA"
n2[["RNA"]]$counts<-as.matrix(n2[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n2_mark <- FindMarkers(n2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n2_mark)
#neg lfc means down in sym; pos lfc means up in sym

n2_mark <- rownames_to_column(n2_mark, var="Preferred_name")

#look at subcluster marker genes
#only 1 subcluster
```

```{r}
#only one subcluster
```

```{r}
#get genes from signficant PCs (2, 3, 6, 8)
print(n2[["pca"]], dims = 1:10, nfeatures = 10)

n2_feats <- c("B3GALT6", "g6055", "CCDC40", "EXOC4", "RAD17", "g13567", "NCAPD2", "GRM6", "CPEB2", "PDXK", "g16972", "g21623", "g23596", "g5136", "FIBCD1.4", "g14938", "g5349", "ACAT2", "CPPED1.1", "g6690", "B3GALT6", "RAD17", "g13567", "DPP8", "TRIM37", "TTC23", "g18589", "g6055", "g3648", "TNR.21", "g22415", "SSC4D", "g26420", "CPEB2", "TSPEAR.2", "GRM6", "FANCD2", "g6561", "TTBK2", "g12695", "NCAN.1", "g3063", "g16371", "ASTL", "g13742", "MCM5", "ARX", "g12284", "LYS1.2", "g3792", "g13567", "RAD17", "g6055", "TRIM37", "GFPT1", "CHORDC1", "g10837", "g26772", "g3648", "PIP5K1B", "ALG9", "KIF19", "g3063", "FGFR4.7", "ANO4", "DACH2", "BLM.1", "g1925", "WNT3", "ARX", "SKA2", "INCENP", "unc.132", "ITGA9.1", "Mcm7", "LIG1", "MCM2", "g18583", "PAICS", "g3649") 
#for now not removing the unnannotated genes

n2_feats_mark <- subset(n2_mark, Preferred_name %in% n2_feats)
# add a column of NAs
n2_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n2_feats_mark$diffexpressed[n2_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n2_feats_mark$diffexpressed[n2_feats_mark$avg_log2FC < 0] <- "apo"

n2_feats2go <- n2_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n2_feats2go2cog <- n2_feats2go %>%
  left_join(gene2cog)
head(n2_feats2go2cog)
n2_feats2go2cog$COG_category[n2_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n2_feats2go2cog2pfam <- n2_feats2go2cog %>%
  left_join(gene2pfam)
n2_feats2go2cog2pfam <- subset(n2_feats2go2cog2pfam, n2_feats2go2cog2pfam$PFAMs!="-")

n2_feats2go2cog2pfam <- n2_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n2_feats_bar_sort <- ggplot(data=n2_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n2_feats_barsort_2025 as 8x4
```

15. Neuron 3 RES OF 0.3
```{r}
n3 <- subset(oculina_named_new, idents = "Neuron 3") 

n3 <- FindVariableFeatures(n3)
n3 <- ScaleData(n3, features = features_to_scale)
n3 <- RunPCA(n3)
n3 <- FindNeighbors(n3, dims = 1:30, reduction = "pca")
n3 <- FindClusters(n3, resolution = 0.3)
#2 clusters
n3 <- RunUMAP(n3, reduction="pca", dims=1:30, return.model=TRUE)

n3@meta.data$seurat_clusters <- as.numeric(as.character(n3@meta.data$seurat_clusters))+1
Idents(n3)<- "seurat_clusters"
n3$seurat_clusters
n3@active.ident <- factor(x = n3@active.ident, levels = 1:2)
n3@active.ident

levels(n3) <- c("1", "2")

plot33 <- DimPlot(n3, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 clusters

plot34 <- DimPlot(n3, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot34b <- DimPlot(n3, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n3_plot33plot34_2025 9x4
#umap_n3_plot34b_2025 8x4 landscape
```

```{r}
n3_topfeat <- TopFeatures(object = n3[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n3[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n3 <- FetchData(n3, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n3 <- FetchData(n3, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n3, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n3, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n3 <- Stdev(n3, reduction = "pca")

pca_n3 <- pc_data_n3
pca_n3$sym_state <- n3@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n3)) #0.196
summary(aov(PC_2 ~ sym_state, data = pca_n3)) #0.663
summary(aov(PC_3 ~ sym_state, data = pca_n3)) #0.358
summary(aov(PC_4 ~ sym_state, data = pca_n3)) #0.772
summary(aov(PC_5 ~ sym_state, data = pca_n3)) #0.0902 .

summary(aov(PC_6 ~ sym_state, data = pca_n3)) #0.151
summary(aov(PC_7 ~ sym_state, data = pca_n3)) #0.306
summary(aov(PC_8 ~ sym_state, data = pca_n3)) #0.0078 **
summary(aov(PC_9 ~ sym_state, data = pca_n3)) #0.151
summary(aov(PC_10 ~ sym_state, data = pca_n3)) #0.598


DefaultAssay(n3) <- "RNA"
n3[["RNA"]]$counts<-as.matrix(n3[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n3_mark <- FindMarkers(n3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n3_mark)
#neg lfc means down in sym; pos lfc means up in sym

n3_mark <- rownames_to_column(n3_mark, var="Preferred_name")

#look at subcluster marker genes
n3_markers_subclust <- FindAllMarkers(n3, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

n3_markers_subclust
n3_markers_subclust <- rownames_to_column(n3_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of n3
n3_1 <- subset(n3, idents="1")
DefaultAssay(n3_1) <- "RNA"

n3_1_mark <- FindMarkers(n3_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
n3_1_mark <- rownames_to_column(n3_1_mark, var="Preferred_name")
head(n3_1_mark)

#get apo vs sym differences for subcluster 2 of n1
n3_2 <- subset(n3, idents="2")
DefaultAssay(n3_2) <- "RNA"

n3_2_mark <- FindMarkers(n3_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
n3_2_mark <- rownames_to_column(n3_2_mark, var="Preferred_name")
head(n3_2_mark)
```

```{r}
#get genes from signficant PCs (8)
print(n3[["pca"]], dims = 1:10, nfeatures = 10)

n3_feats <- c("ASTL", "g201", "g25756", "g6212", "PNLIPRP1", "Mmp1.12", "g5079", "g6876", "g22366", "PLA2G15.3", "g15307", "HIST1H3D.1", "CHRD", "g3058", "NCAPH2", "KIF23", "DEAF1.1", "CNTLN", "DCAF11", "ARL5B") 
#for now not removing the unnannotated genes

n3_feats_mark <- subset(n3_mark, Preferred_name %in% n3_feats)
# add a column of NAs
n3_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n3_feats_mark$diffexpressed[n3_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n3_feats_mark$diffexpressed[n3_feats_mark$avg_log2FC < 0] <- "apo"

n3_feats2go <- n3_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n3_feats2go2cog <- n3_feats2go %>%
  left_join(gene2cog)
head(n3_feats2go2cog)
n3_feats2go2cog$COG_category[n3_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n3_feats2go2cog2pfam <- n3_feats2go2cog %>%
  left_join(gene2pfam)
n3_feats2go2cog2pfam <- subset(n3_feats2go2cog2pfam, n3_feats2go2cog2pfam$PFAMs!="-")

n3_feats2go2cog2pfam <- n3_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n3_feats_bar_sort <- ggplot(data=n3_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n3_feats_barsort_2025 as 4x4
```

16. Neuron 4 RES OF 0.3
```{r}
n4 <- subset(oculina_named_new, idents = "Neuron 4") 

n4 <- FindVariableFeatures(n4)
n4 <- ScaleData(n4, features = features_to_scale)
n4 <- RunPCA(n4)
n4 <- FindNeighbors(n4, dims = 1:30, reduction = "pca")
n4 <- FindClusters(n4, resolution = 0.3)
#1 cluster
n4 <- RunUMAP(n4, reduction="pca", dims=1:30, return.model=TRUE)

n4@meta.data$seurat_clusters <- as.numeric(as.character(n4@meta.data$seurat_clusters))+1
Idents(n4)<- "seurat_clusters"
n4$seurat_clusters
n4@active.ident <- factor(x = n4@active.ident, levels = 1)
n4@active.ident

levels(n4) <- c("1")

plot35 <- DimPlot(n4, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot36 <- DimPlot(n4, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot36b <- DimPlot(n4, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n4_plot35plot36_2025 9x4
#umap_n4_plot36b_2025 8x4 landscape
```

```{r}
n4_topfeat <- TopFeatures(object = n4[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n4[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n4 <- FetchData(n4, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n4 <- FetchData(n4, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n4, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n4, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n4 <- Stdev(n4, reduction = "pca")

pca_n4 <- pc_data_n4
pca_n4$sym_state <- n4@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n4)) #0.3
summary(aov(PC_2 ~ sym_state, data = pca_n4)) #0.0145 *
summary(aov(PC_3 ~ sym_state, data = pca_n4)) #0.183
summary(aov(PC_4 ~ sym_state, data = pca_n4)) #0.000694 ***
summary(aov(PC_5 ~ sym_state, data = pca_n4)) #0.0745 .

summary(aov(PC_6 ~ sym_state, data = pca_n4)) #0.0865 .
summary(aov(PC_7 ~ sym_state, data = pca_n4)) #0.298
summary(aov(PC_8 ~ sym_state, data = pca_n4)) #0.0382 *
summary(aov(PC_9 ~ sym_state, data = pca_n4)) #0.0493 *
summary(aov(PC_10 ~ sym_state, data = pca_n4)) #0.827


DefaultAssay(n4) <- "RNA"
n4[["RNA"]]$counts<-as.matrix(n4[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n4_mark <- FindMarkers(n4, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n4_mark)
#neg lfc means down in sym; pos lfc means up in sym

n4_mark <- rownames_to_column(n4_mark, var="Preferred_name")

#look at subcluster marker genes
#only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of n3
#only one subcluster
```

```{r}
#get genes from signficant PCs (2, 4, 8, 9)
print(n4[["pca"]], dims = 1:10, nfeatures = 10)

n4_feats <- c("g2185", "WDR41.3", "LRRIQ1.1", "ADAMTS2", "PSMD10", "EIF3L", "g11085", "g27965", "g16924", "g7865", "Ly6i", "g4090", "g4317", "ANO8", "g939", "SLC16A2.9", "MCTP2", "g22133", "ABCC4.44", "g28485", "g6517", "CRIP2", "NAA35", "g26049", "g27204", "AHCY", "SIPA1L2", "ARMC4", "SAMD5", "ATP13A2", "g6227", "LDB3", "ENTHD2", "g7212", "g12777", "KDM6A", "MOSPD1", "g10331", "ISL1", "g19841", "RSAD1", "g12147", "STK33", "g10331", "g28731", "g18665", "C17orf85", "ETF1", "g20939", "PRDX6", "LPCAT2.1", "FOXK1", "PDP1.1", "g12863", "HPS5", "RBM25", "SLU7", "PIWIL1", "NOA1", "RACGAP1", "QKI", "g25985", "g26103", "g25919", "MOSPD1", "g8854", "g29652", "g10739", "LRRIQ1.1", "DYSF.2", "g13568", "PREP", "g14904", "SAP30L", "g14943", "g18066", "g4734", "CPEB2", "IQCG", "g7980") 
#for now not removing the unnannotated genes

n4_feats_mark <- subset(n4_mark, Preferred_name %in% n4_feats)
# add a column of NAs
n4_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n4_feats_mark$diffexpressed[n4_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n4_feats_mark$diffexpressed[n4_feats_mark$avg_log2FC < 0] <- "apo"

n4_feats2go <- n4_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n4_feats2go2cog <- n4_feats2go %>%
  left_join(gene2cog)
head(n4_feats2go2cog)
n4_feats2go2cog$COG_category[n4_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n4_feats2go2cog2pfam <- n4_feats2go2cog %>%
  left_join(gene2pfam)
n4_feats2go2cog2pfam <- subset(n4_feats2go2cog2pfam, n4_feats2go2cog2pfam$PFAMs!="-")

n4_feats2go2cog2pfam <- n4_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n4_feats_bar_sort <- ggplot(data=n4_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n4_feats_barsort_2025 as 10x4
```

17. Neuron 5 RES OF 0.5
```{r}
n5 <- subset(oculina_named_new, idents = "Neuron 5") 

n5 <- FindVariableFeatures(n5)
n5 <- ScaleData(n5, features = features_to_scale)
n5 <- RunPCA(n5)
n5 <- FindNeighbors(n5, dims = 1:30, reduction = "pca")
n5 <- FindClusters(n5, resolution = 0.5)
#1 cluster
n5 <- RunUMAP(n5, reduction="pca", dims=1:30, return.model=TRUE)

n5@meta.data$seurat_clusters <- as.numeric(as.character(n5@meta.data$seurat_clusters))+1
Idents(n5)<- "seurat_clusters"
n5$seurat_clusters
n5@active.ident <- factor(x = n5@active.ident, levels = 1)
n5@active.ident

levels(n5) <- c("1")

plot37 <- DimPlot(n5, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot38 <- DimPlot(n5, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot38b <- DimPlot(n5, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n5_plot37plot38_2025 9x4
#umap_n5_plot38b_2025 8x4 landscape
```

```{r}
n5_topfeat <- TopFeatures(object = n5[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n5[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n5 <- FetchData(n5, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n5 <- FetchData(n5, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n5, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n5, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n5 <- Stdev(n5, reduction = "pca")

pca_n5 <- pc_data_n5
pca_n5$sym_state <- n5@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n5)) #0.053 .
summary(aov(PC_2 ~ sym_state, data = pca_n5)) #0.00706 **
summary(aov(PC_3 ~ sym_state, data = pca_n5)) #0.0606 .
summary(aov(PC_4 ~ sym_state, data = pca_n5)) #0.553
summary(aov(PC_5 ~ sym_state, data = pca_n5)) #0.333

summary(aov(PC_6 ~ sym_state, data = pca_n5)) #0.564
summary(aov(PC_7 ~ sym_state, data = pca_n5)) #0.36
summary(aov(PC_8 ~ sym_state, data = pca_n5)) #0.0943 .
summary(aov(PC_9 ~ sym_state, data = pca_n5)) #0.754
summary(aov(PC_10 ~ sym_state, data = pca_n5)) #0.951


DefaultAssay(n5) <- "RNA"
n5[["RNA"]]$counts<-as.matrix(n5[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n5_mark <- FindMarkers(n5, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n5_mark)
#neg lfc means down in sym; pos lfc means up in sym

n5_mark <- rownames_to_column(n5_mark, var="Preferred_name")

#look at subcluster marker genes
#only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of n5
#only one subcluster
```

```{r}
#get genes from signficant PCs (2)
print(n5[["pca"]], dims = 1:10, nfeatures = 10)

n5_feats <- c("g6315", "USH2A", "g18665", "CUBN.16", "g8521", "Mcm7", "ANGPTL1.1", "EDIL3.13", "ZC2HC1C", "g17426", "g11832", "CELA2A", "g25942", "g12921", "g3606", "RPL11", "RPL28", "g21957", "g658", "ABHD11.1") 
#for now not removing the unnannotated genes

n5_feats_mark <- subset(n5_mark, Preferred_name %in% n5_feats)
# add a column of NAs
n5_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n5_feats_mark$diffexpressed[n5_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n5_feats_mark$diffexpressed[n5_feats_mark$avg_log2FC < 0] <- "apo"

n5_feats2go <- n5_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n5_feats2go2cog <- n5_feats2go %>%
  left_join(gene2cog)
head(n5_feats2go2cog)
n5_feats2go2cog$COG_category[n5_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n5_feats2go2cog2pfam <- n5_feats2go2cog %>%
  left_join(gene2pfam)
n5_feats2go2cog2pfam <- subset(n5_feats2go2cog2pfam, n5_feats2go2cog2pfam$PFAMs!="-")

n5_feats2go2cog2pfam <- n5_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n5_feats_bar_sort <- ggplot(data=n5_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n5_feats_barsort_2025 as 4x4
```

18. Neuron 6 RES OF 0.5
```{r}
n6 <- subset(oculina_named_new, idents = "Neuron 6") 

n6 <- FindVariableFeatures(n6)
n6 <- ScaleData(n6, features = features_to_scale)
n6 <- RunPCA(n6)
n6 <- FindNeighbors(n6, dims = 1:30, reduction = "pca")
n6 <- FindClusters(n6, resolution = 0.5)
#1 cluster
n6 <- RunUMAP(n6, reduction="pca", dims=1:30, return.model=TRUE)

n6@meta.data$seurat_clusters <- as.numeric(as.character(n6@meta.data$seurat_clusters))+1
Idents(n6)<- "seurat_clusters"
n6$seurat_clusters
n6@active.ident <- factor(x = n6@active.ident, levels = 1)
n6@active.ident

levels(n6) <- c("1")

plot39 <- DimPlot(n6, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot40 <- DimPlot(n6, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot40b <- DimPlot(n6, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n6_plot39plot40_2025 9x4
#umap_n6_plot40b_2025 8x4 landscape
```

```{r}
n6_topfeat <- TopFeatures(object = n6[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n6[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n6 <- FetchData(n6, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n6 <- FetchData(n6, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n6, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n6, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n6 <- Stdev(n6, reduction = "pca")

pca_n6 <- pc_data_n6
pca_n6$sym_state <- n6@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n6)) #0.658
summary(aov(PC_2 ~ sym_state, data = pca_n6)) #0.00178 **
summary(aov(PC_3 ~ sym_state, data = pca_n6)) #0.00522 **
summary(aov(PC_4 ~ sym_state, data = pca_n6)) #0.595
summary(aov(PC_5 ~ sym_state, data = pca_n6)) #0.303

summary(aov(PC_6 ~ sym_state, data = pca_n6)) #0.000415 ***
summary(aov(PC_7 ~ sym_state, data = pca_n6)) #0.17
summary(aov(PC_8 ~ sym_state, data = pca_n6)) #0.885
summary(aov(PC_9 ~ sym_state, data = pca_n6)) #0.109
summary(aov(PC_10 ~ sym_state, data = pca_n6)) #0.582


DefaultAssay(n6) <- "RNA"
n6[["RNA"]]$counts<-as.matrix(n6[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n6_mark <- FindMarkers(n6, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n6_mark)
#neg lfc means down in sym; pos lfc means up in sym

n6_mark <- rownames_to_column(n6_mark, var="Preferred_name")

#look at subcluster marker genes
#only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of n5
#only one subcluster
```

```{r}
#get genes from signficant PCs (2, 3, 6)
print(n6[["pca"]], dims = 1:10, nfeatures = 10)

n6_feats <- c("g21936", "ALMS1", "ANKMY1", "g19005", "ARID5B", "g15006", "TECPR1", "g17185", "XPO7", "g7950", "BMT2", "CASP14.3", "MORC2", "R3HDM1", "ANXA4", "g14286", "TMCC2", "HERC1", "IBTK", "g2788", "SRA1", "g19436", "g7353", "g8996", "MARCH8", "SLC18A2", "MKNK1", "FBXL13", "PLEKHG4B", "TLK2", "Cyp1", "HNRNPA3", "g24600", "GDPD1", "g21936", "ALMS1", "g19005", "BROX", "ARID5B", "CCDC166.1", "TM9SF2.1", "PUM2", "HMCN1", "SACM1L", "MOXD1.12", "g4255", "ADK", "TRMT44", "g8148", "USP24", "g29670", "ZNFX1.11", "g17544", "SREBF2", "FAM192A", "CUL4A", "g3621", "g8094", "g24943", "HELZ2") 
#for now not removing the unnannotated genes

n6_feats_mark <- subset(n6_mark, Preferred_name %in% n6_feats)
# add a column of NAs
n6_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n6_feats_mark$diffexpressed[n6_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n6_feats_mark$diffexpressed[n6_feats_mark$avg_log2FC < 0] <- "apo"

n6_feats2go <- n6_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n6_feats2go2cog <- n6_feats2go %>%
  left_join(gene2cog)
head(n6_feats2go2cog)
n6_feats2go2cog$COG_category[n6_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n6_feats2go2cog2pfam <- n6_feats2go2cog %>%
  left_join(gene2pfam)
n6_feats2go2cog2pfam <- subset(n6_feats2go2cog2pfam, n6_feats2go2cog2pfam$PFAMs!="-")

n6_feats2go2cog2pfam <- n6_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n6_feats_bar_sort <- ggplot(data=n6_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n6_feats_barsort_2025 as 8x4
```

19. Neuron 7 RES OF 0.5
```{r}
n7 <- subset(oculina_named_new, idents = "Neuron 7") 

n7 <- FindVariableFeatures(n7)
n7 <- ScaleData(n7, features = features_to_scale)
n7 <- RunPCA(n7)
n7 <- FindNeighbors(n7, dims = 1:30, reduction = "pca")
n7 <- FindClusters(n7, resolution = 0.5)
#1 cluster
n7 <- RunUMAP(n7, reduction="pca", dims=1:30, return.model=TRUE)

n7@meta.data$seurat_clusters <- as.numeric(as.character(n7@meta.data$seurat_clusters))+1
Idents(n7)<- "seurat_clusters"
n7$seurat_clusters
n7@active.ident <- factor(x = n7@active.ident, levels = 1)
n7@active.ident

levels(n7) <- c("1")

plot41 <- DimPlot(n7, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot42 <- DimPlot(n7, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot42b <- DimPlot(n7, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n7_plot41plot42_2025 9x4
#umap_n7_plot42b_2025 8x4 landscape
```

```{r}
n7_topfeat <- TopFeatures(object = n7[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n7[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n7 <- FetchData(n7, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n7 <- FetchData(n7, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n7, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n7, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n7 <- Stdev(n7, reduction = "pca")

pca_n7 <- pc_data_n7
pca_n7$sym_state <- n7@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n7)) #0.251
summary(aov(PC_2 ~ sym_state, data = pca_n7)) #0.298
summary(aov(PC_3 ~ sym_state, data = pca_n7)) #0.163
summary(aov(PC_4 ~ sym_state, data = pca_n7)) #0.858
summary(aov(PC_5 ~ sym_state, data = pca_n7)) #0.232

summary(aov(PC_6 ~ sym_state, data = pca_n7)) #0.359
summary(aov(PC_7 ~ sym_state, data = pca_n7)) #0.273
summary(aov(PC_8 ~ sym_state, data = pca_n7)) #0.303
summary(aov(PC_9 ~ sym_state, data = pca_n7)) #0.0468 *
summary(aov(PC_10 ~ sym_state, data = pca_n7)) #0.902


DefaultAssay(n7) <- "RNA"
n7[["RNA"]]$counts<-as.matrix(n7[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n7_mark <- FindMarkers(n7, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n7_mark)
#neg lfc means down in sym; pos lfc means up in sym

n7_mark <- rownames_to_column(n7_mark, var="Preferred_name")

#look at subcluster marker genes
#only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of n5
#only one subcluster
```

```{r}
#get genes from signficant PCs (9)
print(n7[["pca"]], dims = 1:10, nfeatures = 10)

n7_feats <- c("g20696", "ZMAT2", "ADAM33", "SLC29A1", "g20245", "UBAP2", "g3825", "ACTG1.1", "HRAS", "SMIM15", "FPR2.1", "MICU3", "ACTR5", "USP15", "g22976", "TTN", "POMGNT1.1", "g3220", "g24057", "g25145") 
#for now not removing the unnannotated genes

n7_feats_mark <- subset(n7_mark, Preferred_name %in% n7_feats)
# add a column of NAs
n7_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n7_feats_mark$diffexpressed[n7_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n7_feats_mark$diffexpressed[n7_feats_mark$avg_log2FC < 0] <- "apo"

n7_feats2go <- n7_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n7_feats2go2cog <- n7_feats2go %>%
  left_join(gene2cog)
head(n7_feats2go2cog)
n7_feats2go2cog$COG_category[n7_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n7_feats2go2cog2pfam <- n7_feats2go2cog %>%
  left_join(gene2pfam)
n7_feats2go2cog2pfam <- subset(n7_feats2go2cog2pfam, n7_feats2go2cog2pfam$PFAMs!="-")

n7_feats2go2cog2pfam <- n7_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n7_feats_bar_sort <- ggplot(data=n7_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n7_feats_barsort_2025 as 4x4
```

20. Neuron 8 RES OF 0.5
```{r}
n8 <- subset(oculina_named_new, idents = "Neuron 8") 

n8 <- FindVariableFeatures(n8)
n8 <- ScaleData(n8, features = features_to_scale)
n8 <- RunPCA(n8)
n8 <- FindNeighbors(n8, dims = 1:30, reduction = "pca")
n8 <- FindClusters(n8, resolution = 0.5)
#1 cluster
n8 <- RunUMAP(n8, reduction="pca", dims=1:30, return.model=TRUE)

n8@meta.data$seurat_clusters <- as.numeric(as.character(n8@meta.data$seurat_clusters))+1
Idents(n8)<- "seurat_clusters"
n8$seurat_clusters
n8@active.ident <- factor(x = n8@active.ident, levels = 1)
n8@active.ident

levels(n8) <- c("1")

plot43 <- DimPlot(n8, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot44 <- DimPlot(n8, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot44b <- DimPlot(n8, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n8_plot43plot44_2025 9x4
#umap_n8_plot44b_2025 8x4 landscape
```

```{r}
n8_topfeat <- TopFeatures(object = n8[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n8[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n8 <- FetchData(n8, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n8 <- FetchData(n8, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n8, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n8, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n8 <- Stdev(n8, reduction = "pca")

pca_n8 <- pc_data_n8
pca_n8$sym_state <- n8@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n8)) #0.0993 .
summary(aov(PC_2 ~ sym_state, data = pca_n8)) #0.685
summary(aov(PC_3 ~ sym_state, data = pca_n8)) #0.0692 .
summary(aov(PC_4 ~ sym_state, data = pca_n8)) #0.0199 *
summary(aov(PC_5 ~ sym_state, data = pca_n8)) #0.408

summary(aov(PC_6 ~ sym_state, data = pca_n8)) #0.351
summary(aov(PC_7 ~ sym_state, data = pca_n8)) #0.731
summary(aov(PC_8 ~ sym_state, data = pca_n8)) #0.729
summary(aov(PC_9 ~ sym_state, data = pca_n8)) #0.561
summary(aov(PC_10 ~ sym_state, data = pca_n8)) #0.912


DefaultAssay(n8) <- "RNA"
n8[["RNA"]]$counts<-as.matrix(n8[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n8_mark <- FindMarkers(n8, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n8_mark)
#neg lfc means down in sym; pos lfc means up in sym

n8_mark <- rownames_to_column(n8_mark, var="Preferred_name")

#look at subcluster marker genes
#only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of n5
#only one subcluster
```

```{r}
#get genes from signficant PCs (4)
print(n8[["pca"]], dims = 1:10, nfeatures = 10)

n8_feats <- c("LOXHD1", "RRAS2", "g19006", "TSEN2", "LYRM4", "g13375", "g17654", "SAMD9L.1", "g4740", "g30138", "LIG3", "g14720", "g1471", "FRMD5", "Pld", "EXOC4", "DYSF.2", "g11207", "g4939", "ABHD5.1") 
#for now not removing the unnannotated genes

n8_feats_mark <- subset(n8_mark, Preferred_name %in% n8_feats)
# add a column of NAs
n8_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n8_feats_mark$diffexpressed[n8_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n8_feats_mark$diffexpressed[n8_feats_mark$avg_log2FC < 0] <- "apo"

n8_feats2go <- n8_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n8_feats2go2cog <- n8_feats2go %>%
  left_join(gene2cog)
head(n8_feats2go2cog)
n8_feats2go2cog$COG_category[n8_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n8_feats2go2cog2pfam <- n8_feats2go2cog %>%
  left_join(gene2pfam)
n8_feats2go2cog2pfam <- subset(n8_feats2go2cog2pfam, n8_feats2go2cog2pfam$PFAMs!="-")

n8_feats2go2cog2pfam <- n8_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n8_feats_bar_sort <- ggplot(data=n8_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n8_feats_barsort_2025 as 4x4
```

21. Neuron 9 RES OF 0.5
```{r}
n9 <- subset(oculina_named_new, idents = "Neuron 9") 

n9 <- FindVariableFeatures(n9)
n9 <- ScaleData(n9, features = features_to_scale)
n9 <- RunPCA(n9, npcs = 20) #because fewer than 50 cells (fewer than 30 cells even)
n9 <- FindNeighbors(n9, dims = 1:20, reduction = "pca")
n9 <- FindClusters(n9, resolution = 0.5)
#1 cluster
n9 <- RunUMAP(n9, reduction="pca", dims=1:20, return.model=TRUE, n.neighbors = 20)

n9@meta.data$seurat_clusters <- as.numeric(as.character(n9@meta.data$seurat_clusters))+1
Idents(n9)<- "seurat_clusters"
n9$seurat_clusters
n9@active.ident <- factor(x = n9@active.ident, levels = 1)
n9@active.ident

levels(n9) <- c("1")

plot45 <- DimPlot(n9, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot46 <- DimPlot(n9, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot46b <- DimPlot(n9, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n6_plot45plot46_2025 9x4
#umap_n9_plot46b_2025 8x4 landscape
```

```{r}
n9_topfeat <- TopFeatures(object = n9[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n9[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n9 <- FetchData(n9, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n9 <- FetchData(n9, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n9, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n9, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n9 <- Stdev(n9, reduction = "pca")

pca_n9 <- pc_data_n9
pca_n9$sym_state <- n9@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n9)) #0.0996 .
summary(aov(PC_2 ~ sym_state, data = pca_n9)) #0.547
summary(aov(PC_3 ~ sym_state, data = pca_n9)) #0.905
summary(aov(PC_4 ~ sym_state, data = pca_n9)) #0.89
summary(aov(PC_5 ~ sym_state, data = pca_n9)) #0.901

summary(aov(PC_6 ~ sym_state, data = pca_n9)) #0.717
summary(aov(PC_7 ~ sym_state, data = pca_n9)) #0.612
summary(aov(PC_8 ~ sym_state, data = pca_n9)) #0.0211 *
summary(aov(PC_9 ~ sym_state, data = pca_n9)) #0.0382 *
summary(aov(PC_10 ~ sym_state, data = pca_n9)) #0.285


DefaultAssay(n9) <- "RNA"
n9[["RNA"]]$counts<-as.matrix(n9[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n9_mark <- FindMarkers(n9, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n9_mark)
#neg lfc means down in sym; pos lfc means up in sym

n9_mark <- rownames_to_column(n9_mark, var="Preferred_name")

#look at subcluster marker genes
#only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of n5
#only one subcluster
```

```{r}
#get genes from signficant PCs (8, 9)
print(n9[["pca"]], dims = 1:10, nfeatures = 10)

n9_feats <- c("g3709", "FBXO11", "UNC119", "F14E5.2", "ZNFX1.6", "RPL27", "REPS1", "RPS10", "SREBF2", "g29679", "NUG1", "RPL15", "SMAP1", "MYO3B", "g23155", "DTNA", "g16687", "PI4KA", "PWP2", "g5666", "ATF6B", "MYO3B", "g29023", "NAGPA", "CD59.2", "DYSF.2", "znf711", "EIF4H", "g30337", "IWS1", "TMA7", "HINT1", "BAZ1A", "g19484", "g18493", "MTIF2", "g8955", "NSA2", "PRRC2C", "RPL27") 
#for now not removing the unnannotated genes

n9_feats_mark <- subset(n9_mark, Preferred_name %in% n9_feats)
# add a column of NAs
n9_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n9_feats_mark$diffexpressed[n9_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n9_feats_mark$diffexpressed[n9_feats_mark$avg_log2FC < 0] <- "apo"

n9_feats2go <- n9_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n9_feats2go2cog <- n9_feats2go %>%
  left_join(gene2cog)
head(n9_feats2go2cog)
n9_feats2go2cog$COG_category[n9_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n9_feats2go2cog2pfam <- n9_feats2go2cog %>%
  left_join(gene2pfam)
n9_feats2go2cog2pfam <- subset(n9_feats2go2cog2pfam, n9_feats2go2cog2pfam$PFAMs!="-")

n9_feats2go2cog2pfam <- n9_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n9_feats_bar_sort <- ggplot(data=n9_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n9_feats_barsort_2025 as 4x4
```

22. Neuron 10 RES OF 0.5
```{r}
n10 <- subset(oculina_named_new, idents = "Neuron 10") 

n10 <- FindVariableFeatures(n10)
n10 <- ScaleData(n10, features = features_to_scale)
n10 <- RunPCA(n10, npcs = 20) #because fewer than 50 cells (fewer than 30 cells even)
n10 <- FindNeighbors(n10, dims = 1:20, reduction = "pca")
n10 <- FindClusters(n10, resolution = 0.5)
#1 cluster
n10 <- RunUMAP(n10, reduction="pca", dims=1:20, return.model=TRUE, n.neighbors = 20)

n10@meta.data$seurat_clusters <- as.numeric(as.character(n10@meta.data$seurat_clusters))+1
Idents(n10)<- "seurat_clusters"
n10$seurat_clusters
n10@active.ident <- factor(x = n10@active.ident, levels = 1)
n10@active.ident

levels(n10) <- c("1")

plot47 <- DimPlot(n10, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot48 <- DimPlot(n10, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot48b <- DimPlot(n10, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_n10_plot47plot48_2025 9x4
#umap_n10_plot48b_2025 8x4 landscape
```

```{r}
n10_topfeat <- TopFeatures(object = n10[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
n10[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_n10 <- FetchData(n10, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_n10 <- FetchData(n10, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_n10, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_n10, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_n10 <- Stdev(n10, reduction = "pca")

pca_n10 <- pc_data_n10
pca_n10$sym_state <- n10@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_n10)) #0.000295 ***
summary(aov(PC_2 ~ sym_state, data = pca_n10)) #0.835
summary(aov(PC_3 ~ sym_state, data = pca_n10)) #0.174
summary(aov(PC_4 ~ sym_state, data = pca_n10)) #0.0852 .
summary(aov(PC_5 ~ sym_state, data = pca_n10)) #0.76

summary(aov(PC_6 ~ sym_state, data = pca_n10)) #0.0894 .
summary(aov(PC_7 ~ sym_state, data = pca_n10)) #0.458
summary(aov(PC_8 ~ sym_state, data = pca_n10)) #0.802
summary(aov(PC_9 ~ sym_state, data = pca_n10)) #0.906
summary(aov(PC_10 ~ sym_state, data = pca_n10)) #0.603


DefaultAssay(n10) <- "RNA"
n10[["RNA"]]$counts<-as.matrix(n10[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
n10_mark <- FindMarkers(n10, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(n10_mark)
#neg lfc means down in sym; pos lfc means up in sym

n10_mark <- rownames_to_column(n10_mark, var="Preferred_name")

#look at subcluster marker genes
#only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of n10
#only one subcluster
```

```{r}
#get genes from signficant PCs (1)
print(n10[["pca"]], dims = 1:10, nfeatures = 10)

n10_feats <- c("g10424", "g5731", "g18630", "g22336", "g25372", "g19000", "g18999", "ZNF333", "g7832", "g4091", "RPS13", "RPS26", "RPLP1", "RPS28", "RPL13", "RPL24", "RPSA", "eef1g", "rps17", "TMEM258") 
#for now not removing the unnannotated genes

n10_feats_mark <- subset(n10_mark, Preferred_name %in% n10_feats)
# add a column of NAs
n10_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
n10_feats_mark$diffexpressed[n10_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
n10_feats_mark$diffexpressed[n10_feats_mark$avg_log2FC < 0] <- "apo"

n10_feats2go <- n10_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

n10_feats2go2cog <- n10_feats2go %>%
  left_join(gene2cog)
head(n10_feats2go2cog)
n10_feats2go2cog$COG_category[n10_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
n10_feats2go2cog2pfam <- n10_feats2go2cog %>%
  left_join(gene2pfam)
n10_feats2go2cog2pfam <- subset(n10_feats2go2cog2pfam, n10_feats2go2cog2pfam$PFAMs!="-")

n10_feats2go2cog2pfam <- n10_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

n10_feats_bar_sort <- ggplot(data=n10_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### n10_feats_barsort_2025 as 4x4
```

23. Gland 1 RES OF 0.3
```{r}
gland1 <- subset(oculina_named_new, idents = "Gland 1") 

gland1 <- FindVariableFeatures(gland1)
gland1 <- ScaleData(gland1, features = features_to_scale)
gland1 <- RunPCA(gland1)
gland1 <- FindNeighbors(gland1, dims = 1:30, reduction = "pca")
gland1 <- FindClusters(gland1, resolution = 0.3)
#2 clusters
gland1 <- RunUMAP(gland1, reduction="pca", dims=1:30, return.model=TRUE)

gland1@meta.data$seurat_clusters <- as.numeric(as.character(gland1@meta.data$seurat_clusters))+1
Idents(gland1)<- "seurat_clusters"
gland1$seurat_clusters
gland1@active.ident <- factor(x = gland1@active.ident, levels = 1:2)
gland1@active.ident

levels(gland1) <- c("1", "2")

plot49 <- DimPlot(gland1, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 cluster

plot50 <- DimPlot(gland1, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot50b <- DimPlot(gland1, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gland1_plot49plot50_2025 9x4
#umap_gland1_plot50b_2025 8x4 landscape
```

```{r}
gland1_topfeat <- TopFeatures(object = gland1[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gland1[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gland1 <- FetchData(gland1, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gland1 <- FetchData(gland1, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gland1, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gland1, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_gland1 <- Stdev(gland1, reduction = "pca")

pca_gland1 <- pc_data_gland1
pca_gland1$sym_state <- gland1@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gland1)) #0.65
summary(aov(PC_2 ~ sym_state, data = pca_gland1)) #0.636
summary(aov(PC_3 ~ sym_state, data = pca_gland1)) #0.541
summary(aov(PC_4 ~ sym_state, data = pca_gland1)) #0.276
summary(aov(PC_5 ~ sym_state, data = pca_gland1)) #0.845

summary(aov(PC_6 ~ sym_state, data = pca_gland1)) #0.0625 .
summary(aov(PC_7 ~ sym_state, data = pca_gland1)) #0.433
summary(aov(PC_8 ~ sym_state, data = pca_gland1)) #0.132
summary(aov(PC_9 ~ sym_state, data = pca_gland1)) #0.0482 *
summary(aov(PC_10 ~ sym_state, data = pca_gland1)) #0.41


DefaultAssay(gland1) <- "RNA"
gland1[["RNA"]]$counts<-as.matrix(gland1[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gland1_mark <- FindMarkers(gland1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gland1_mark)
#neg lfc means down in sym; pos lfc means up in sym

gland1_mark <- rownames_to_column(gland1_mark, var="Preferred_name")

#look at subcluster marker genes
gland1_markers_subclust <- FindAllMarkers(gland1, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

gland1_markers_subclust
gland1_markers_subclust <- rownames_to_column(gland1_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of gland1
gland1_1 <- subset(gland1, idents="1")
DefaultAssay(gland1_1) <- "RNA"

gland1_1_mark <- FindMarkers(gland1_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gland1_1_mark <- rownames_to_column(gland1_1_mark, var="Preferred_name")
head(gland1_1_mark)

#get apo vs sym differences for subcluster 2 of gland1
gland1_2 <- subset(gland1, idents="2")
DefaultAssay(gland1_2) <- "RNA"

gland1_2_mark <- FindMarkers(gland1_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gland1_2_mark <- rownames_to_column(gland1_2_mark, var="Preferred_name")
head(gland1_2_mark)
```

```{r}
#get genes from signficant PCs (9)
print(gland1[["pca"]], dims = 1:10, nfeatures = 10)

gland1_feats <- c("g21611", "VPS26B", "g13981", "YTHDC2", "TBC1D12", "ALDH4A1", "ALDH2.1", "TBC1D4", "TADA2B", "CNOT10", "STK32B", "RANBP1", "g25222", "g10249", "PELO", "NUDCD1", "STPG2", "TRAPPC5", "Picot", "g25219") 
#for now not removing the unnannotated genes

gland1_feats_mark <- subset(gland1_mark, Preferred_name %in% gland1_feats)
# add a column of NAs
gland1_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
gland1_feats_mark$diffexpressed[gland1_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
gland1_feats_mark$diffexpressed[gland1_feats_mark$avg_log2FC < 0] <- "apo"

gland1_feats2go <- gland1_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

gland1_feats2go2cog <- gland1_feats2go %>%
  left_join(gene2cog)
head(gland1_feats2go2cog)
gland1_feats2go2cog$COG_category[gland1_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
gland1_feats2go2cog2pfam <- gland1_feats2go2cog %>%
  left_join(gene2pfam)
gland1_feats2go2cog2pfam <- subset(gland1_feats2go2cog2pfam, gland1_feats2go2cog2pfam$PFAMs!="-")

gland1_feats2go2cog2pfam <- gland1_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

gland1_feats_bar_sort <- ggplot(data=gland1_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### gland1_feats_barsort_2025 as 4x4
```

24. Gland 2 RES OF 0.3
```{r}
gland2 <- subset(oculina_named_new, idents = "Gland 2") 

gland2 <- FindVariableFeatures(gland2)
gland2 <- ScaleData(gland2, features = features_to_scale)
gland2 <- RunPCA(gland2)
gland2 <- FindNeighbors(gland2, dims = 1:30, reduction = "pca")
gland2 <- FindClusters(gland2, resolution = 0.3)
#1 cluster
gland2 <- RunUMAP(gland2, reduction="pca", dims=1:30, return.model=TRUE)

gland2@meta.data$seurat_clusters <- as.numeric(as.character(gland2@meta.data$seurat_clusters))+1
Idents(gland2)<- "seurat_clusters"
gland2$seurat_clusters
gland2@active.ident <- factor(x = gland2@active.ident, levels = 1)
gland2@active.ident

levels(gland2) <- c("1")

plot51 <- DimPlot(gland2, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot52 <- DimPlot(gland2, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot52b <- DimPlot(gland2, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gland2_plot51plot52_2025 9x4
#umap_gland2_plot52b_2025 8x4 landscape
```

```{r}
gland2_topfeat <- TopFeatures(object = gland2[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gland2[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gland2 <- FetchData(gland2, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gland2 <- FetchData(gland2, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gland2, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gland2, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_gland2 <- Stdev(gland2, reduction = "pca")

pca_gland2 <- pc_data_gland2
pca_gland2$sym_state <- gland2@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gland2)) #0.591
summary(aov(PC_2 ~ sym_state, data = pca_gland2)) #0.417
summary(aov(PC_3 ~ sym_state, data = pca_gland2)) #0.0582 .
summary(aov(PC_4 ~ sym_state, data = pca_gland2)) #0.246
summary(aov(PC_5 ~ sym_state, data = pca_gland2)) #0.112

summary(aov(PC_6 ~ sym_state, data = pca_gland2)) #0.141 .
summary(aov(PC_7 ~ sym_state, data = pca_gland2)) #0.423
summary(aov(PC_8 ~ sym_state, data = pca_gland2)) #0.0675 .
summary(aov(PC_9 ~ sym_state, data = pca_gland2)) #0.671
summary(aov(PC_10 ~ sym_state, data = pca_gland2)) #0.782


DefaultAssay(gland2) <- "RNA"
gland2[["RNA"]]$counts<-as.matrix(gland2[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gland2_mark <- FindMarkers(gland2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gland2_mark)
#neg lfc means down in sym; pos lfc means up in sym

gland2_mark <- rownames_to_column(gland2_mark, var="Preferred_name")

#look at subcluster marker genes
#Only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of gland1
#Only 1 subcluster
```

```{r}
#no significant PCs
```

25. Gland 3 RES OF 0.5
```{r}
gland3 <- subset(oculina_named_new, idents = "Gland 3") 

gland3 <- FindVariableFeatures(gland3)
gland3 <- ScaleData(gland3, features = features_to_scale)
gland3 <- RunPCA(gland3, npcs = 30)
gland3 <- FindNeighbors(gland3, dims = 1:30, reduction = "pca")
gland3 <- FindClusters(gland3, resolution = 0.5)
#1 cluster
gland3 <- RunUMAP(gland3, reduction="pca", dims=1:30, return.model=TRUE)

gland3@meta.data$seurat_clusters <- as.numeric(as.character(gland3@meta.data$seurat_clusters))+1
Idents(gland3)<- "seurat_clusters"
gland3$seurat_clusters
gland3@active.ident <- factor(x = gland3@active.ident, levels = 1)
gland3@active.ident

levels(gland3) <- c("1")

plot53 <- DimPlot(gland3, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot54 <- DimPlot(gland3, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot54b <- DimPlot(gland3, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_gland3_plot53plot54_2025 9x4
#umap_gland3_plot54b_2025 8x4 landscape
```

```{r}
gland3_topfeat <- TopFeatures(object = gland3[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
gland3[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_gland3 <- FetchData(gland3, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_gland3 <- FetchData(gland3, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_gland3, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_gland3, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_gland3 <- Stdev(gland3, reduction = "pca")

pca_gland3 <- pc_data_gland3
pca_gland3$sym_state <- gland3@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_gland3)) #0.0578 .
summary(aov(PC_2 ~ sym_state, data = pca_gland3)) #0.0564 .
summary(aov(PC_3 ~ sym_state, data = pca_gland3)) #0.838
summary(aov(PC_4 ~ sym_state, data = pca_gland3)) #0.493
summary(aov(PC_5 ~ sym_state, data = pca_gland3)) #0.086 .

summary(aov(PC_6 ~ sym_state, data = pca_gland3)) #0.482
summary(aov(PC_7 ~ sym_state, data = pca_gland3)) #0.193
summary(aov(PC_8 ~ sym_state, data = pca_gland3)) #0.316
summary(aov(PC_9 ~ sym_state, data = pca_gland3)) #0.589
summary(aov(PC_10 ~ sym_state, data = pca_gland3)) #0.829


DefaultAssay(gland3) <- "RNA"
gland3[["RNA"]]$counts<-as.matrix(gland3[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
gland3_mark <- FindMarkers(gland3, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(gland3_mark)
#neg lfc means down in sym; pos lfc means up in sym

gland3_mark <- rownames_to_column(gland3_mark, var="Preferred_name")

#look at subcluster marker genes
#Only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of gland1
#Only 1 subcluster
```

```{r}
#no significant PCs
```

26. Calicoblast RES OF 0.5
```{r}
calico <- subset(oculina_named_new, idents = "Calicoblast") 

calico <- FindVariableFeatures(calico)
calico <- ScaleData(calico, features = features_to_scale)
calico <- RunPCA(calico, npcs = 30)
calico <- FindNeighbors(calico, dims = 1:30, reduction = "pca")
calico <- FindClusters(calico, resolution = 0.5)
#1 cluster
calico <- RunUMAP(calico, reduction="pca", dims=1:30, return.model=TRUE)

calico@meta.data$seurat_clusters <- as.numeric(as.character(calico@meta.data$seurat_clusters))+1
Idents(calico)<- "seurat_clusters"
calico$seurat_clusters
calico@active.ident <- factor(x = calico@active.ident, levels = 1)
calico@active.ident

levels(calico) <- c("1")

plot55 <- DimPlot(calico, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot56 <- DimPlot(calico, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot56b <- DimPlot(calico, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_calico_plot55plot56_2025 9x4
#umap_calico_plot56b_2025 8x4 landscape
```

```{r}
calico_topfeat <- TopFeatures(object = calico[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
calico[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_calico <- FetchData(calico, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_calico <- FetchData(calico, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_calico, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_calico, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_calico <- Stdev(calico, reduction = "pca")

pca_calico <- pc_data_calico
pca_calico$sym_state <- calico@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_calico)) #0.125
summary(aov(PC_2 ~ sym_state, data = pca_calico)) #0.793
summary(aov(PC_3 ~ sym_state, data = pca_calico)) #0.796
summary(aov(PC_4 ~ sym_state, data = pca_calico)) #0.312
summary(aov(PC_5 ~ sym_state, data = pca_calico)) #0.158

summary(aov(PC_6 ~ sym_state, data = pca_calico)) #0.986
summary(aov(PC_7 ~ sym_state, data = pca_calico)) #0.000761 ***
summary(aov(PC_8 ~ sym_state, data = pca_calico)) #0.289
summary(aov(PC_9 ~ sym_state, data = pca_calico)) #0.264
summary(aov(PC_10 ~ sym_state, data = pca_calico)) #0.0498 *


DefaultAssay(calico) <- "RNA"
calico[["RNA"]]$counts<-as.matrix(calico[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
calico_mark <- FindMarkers(calico, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(calico_mark)
#neg lfc means down in sym; pos lfc means up in sym

calico_mark <- rownames_to_column(calico_mark, var="Preferred_name")

#look at subcluster marker genes
#Only 1 subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of calico
#Only 1 subcluster
```

```{r}
#get genes from signficant PCs (7, 10)
print(calico[["pca"]], dims = 1:10, nfeatures = 10)

calico_feats <- c("g28066", "RRP15", "RPL11", "NSA2", "EIF4A2", "g17564", "g9535", "RPS13", "FAM192A", "g21546", "NME8", "g19640", "HMCN1.26", "DCTN4", "IMP4", "ARAP1.7", "TRIB1", "TTC12", "g9984", "SNRPC", "g8061", "ARL3", "g12284", "PLCG2", "g25599", "g13668", "g22451", "ELF2", "g546", "ALOXE3.2", "GPATCH2", "DYNC1LI1", "g423", "g19284", "UFC1", "RpS27A", "SERBP1", "PNLDC1", "ADAT2", "LRSAM1") 
#for now not removing the unnannotated genes

calico_feats_mark <- subset(calico_mark, Preferred_name %in% calico_feats)
# add a column of NAs
calico_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
calico_feats_mark$diffexpressed[calico_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
calico_feats_mark$diffexpressed[calico_feats_mark$avg_log2FC < 0] <- "apo"

calico_feats2go <- calico_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

calico_feats2go2cog <- calico_feats2go %>%
  left_join(gene2cog)
head(calico_feats2go2cog)
calico_feats2go2cog$COG_category[calico_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
calico_feats2go2cog2pfam <- calico_feats2go2cog %>%
  left_join(gene2pfam)
calico_feats2go2cog2pfam <- subset(calico_feats2go2cog2pfam, calico_feats2go2cog2pfam$PFAMs!="-")

calico_feats2go2cog2pfam <- calico_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

calico_feats_bar_sort <- ggplot(data=calico_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### calico_feats_barsort_2025 as 6x4
```

27. Cnidocyte 1 RES OF 0.5
```{r}
cnido1 <- subset(oculina_named_new, idents = "Cnidocyte 1") 

cnido1 <- FindVariableFeatures(cnido1)
cnido1 <- ScaleData(cnido1, features = features_to_scale)
cnido1 <- RunPCA(cnido1)
cnido1 <- FindNeighbors(cnido1, dims = 1:30, reduction = "pca")
cnido1 <- FindClusters(cnido1, resolution = 0.5)
#2 cluster
cnido1 <- RunUMAP(cnido1, reduction="pca", dims=1:30, return.model=TRUE)

cnido1@meta.data$seurat_clusters <- as.numeric(as.character(cnido1@meta.data$seurat_clusters))+1
Idents(cnido1)<- "seurat_clusters"
cnido1$seurat_clusters
cnido1@active.ident <- factor(x = cnido1@active.ident, levels = 1:2)
cnido1@active.ident

levels(cnido1) <- c("1", "2")

plot57 <- DimPlot(cnido1, reduction = "umap", label = T, cols=c("#CC6699", "#33CC99"), alpha=0.5)
#2 clusters

plot58 <- DimPlot(cnido1, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot58b <- DimPlot(cnido1, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699", "#33CC99"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_cnido_plot57plot58_2025 9x4
#umap_cnido_plot58b_2025 8x4 landscape
```

```{r}
cnido1_topfeat <- TopFeatures(object = cnido1[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
cnido1[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_cnido1 <- FetchData(cnido1, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_cnido1 <- FetchData(cnido1, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_cnido1, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_cnido1, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_cnido1 <- Stdev(cnido1, reduction = "pca")

pca_cnido1 <- pc_data_cnido1
pca_cnido1$sym_state <- cnido1@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_cnido1)) #0.105
summary(aov(PC_2 ~ sym_state, data = pca_cnido1)) #0.000608 ***
summary(aov(PC_3 ~ sym_state, data = pca_cnido1)) #0.051 .
summary(aov(PC_4 ~ sym_state, data = pca_cnido1)) #0.128
summary(aov(PC_5 ~ sym_state, data = pca_cnido1)) #0.98

summary(aov(PC_6 ~ sym_state, data = pca_cnido1)) #0.0372 *
summary(aov(PC_7 ~ sym_state, data = pca_cnido1)) #0.47
summary(aov(PC_8 ~ sym_state, data = pca_cnido1)) #0.81
summary(aov(PC_9 ~ sym_state, data = pca_cnido1)) #0.000474 ***
summary(aov(PC_10 ~ sym_state, data = pca_cnido1)) #0.709


DefaultAssay(cnido1) <- "RNA"
cnido1[["RNA"]]$counts<-as.matrix(cnido1[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
cnido1_mark <- FindMarkers(cnido1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(cnido1_mark)
#neg lfc means down in sym; pos lfc means up in sym

cnido1_mark <- rownames_to_column(cnido1_mark, var="Preferred_name")

#look at subcluster marker genes
cnido1_markers_subclust <- FindAllMarkers(cnido1, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

cnido1_markers_subclust
cnido1_markers_subclust <- rownames_to_column(cnido1_markers_subclust, var="Preferred_name")
```

```{r}
#get apo vs sym differences for subcluster 1 of cnido1
cnido1_1 <- subset(cnido1, idents="1")
DefaultAssay(cnido1_1) <- "RNA"

cnido1_1_mark <- FindMarkers(cnido1_1, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
cnido1_1_mark <- rownames_to_column(cnido1_1_mark, var="Preferred_name")
head(cnido1_1_mark)

#get apo vs sym differences for subcluster 2 of cnido1
cnido1_2 <- subset(cnido1, idents="2")
DefaultAssay(cnido1_2) <- "RNA"

cnido1_2_mark <- FindMarkers(cnido1_2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
cnido1_2_mark <- rownames_to_column(cnido1_2_mark, var="Preferred_name")
head(cnido1_2_mark)
```

```{r}
#get genes from signficant PCs (2, 6, 9)
print(cnido1[["pca"]], dims = 1:10, nfeatures = 10)

cnido1_feats <- c("C7orf57.1", "LRRC48", "g24668", "g15344", "g23953", "PDCD2", "PEPD.2", "DLC1", "ZNFX1.1", "hmcn2.5", "awd.1", "CYB5A", "NHP2L1", "g9833", "PSMB2", "g25991", "g14100", "tomm7", "g11285", "CCT6A", "g18994", "PFN4", "g18152", "g20176", "SLC35A4", "g9061", "g24460", "SREK1IP1", "g1726", "g14006", "g5543", "g11424", "MFI2.4", "g22782", "g10149", "LIG4", "APOPT1", "g28399", "OSBPL8", "g13754", "TNPO1", "g6050", "TRAF3.3", "SH3PXD2B.1", "NOXO1", "S6kII", "g27090", "BAG4", "g19330", "g27230", "g727", "g16178", "ELK3.3", "TUBG2", "DGKE", "ZNFX1.1", "g15945", "g582", "g24244", "MIB2.2") 
#for now not removing the unnannotated genes

cnido1_feats_mark <- subset(cnido1_mark, Preferred_name %in% cnido1_feats)
# add a column of NAs
cnido1_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
cnido1_feats_mark$diffexpressed[cnido1_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
cnido1_feats_mark$diffexpressed[cnido1_feats_mark$avg_log2FC < 0] <- "apo"

cnido1_feats2go <- cnido1_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

cnido1_feats2go2cog <- cnido1_feats2go %>%
  left_join(gene2cog)
head(cnido1_feats2go2cog)
cnido1_feats2go2cog$COG_category[cnido1_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
cnido1_feats2go2cog2pfam <- cnido1_feats2go2cog %>%
  left_join(gene2pfam)
cnido1_feats2go2cog2pfam <- subset(cnido1_feats2go2cog2pfam, cnido1_feats2go2cog2pfam$PFAMs!="-")

cnido1_feats2go2cog2pfam <- cnido1_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

cnido1_feats_bar_sort <- ggplot(data=cnido1_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### cnido1_feats_barsort_2025 as 8x4
```

28. Cnidocyte 2 RES OF 0.5
```{r}
cnido2 <- subset(oculina_named_new, idents = "Cnidocyte 2") 

cnido2 <- FindVariableFeatures(cnido2)
cnido2 <- ScaleData(cnido2, features = features_to_scale)
cnido2 <- RunPCA(cnido2)
cnido2 <- FindNeighbors(cnido2, dims = 1:30, reduction = "pca")
cnido2 <- FindClusters(cnido2, resolution = 0.5)
#1 cluster
cnido2 <- RunUMAP(cnido2, reduction="pca", dims=1:30, return.model=TRUE)

cnido2@meta.data$seurat_clusters <- as.numeric(as.character(cnido2@meta.data$seurat_clusters))+1
Idents(cnido2)<- "seurat_clusters"
cnido2$seurat_clusters
cnido2@active.ident <- factor(x = cnido2@active.ident, levels = 1)
cnido2@active.ident

levels(cnido2) <- c("1")

plot59 <- DimPlot(cnido2, reduction = "umap", label = T, cols=c("#CC6699"), alpha=0.5)
#1 cluster

plot60 <- DimPlot(cnido2, reduction = "umap", label=F, order=c("sym", "apo"), group.by = "symstate", cols=c("gray40", "#CC6600"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

plot60b <- DimPlot(cnido2, reduction = "umap", label=T, split.by = "symstate", cols=c("#CC6699"), alpha=0.5) + ggtitle("") + theme(legend.text=element_text(size=10))

#saved as umap_cnido2_plot59plot60_2025 9x4
#umap_cnido2_plot60b_2025 8x4 landscape
```

```{r}
cnido2_topfeat <- TopFeatures(object = cnido2[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
cnido2[["umap"]]@cell.embeddings
##https://hbctraining.github.io/scRNA-seq_online/lessons/08_SC_clustering_quality_control.html
columns <- c(paste0("PC_", 1:10),
            "ident",
            "umap_1", "umap_2")
pc_data_cnido2 <- FetchData(cnido2, 
                     vars = columns)
# Adding cluster label to center of cluster on UMAP
umap_label_cnido2 <- FetchData(cnido2, 
  vars = c("ident", "umap_1", "umap_2"))  %>%
  group_by(ident) %>%
  summarise(x=mean(umap_1), y=mean(umap_2))
# Plotting a UMAP plot for each of the PCs

map(paste0("PC_", 1:10), function(pc){
        ggplot(pc_data_cnido2, 
               aes(umap_1, umap_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label_cnido2, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %>% 
        plot_grid(plotlist = .)

stdev_cnido2 <- Stdev(cnido2, reduction = "pca")

pca_cnido2 <- pc_data_cnido2
pca_cnido2$sym_state <- cnido2@meta.data$symstate
summary(aov(PC_1 ~ sym_state, data = pca_cnido2)) #0.43
summary(aov(PC_2 ~ sym_state, data = pca_cnido2)) #0.183
summary(aov(PC_3 ~ sym_state, data = pca_cnido2)) #0.714
summary(aov(PC_4 ~ sym_state, data = pca_cnido2)) #0.0165 *
summary(aov(PC_5 ~ sym_state, data = pca_cnido2)) #0.366

summary(aov(PC_6 ~ sym_state, data = pca_cnido2)) #0.111
summary(aov(PC_7 ~ sym_state, data = pca_cnido2)) #0.247
summary(aov(PC_8 ~ sym_state, data = pca_cnido2)) #0.876
summary(aov(PC_9 ~ sym_state, data = pca_cnido2)) #0.229
summary(aov(PC_10 ~ sym_state, data = pca_cnido2)) #0.247


DefaultAssay(cnido2) <- "RNA"
cnido2[["RNA"]]$counts<-as.matrix(cnido2[["RNA"]]$counts)+1

#this to look at apo vs sym differentially expressed genes
cnido2_mark <- FindMarkers(cnido2, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
head(cnido2_mark)
#neg lfc means down in sym; pos lfc means up in sym

cnido2_mark <- rownames_to_column(cnido2_mark, var="Preferred_name")

#look at subcluster marker genes
#only one subcluster
```

```{r}
#get apo vs sym differences for subcluster 1 of cnido2
#only one subcluster
```

```{r}
#get genes from signficant PCs (4)
print(cnido2[["pca"]], dims = 1:10, nfeatures = 10)

cnido2_feats <- c("g7715", "PSEN1", "htl", "g26349", "g11635", "CLSTN2", "SH3PXD2B.1", "gcy.11", "NOTCH1", "g29970", "g14732", "g21664", "g10023", "g19821", "g8194", "g4673", "RAB6B", "IRF1.2", "DDR1.1", "TRAPPC9") 
#for now not removing the unnannotated genes

cnido2_feats_mark <- subset(cnido2_mark, Preferred_name %in% cnido2_feats)
# add a column of NAs
cnido2_feats_mark$diffexpressed <- "ns"
# if log2Foldchange > 0 set as "SYM"
cnido2_feats_mark$diffexpressed[cnido2_feats_mark$avg_log2FC > 0] <- "sym"
# if log2Foldchange < 0 set as "APO"
cnido2_feats_mark$diffexpressed[cnido2_feats_mark$avg_log2FC < 0] <- "apo"

cnido2_feats2go <- cnido2_feats_mark %>%
  left_join(gene2go)
#write.table(agh_feats2go, file="./files/epi1_feats2go2025.txt", quote=F, sep="\t", row.names=FALSE)

cnido2_feats2go2cog <- cnido2_feats2go %>%
  left_join(gene2cog)
head(cnido2_feats2go2cog)
cnido2_feats2go2cog$COG_category[cnido2_feats2go2cog$COG_category == "-"] <- NA

#remove genes with no pfam, reorder by COG value, then replot
cnido2_feats2go2cog2pfam <- cnido2_feats2go2cog %>%
  left_join(gene2pfam)
cnido2_feats2go2cog2pfam <- subset(cnido2_feats2go2cog2pfam, cnido2_feats2go2cog2pfam$PFAMs!="-")

cnido2_feats2go2cog2pfam <- cnido2_feats2go2cog2pfam %>%
  arrange(COG_category) %>% 
  mutate(Preferred_name=factor(Preferred_name, levels=Preferred_name))

cnido2_feats_bar_sort <- ggplot(data=cnido2_feats2go2cog2pfam, aes(y=factor(Preferred_name, levels = rev(levels(factor(Preferred_name)))), x=avg_log2FC, fill=diffexpressed)) +
  geom_bar(position="stack", stat="identity", color="black")+
  scale_fill_manual(values=alpha(c("#CC6600", "gray70"), 1), breaks = c("sym", "apo"))+
  theme_classic()
#### cnido2_feats_barsort_2025 as 4x4
```

#Trajectory analysis
1. Gastrodermal cells and algal-hosting cells only
```{r}
gderm_all <- subset(oculina_named_new, idents = c("Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gastrodermis 4", "Gastrodermis 5", "Algal-Hosting")) 

gderm.cds <- as.cell_data_set(gderm_all)
gderm.cds <- cluster_cells(cds = gderm.cds, reduction_method = "UMAP")
gderm.cds <- learn_graph(gderm.cds, use_partition = FALSE)
gderm_plot_cells <- plot_cells(gderm.cds,
           color_cells_by = "cluster",
           label_groups_by_cluster=FALSE,
           label_leaves=TRUE,
           label_branch_points=TRUE)

gderm_all_sym <- subset(gderm_all, subset = symstate == "sym")
gderm_sym.cds <- as.cell_data_set(gderm_all_sym)
gderm_sym.cds <- cluster_cells(cds = gderm_sym.cds, reduction_method = "UMAP", k=15)
gderm_sym.cds <- learn_graph(gderm_sym.cds, use_partition = FALSE)
gderm_sym_plot_cells <- plot_cells(gderm_sym.cds,
          cell_size = 0.85,
          #alpha = 0.75,
           color_cells_by = "ident",
          group_label_size = 3,
          cell_stroke = 0,
           label_groups_by_cluster=FALSE,
           label_leaves=TRUE,
           label_branch_points=TRUE) + scale_color_manual(values = c("#C7B8E6", "#A28AD2", "#7D5CBF", "#582EAC", "#330099", "#000066")) + ggtitle("") + theme(legend.text=element_text(size=10))
#saved as gderm_sym_trajectory as portrait 5x4

#gderm_sym.cds <- order_cells(gderm_sym.cds, reduction_method = "UMAP")
#plot_cells(
#  cds = gderm_sym.cds,
#  color_cells_by = "pseudotime",
#  show_trajectory_graph = TRUE
#)

gderm_all_apo <- subset(gderm_all, subset = symstate == "apo")
gderm_apo.cds <- as.cell_data_set(gderm_all_apo)
gderm_apo.cds <- cluster_cells(cds = gderm_apo.cds, reduction_method = "UMAP", k=15)
gderm_apo.cds <- learn_graph(gderm_apo.cds, use_partition = FALSE)
gderm_apo_plot_cells <- plot_cells(gderm_apo.cds,
           cell_size = 0.85,
          #alpha = 0.5,
           color_cells_by = "ident",
          group_label_size = 3,
          cell_stroke = 0,
           label_groups_by_cluster=FALSE,
           label_leaves=TRUE,
           label_branch_points=TRUE) + scale_color_manual(values = c("#C7B8E6", "#A28AD2", "#7D5CBF", "#582EAC", "#330099", "#000066")) + ggtitle("") + theme(legend.text=element_text(size=10))
#saved as gderm_apo_trajectory as portrait 5x4
```

# Look at localization of nutrient genes of interest
```{r}
head(oc_full_aposym_mark)
head(oc_DE_05)
head(oc_DE_10)
head(gene)

oc_full_mark_name <- oc_full_aposym_mark %>%
  left_join(gene)
```

1. Candidates for genes involved in glutamine production
- N-acetylglutamate synthase: NAGS, not sig
- Glutamate/Leucine/Phenylalanine/Valine dehydrogenase: May be the equivalent of DHE4, 3 genes are g20747, g20748, and g16789; first two are significantly higher in sym and are markers for gderm1
- glutamate dehydrogenase [NAD(P)+] activity: Gdh, is not significant
- tetrahydrofolylpolyglutamate synthase activity: FPGS, is not significant

--> move forward with g20747 and g20748 as putative glutamate dehydrogenases
- only signficantly differential expression in apo/sym in gderm1

g20747
```{r}
#In whole dataset
#g20747: avg_log2FC is 2.16229 and p_val_adj is 5.707281e-13

g20747_feat_full <- FeaturePlot(oculina_named_new, reduction = "umap", features = c("g20747"), order=TRUE,  split.by = "symstate", min.cutoff = 1.5) + theme(legend.position = c(0.98,0.98))
#g20747_glutdeh_full as 8x4 landscape
g20747_vln_full <- VlnPlot(object =oculina_named_new, features = c("g20747"), pt.size= 0, layer="data", combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#g20747_vln_full 8x3 --> good

#in gderm1
#using slot/layer = data, which is the normalized data ("Feature counts for each cell are divided by the total counts for that cell and multiplied by the scale.factor. This is then natural-log transformed using log1p")
g20747_feat_gd1 <- FeaturePlot(gderm1, reduction = "umap", features = c("g20747"), order=TRUE,  split.by = "symstate") + theme(legend.position = c(0.98,0.98))
#in gderm1_mark log2fc = 2.821509 pval_adj = 1.092582e-14
#g20747_glutdeh_feat_gd1 8x4

g20747_vln1_gderm1 <- VlnPlot(object =gderm1, features = c("g20747"), pt.size= 0, layer="data", combine = F,assay = "RNA", group.by = "symstate", cols=c("gray40", "#CC6600"))
#g20747_vln1_gderm1 as 3x3 portrait

g20747_vln2_gderm1 <- VlnPlot(object =gderm1, features = c("g20747"), pt.size= 0,combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#g20747_vln2_gderm1 as 4x3

g20747_vln3_gderm1 <- VlnPlot(object =gderm1, features = c("g20747"), pt.size= 0,combine = F,assay = "RNA", cols=alpha(c("#CC6699", "#33CC99", "#3333FF", "#FF9900"), 0.5)) 
#g20747_vln3_gderm1 as 4x3

#in gderm2
g20747_feat_gd2 <- FeaturePlot(gderm2, reduction = "umap", features = c("g20747"), order=TRUE,  split.by = "symstate") + theme(legend.position = c(0.98,0.98))
#log2fc = 2.214928 pval_adj = 1
#g20747_glutdeh_feat_gd2 8x4

g20747_vln1_gderm2 <- VlnPlot(object =gderm2, features = c("g20747"), pt.size= 0, layer="data", combine = F,assay = "RNA", group.by = "symstate", cols=c("gray40", "#CC6600"))
#g20747_vln1_gderm2 as 3x3 portrait --> not sig

g20747_vln2_gderm2 <- VlnPlot(object =gderm2, features = c("g20747"), pt.size= 0,combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#g20747_vln2_gderm2 as 4x3

g20747_vln3_gderm2 <- VlnPlot(object =gderm2, features = c("g20747"), pt.size= 0,combine = F,assay = "RNA", cols=alpha(c("#CC6699", "#33CC99", "#3333FF"), 0.5))
#g20747_vln3_gderm2 as 4x3
```

g20748
```{r}
#In whole dataset
#g20748: avg_log2FC is 2.424139 and p_val_adj is 5.126274e-07

g20748_feat_full <- FeaturePlot(oculina_named_new, reduction = "umap", features = c("g20748"), order=TRUE,  split.by = "symstate") + theme(legend.position = c(0.98,0.98))
#g20748_glutdeh_full as 8x4 landscape
g20748_vln_full <- VlnPlot(object =oculina_named_new, features = c("g20748"), pt.size= 0, layer="data", combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#g20748_vln_full 8x3 --> good

#in gderm1
#using slot/layer = data, which is the normalized data ("Feature counts for each cell are divided by the total counts for that cell and multiplied by the scale.factor. This is then natural-log transformed using log1p")
g20748_feat_gd1 <- FeaturePlot(gderm1, reduction = "umap", features = c("g20748"), order=TRUE,  split.by = "symstate") + theme(legend.position = c(0.98,0.98))
#in gderm1_mark log2fc = 3.182468 pval_adj = 1.60915e-09
#g20748_glutdeh_feat_gd1 8x4

g20748_vln1_gderm1 <- VlnPlot(object =gderm1, features = c("g20748"), pt.size= 0, layer="data", combine = F,assay = "RNA", group.by = "symstate", cols=c("gray40", "#CC6600"))
#g20748_vln1_gderm1 as 3x3 portrait

g20748_vln2_gderm1 <- VlnPlot(object =gderm1, features = c("g20748"), pt.size= 0,combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#g20748_vln2_gderm1 as 4x3

g20748_vln3_gderm1 <- VlnPlot(object =gderm1, features = c("g20748"), pt.size= 0,combine = F,assay = "RNA", cols=alpha(c("#CC6699", "#33CC99", "#3333FF", "#FF9900"), 0.5)) 
#g20748_vln3_gderm1 as 4x3

#in gderm2
g20748_feat_gd2 <- FeaturePlot(gderm2, reduction = "umap", features = c("g20748"), order=TRUE,  split.by = "symstate") + theme(legend.position = c(0.98,0.98))
#log2fc = 2.573632 pval_adj = 1
#g20748_glutdeh_feat_gd2 8x4

g20748_vln1_gderm2 <- VlnPlot(object =gderm2, features = c("g20748"), pt.size= 0, layer="data", combine = F,assay = "RNA", group.by = "symstate", cols=c("gray40", "#CC6600"))
#g20748_vln1_gderm2 as 3x3 portrait --> not sig

g20748_vln2_gderm2 <- VlnPlot(object =gderm2, features = c("g20748"), pt.size= 0,combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#g20748_vln2_gderm2 as 4x3

g20748_vln3_gderm2 <- VlnPlot(object =gderm2, features = c("g20748"), pt.size= 0,combine = F,assay = "RNA", cols=alpha(c("#CC6699", "#33CC99", "#3333FF"), 0.5))
#g20748_vln3_gderm2 as 4x3
```

2. Candidates for genes involved in sugar or lipid transport/production
- no apolipophorin, but there is APOB, which is annotated as triglyceride mobilization, higher in sym and marker for gderms1-4 but idk
- several long-chain fatty acid CoA ligase: ABCD2, SLC27A4, SLC27A4.1, SLC27A4.2; ACSBG1 is marker for 1 (gderm1) and 8 and is higher in sym at full dataset and in gderm1
- several acyl-coenzyme A thioesterase: ACOT4.2 is higher in sym and marker for gderm1; and higher in gderm1 sym
- two diacylglycerol O-acyltransferase: not sig and not markers
- one myo-inositol transporter: g19393 but not sig and not marker
- many Solute carrier family 2, facilitated glucose transporter members , one that is for member 8: none significant or markers
- two StAR-related lipid transfer guys: not sig and not markers
- 2-acylglycerol O-acyltransferase: MOGAT3 not sig nor marker
- two acyl-Coenzyme A oxidase: not sig or marker
- three elongation of very long chain fatty acids protein: g10038 is marker for gderm1; not sig diff in whole dataset or in gderm1/2

genes to further explore: ACSBG1 (long-chain fatty acid CoA ligase), ACOT4.2 (acyl-coenzyme A thioesterase)

ACSBG1
```{r}
#In whole dataset
#ACSBG1: avg_log2FC is 0.7859174 and p_val_adj is 0.0009168367

ACSBG1_feat_full <- FeaturePlot(oculina_named_new, reduction = "umap", features = c("ACSBG1"), order=TRUE,  split.by = "symstate", min.cutoff = 1.5) + theme(legend.position = c(0.98,0.98))
#ACSBG1_full as 8x4 landscape
ACSBG1_vln_full <- VlnPlot(object =oculina_named_new, features = c("ACSBG1"), pt.size= 0, layer="data", combine = F,assay = "RNA", split.by = "symstate", log=TRUE, cols=c("gray40", "#CC6600")) #meh not great
#ACSBG1_vln_full

#in gderm1
ACSBG1_feat_gd1 <- FeaturePlot(gderm1, reduction = "umap", features = c("ACSBG1"), order=TRUE,  split.by = "symstate", min.cutoff = 1.5) + theme(legend.position = c(0.98,0.98))
#in gderm1_mark log2fc = 1.917375 pval_adj = 5.863831e-07
#ACSBG1_feat_gd1 8x4
#--> this one is good!

ACSBG1_vln1_gderm1 <- VlnPlot(object =gderm1, features = c("ACSBG1"), pt.size= 0, layer="data", combine = F,assay = "RNA", group.by = "symstate", cols=c("gray40", "#CC6600"))
#ACSBG1_vln1_gderm1 as 3x3 portrait --> less good

ACSBG1_vln2_gderm1 <- VlnPlot(object =gderm1, features = c("ACSBG1"), pt.size= 0,combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#ACSBG1_vln2_gderm1 as 4x3 --> GOOD!

ACSBG1_vln3_gderm1 <- VlnPlot(object =gderm1, features = c("ACSBG1"), pt.size= 0,combine = F,assay = "RNA", cols=alpha(c("#CC6699", "#33CC99", "#3333FF", "#FF9900"), 0.5)) 
#ACSBG1_vln3_gderm1 as 4x3 --> GOOD!

#in gderm2
ACSBG1_feat_gd2 <- FeaturePlot(gderm2, reduction = "umap", features = c("ACSBG1"), order=TRUE,  split.by = "symstate", min.cutoff = 1.5) + theme(legend.position = c(0.98,0.98))
#log2fc = 1.985001 pval_adj = 1
#ACSBG1_feat_gd2 8x4 --> fine

ACSBG1_vln1_gderm2 <- VlnPlot(object =gderm2, features = c("ACSBG1"), pt.size= 0, layer="data", combine = F,assay = "RNA", group.by = "symstate", cols=c("gray40", "#CC6600"))
#ACSBG1_vln1_gderm2 as 3x3 portrait --> not sig

ACSBG1_vln2_gderm2 <- VlnPlot(object =gderm2, features = c("ACSBG1"), pt.size= 0,combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#ACSBG1_vln2_gderm2 as 4x3 --> actually pretty good

ACSBG1_vln3_gderm2 <- VlnPlot(object =gderm2, features = c("ACSBG1"), pt.size= 0,combine = F,assay = "RNA", cols=alpha(c("#CC6699", "#33CC99", "#3333FF"), 0.5))
#ACSBG1_vln3_gderm2 as 4x3 --> good
```

ACOT4.2
```{r}
#In whole dataset
#ACOT4.2: avg_log2FC is 1.628593 and p_val_adj is 0.001908527

ACOT4.2_feat_full <- FeaturePlot(oculina_named_new, reduction = "umap", features = c("ACOT4.2"), order=TRUE,  split.by = "symstate", min.cutoff = 1.5) + theme(legend.position = c(0.98,0.98))
#ACOT4.2_full as 8x4 landscape, good
ACOT4.2_vln_full <- VlnPlot(object =oculina_named_new, features = c("ACOT4.2"), pt.size= 0, layer="data", combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600")) #good!!!
#ACOT4.2_vln_full

#in gderm1
ACOT4.2_feat_gd1 <- FeaturePlot(gderm1, reduction = "umap", features = c("ACOT4.2"), order=TRUE,  split.by = "symstate", min.cutoff = 1.5) + theme(legend.position = c(0.98,0.98))
#in gderm1_mark log2fc = 1.718976 pval_adj = 0.0001523843
#ACOT4.2_feat_gd1 8x4
#--> this one is good!

ACOT4.2_vln1_gderm1 <- VlnPlot(object =gderm1, features = c("ACOT4.2"), pt.size= 0, layer="data", combine = F,assay = "RNA", group.by = "symstate", cols=c("gray40", "#CC6600"))
#ACOT4.2_vln1_gderm1 as 3x3 portrait --> good!!

ACOT4.2_vln2_gderm1 <- VlnPlot(object =gderm1, features = c("ACOT4.2"), pt.size= 0,combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#ACOT4.2_vln2_gderm1 as 4x3 --> GOOD!

ACOT4.2_vln3_gderm1 <- VlnPlot(object =gderm1, features = c("ACOT4.2"), pt.size= 0,combine = F,assay = "RNA", cols=alpha(c("#CC6699", "#33CC99", "#3333FF", "#FF9900"), 0.5)) 
#ACOT4.2_vln3_gderm1 as 4x3 --> GOOD!

#in gderm2
ACOT4.2_feat_gd2 <- FeaturePlot(gderm2, reduction = "umap", features = c("ACOT4.2"), order=TRUE,  split.by = "symstate", min.cutoff = 0) + theme(legend.position = c(0.98,0.98))
#log2fc = 2.628784 pval_adj = 1
#ACOT4.2_feat_gd2 8x4 --> fine

ACOT4.2_vln1_gderm2 <- VlnPlot(object =gderm2, features = c("ACOT4.2"), pt.size= 0, layer="data", combine = F,assay = "RNA", group.by = "symstate", cols=c("gray40", "#CC6600"))
#ACOT4.2_vln1_gderm2 as 3x3 portrait --> not sig

ACOT4.2_vln2_gderm2 <- VlnPlot(object =gderm2, features = c("ACOT4.2"), pt.size= 0,combine = F,assay = "RNA", split.by = "symstate", cols=c("gray40", "#CC6600"))
#ACOT4.2_vln2_gderm2 as 4x3 --> good

ACOT4.2_vln3_gderm2 <- VlnPlot(object =gderm2, features = c("ACOT4.2"), pt.size= 0,combine = F,assay = "RNA", cols=alpha(c("#CC6699", "#33CC99", "#3333FF"), 0.5))
#ACOT4.2_vln3_gderm2 as 4x3 --> good
```

more sugar genes to look at bc they have sugar_tr domains in pfam: 
MFSD12.4 --> good coexpression, marker for 3 but not sig higher in sym in gderm1



3. Coexpression of putative glutamate dehydrogenases (g20747 and g20748), ACSBG1 (long-chain fatty acid CoA ligase), ACOT4.2 (acyl-coenzyme A thioesterase)
(also issues with the blend threshold are here https://github.com/satijalab/seurat/issues/8940 cells just get assigned discrete value)
```{r}
col=c("lightgrey", "#ff0000", "#3333ff")

#gderm1
gderm1_sym <- subset(gderm1, subset=symstate=="sym") 

g20747_g20748_gderm1_sym <- FeaturePlot(object = gderm1_sym, reduction="umap", features = c("g20747", "g20748"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_g20748_gderm1_sym as 12x4
#lots of coexpression in cluster 3

g20747_ACSBG1_gderm1_sym <- FeaturePlot(object = gderm1_sym, reduction="umap", features = c("g20747", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_ACSBG1_gderm1_sym as 12x4
#coexpression in cluster 3

g20747_ACOT4.2_gderm1_sym <- FeaturePlot(object = gderm1_sym, reduction="umap", features = c("g20747", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_ACOT4.2_gderm1_sym as 12x4
#coexpression in cluster 3

g20748_ACSBG1_gderm1_sym <- FeaturePlot(object = gderm1_sym, reduction="umap", features = c("g20748", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20748_ACSBG1_gderm1_sym as 12x4
#coexpression in cluster 3

g20748_ACOT4.2_gderm1_sym <- FeaturePlot(object = gderm1_sym, reduction="umap", features = c("g20748", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20748_ACOT4.2_gderm1_sym as 12x4
#coexpression in cluster 3

ACSBG1_ACOT4.2_gderm1_sym <- FeaturePlot(object = gderm1_sym, reduction="umap", features = c("ACSBG1", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#ACSBG1_ACOT4.2_gderm1_sym as 12x4
#coexpression in cluster 3

#gderm2
gderm2_sym <- subset(gderm2, subset=symstate=="sym") 

g20747_g20748_gderm2_sym <- FeaturePlot(object = gderm2_sym, reduction="umap", features = c("g20747", "g20748"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_g20748_gderm2_sym as 12x4

g20747_ACSBG1_gderm2_sym <- FeaturePlot(object = gderm2_sym, reduction="umap", features = c("g20747", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_ACSBG1_gderm2_sym as 12x4

g20747_ACOT4.2_gderm2_sym <- FeaturePlot(object = gderm2_sym, reduction="umap", features = c("g20747", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_ACOT4.2_gderm2_sym as 12x4

g20748_ACSBG1_gderm2_sym <- FeaturePlot(object = gderm2_sym, reduction="umap", features = c("g20748", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20748_ACSBG1_gderm2_sym as 12x4

g20748_ACOT4.2_gderm2_sym <- FeaturePlot(object = gderm2_sym, reduction="umap", features = c("g20748", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20748_ACOT4.2_gderm2_sym as 12x4

ACSBG1_ACOT4.2_gderm2_sym <- FeaturePlot(object = gderm2_sym, reduction="umap", features = c("ACSBG1", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#ACSBG1_ACOT4.2_gderm2_sym as 12x4
```

agh
```{r}
agh_sym <- subset(agh, subset=symstate=="sym") 

g20747_g20748_agh_sym <- FeaturePlot(object = agh_sym, reduction="umap", features = c("g20747", "g20748"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)

g20747_ACSBG1_agh_sym <- FeaturePlot(object = agh_sym, reduction="umap", features = c("g20747", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)

g20747_ACOT4.2_agh_sym <- FeaturePlot(object = agh_sym, reduction="umap", features = c("g20747", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)

g20748_ACSBG1_agh_sym <- FeaturePlot(object = agh_sym, reduction="umap", features = c("g20748", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)

g20748_ACOT4.2_agh_sym <- FeaturePlot(object = agh_sym, reduction="umap", features = c("g20748", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)

ACSBG1_ACOT4.2_agh_sym <- FeaturePlot(object = agh_sym, reduction="umap", features = c("ACSBG1", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)

```


checking the coexpression of these genes in other cell clusters
```{r}
epi1_sym <- subset(epi1, subset=symstate=="sym") 

g20747_g20748_epi1_sym <- FeaturePlot(object = epi1_sym, reduction="umap", features = c("g20747", "g20748"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_g20748_epi1_sym as 12x4

g20747_ACSBG1_epi1_sym <- FeaturePlot(object = epi1_sym, reduction="umap", features = c("g20747", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_ACSBG1_epi1_sym as 12x4

g20747_ACOT4.2_epi1_sym <- FeaturePlot(object = epi1_sym, reduction="umap", features = c("g20747", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_ACOT4.2_epi1_sym as 12x4

g20748_ACSBG1_epi1_sym <- FeaturePlot(object = epi1_sym, reduction="umap", features = c("g20748", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20748_ACSBG1_epi1_sym as 12x4

g20748_ACOT4.2_epi1_sym <- FeaturePlot(object = epi1_sym, reduction="umap", features = c("g20748", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20748_ACOT4.2_epi1_sym as 12x4

ACSBG1_ACOT4.2_epi1_sym <- FeaturePlot(object = epi1_sym, reduction="umap", features = c("ACSBG1", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#ACSBG1_ACOT4.2_epi1_sym as 12x4

#n1
n1_sym <- subset(n1, subset=symstate=="sym") 

g20747_g20748_n1_sym <- FeaturePlot(object = n1_sym, reduction="umap", features = c("g20747", "g20748"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_g20748_n1_sym as 12x4

g20747_ACSBG1_n1_sym <- FeaturePlot(object = n1_sym, reduction="umap", features = c("g20747", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_ACSBG1_n1_sym as 12x4

g20747_ACOT4.2_n1_sym <- FeaturePlot(object = n1_sym, reduction="umap", features = c("g20747", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20747_ACOT4.2_n1_sym as 12x4

g20748_ACSBG1_n1_sym <- FeaturePlot(object = n1_sym, reduction="umap", features = c("g20748", "ACSBG1"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20748_ACSBG1_n1_sym as 12x4

g20748_ACOT4.2_n1_sym <- FeaturePlot(object = n1_sym, reduction="umap", features = c("g20748", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#g20748_ACOT4.2_n1_sym as 12x4

ACSBG1_ACOT4.2_n1_sym <- FeaturePlot(object = n1_sym, reduction="umap", features = c("ACSBG1", "ACOT4.2"), blend = TRUE, keep.scale = "feature", blend.threshold = 0.1, cols=c("lightgrey", "#ff0000", "#3333ff"), order=TRUE)
#ACSBG1_ACOT4.2_n1_sym as 12x4
```

1. reading in the xenia rds data and the ortholog data from broccoli step 3 (let's look at 	table_OGs_protein_names.txt)
```{r}
xenia_nonregen_int <- readRDS(file.choose())
head(xenia_nonregen_int@assays)

orthos_table <- read.table(file.choose(), header = FALSE, sep = "\t")
head(orthos_table) #v6 is oculina, v12 is xenia
orthos_ocuxen <- orthos_table[,c(1,6,12)]
orthos_ocuxen <- orthos_ocuxen[!(orthos_ocuxen$V6==""),]
orthos_ocuxen <- orthos_ocuxen[!(orthos_ocuxen$V12==""),]
#8529 orthos
```

2. create a list of all the gene names from the Broccoli orthos
```{r}
orthos_ocu <- orthos_ocuxen[,c(1,2)]
orthos_ocu <- orthos_ocu %>%
  mutate(V6 = gsub(".t1", "", V6)) %>%
  mutate(V6 = gsub(".t2", "", V6)) %>%
  mutate(V6 = gsub(".t3", "", V6)) %>%
  mutate(V6 = gsub(".t4", "", V6)) %>%
  mutate(V6 = gsub(".t5", "", V6)) %>%
  mutate(V6 = gsub(".t6", "", V6)) %>%
  mutate(V6 = gsub(".t7", "", V6)) %>%
  mutate(V6 = gsub(".t8", "", V6)) %>%
  mutate(V6 = gsub(".t9", "", V6))

orthos_ocu <- orthos_ocu %>% 
  mutate(query=strsplit(V6, " ")) %>% 
  unnest(query) %>%
  left_join(gene2gene_symbol) %>%
  filter(!duplicated(query))

xenia_id2sym <- read.delim(file="./files/xenia_id2symbol.txt")
head(xenia_id2sym)
colnames(xenia_id2sym) <- paste(c("Protein_ID", "gene_symbol"))

orthos_xen <- orthos_ocuxen[,c(1,3)]
orthos_xen <- orthos_xen %>%
  mutate(V12 = gsub("-T1", "", V12)) %>%
  mutate(V12 = gsub("-T2", "", V12)) %>%
  mutate(V12 = gsub("-T3", "", V12)) %>%
  mutate(V12 = gsub("-T4", "", V12)) %>%
  mutate(V12 = gsub("-T5", "", V12)) %>%
  mutate(V12 = gsub("-T6", "", V12))

orthos_xen <- orthos_xen %>% 
  mutate(Protein_ID=strsplit(V12, " ")) %>% 
  unnest(Protein_ID) %>%
  left_join(xenia_id2sym) %>%
  mutate(Protein_ID = gsub("_", "-", Protein_ID)) %>%
  mutate(gene_symbol = gsub("_", "-", gene_symbol))
orthos_xen <- orthos_xen %>%
  filter(!duplicated(Protein_ID))
#removing the row that contains "Rieske [2Fe-2S] domain" because the weird character is giving downstream issues
orthos_xen <- orthos_xen[!(orthos_xen$Protein_ID=="Xe-008331"),]
```

3. prep your two seurat objects
```{r}
head(oculina_sym)

#update xenia
xenia_nonregen_int <- UpdateSeuratObject(object = xenia_nonregen_int)
#colnames(xenia_nonregen_int)<-Cells(xenia_nonregen_int[['integrated_snn']])
#all(Cells(Data@graphs$integrated_snn)==colnames(Data))
#Data <- Seurat::UpdateSeuratObject(object = Data)
#xenia_16 <- subset(xenia_nonregen_int, subset = seurat_clusters == 16)

#set default assay to RNA
DefaultAssay(oculina_sym) <- "RNA"
DefaultAssay(xenia_nonregen_int) <- "RNA"

# create a list of all the gene names
xeniagenes<-as.vector(unlist(orthos_xen$gene_symbol))
oculinagenes<-as.vector(unlist(orthos_ocu$Preferred_name))
```

4. 
```{r}
xenia <- NormalizeData(xenia_nonregen_int)
xenia <- FindVariableFeatures(xenia)
xenia<- ScaleData(xenia)
xenia <- RunPCA(xenia)
xenia <- FindNeighbors(xenia, dims = 1:30, reduction = "pca")
xenia <- FindClusters(xenia, resolution = 0.18)
#16 clusters at res of 0.18
xenia <- RunUMAP(xenia, dims = 1:30)
####
DefaultAssay(xenia) <- "RNA"
xenia@meta.data$seurat_clusters <- as.numeric(as.character(xenia@meta.data$seurat_clusters))+1
Idents(xenia)<- "seurat_clusters"
xenia$seurat_clusters
xenia@active.ident <- factor(x = xenia@active.ident, levels = 1:16)
xenia@active.ident
plotx <- DimPlot(xenia, reduction = "umap", label = T, alpha=0.5)

rnaseq_xenia<- FindAllMarkers(xenia, 
                          min.pct = -Inf, 
                          logfc.threshold = .5, 
                          only.pos = T, 
                          test.use = "MAST",
                          latent.vars = c('percent.mt', 'nFeature_RNA'))

DefaultAssay(xenia) <- "RNA"
xenia[["RNA"]]$counts<-as.matrix(xenia[["RNA"]]$counts)+1

x_algal_gderm3_mark <- FindMarkers(xenia, ident.1 = "15", ident.2 = c("5", "6"), test.use="DESeq2", slot = "counts")
x_algal_gderm3_mark <- rownames_to_column(x_algal_gderm3_mark, var="gene_symbol")
x_algal_gderm3_05 = subset(x_algal_gderm3_mark, p_val_adj < 0.05)
nrow(x_algal_gderm3_05)
#298
```

Get DEGs from oculina_named_new to compare apo vs sym in gderm1-4+agh (not gderm5 bc on a different trajectory in both apo and sym)
```{r}
gderm1234a <- subset(oculina_named_new, idents = c("Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gastrodermis 4", "Algal-Hosting"))

DefaultAssay(gderm1234a) <- "RNA"
gderm1234a[["RNA"]]$counts<-as.matrix(gderm1234a[["RNA"]]$counts)+1

gd1234a_mark <- FindMarkers(gderm1234a, ident.1 = "sym", ident.2="apo", group.by = "symstate", test.use="DESeq2", slot = "counts")
gd1234a_mark <- rownames_to_column(gd1234a_mark, var="Preferred_name")
head(gd1234a_mark)
gd1234a_05 = subset(gd1234a_mark, p_val_adj < 0.05)
nrow(gd1234a_05)
#343
```

combo the DEGs with their orthos
```{r}
head(orthos_xen)
head(orthos_ocu)

orthos_xen_05 <- x_algal_gderm3_05 %>%
  inner_join(orthos_xen) %>%
  select(gene_symbol, V1, avg_log2FC)
nrow(orthos_xen_05)
#238
colnames(orthos_xen_05) <- paste(c("gene_symbol_xenia", "V1", "log2FC_xenia"))
length(unique(orthos_xen_05$V1))

orthos_gd1234a_05 <- gd1234a_05 %>%
  inner_join(orthos_ocu) %>%
  select(Preferred_name, V1, avg_log2FC)
nrow(orthos_gd1234a_05)
#260
colnames(orthos_gd1234a_05) <- paste(c("gene_symbol_oculina", "V1", "log2FC_oculina"))
length(unique(orthos_gd1234a_05$V1))
```

venn
```{r}
library("VennDiagram")

candidates_xen_gd1234a=list("Xenia"=orthos_xen_05$V1, "Oculina"=orthos_gd1234a_05$V1)

prettyvenn3=venn.diagram(
  x = candidates_xen_gd1234a,
  category.names=c("Xenia", "Oculina"), #this is the number of unique orthos
  col=c("#990000","#000099"),
  lwd=c(3,3),
  filename=NULL,
  #main = "",
  main.cex = 2,
  main.fontface = "bold",
  main.fontfamily = "sans",
  #col = "transparent",
  fill = c("#990000","#000099"),
  alpha = 0.1,
  cex = 2,
  fontfamily = "sans",
  fontface = "bold",
  cat.default.pos = "text",
  cat.col = c("black", "black"),
  cat.cex = 1.5,
  cat.fontfamily = "sans",
  cat.dist = c(0.25, 0.25),
  cat.pos = c(-15, 5)
)

grid.draw(prettyvenn3)
#venn_xen_gd1234a 4x4
```

Do Fishers for shared and unique orthos
using the oculina genes
```{r}
intersect_xen_gd1234a <- orthos_xen_05 %>%
  inner_join(orthos_gd1234a_05)
gd_shared <- as.data.frame(unique(intersect_xen_gd1234a$gene_symbol_oculina))
colnames(gd_shared) <- paste(c("Preferred_name"))
gd_shared <- gd_shared %>%
  mutate(Fishers = 1) %>%
  full_join(gene2gene_symbol) %>%
  mutate(Fishers = if_else(is.na(Fishers), 0, 1)) %>%
  select(Preferred_name, Fishers)
write.csv(gd_shared, file="./files/gd1234a_shared_GO.csv", quote=F, row.names=FALSE)
```

using the xenia genes
```{r}
#xenia genes of the shared bit
xenia_id2sym <- read.delim(file="./files/xenia_id2symbol.txt")
head(xenia_id2sym)
colnames(xenia_id2sym) <- paste(c("Protein_ID", "gene_symbol"))
head(xenia_id2sym)
xenia_id2sym2 <- xenia_id2sym %>%
  mutate(gene_symbol = gsub("_", "-", gene_symbol))

gd1234aXEN_shared <- as.data.frame(unique(intersect_xen_gd1234a$gene_symbol_xenia))
colnames(gd1234aXEN_shared) <- paste(c("gene_symbol"))
gd1234aXEN_shared <- gd1234aXEN_shared %>%
  mutate(Fishers = 1) %>%
  full_join(xenia_id2sym2) %>%
  mutate(Fishers = if_else(is.na(Fishers), 0, 1)) %>%
  select(Protein_ID, Fishers)
write.csv(gd1234aXEN_shared, file="./files/gd1234aXEN_shared_GO.csv", quote=F, row.names=FALSE)
```

```{r}
head(orthos_xen_05)
head(orthos_gd1234a_05)

orthos_xen_05_Fish <- as.data.frame(orthos_xen_05) %>%
  mutate(Fishers_xenia = 1)%>%
  dplyr::select(gene_symbol_xenia, Fishers_xenia, V1)

orthos_gd1234a_05_Fish <- as.data.frame(orthos_gd1234a_05) %>%
  mutate(Fishers_oculina = 1)%>%
  dplyr::select(gene_symbol_oculina, Fishers_oculina, V1)

fj_gd <- full_join(orthos_xen_05_Fish, orthos_gd1234a_05_Fish)
tail(fj_gd)

fj_gd1234a <- fj_gd %>%
  drop_na(gene_symbol_oculina)
fj_gd1234a_only <- subset(fj_gd1234a, !(gene_symbol_oculina %in% intersect_xen_gd1234a$gene_symbol_oculina)) %>%
  select(-gene_symbol_xenia, -Fishers_xenia, -V1)
#fj_ocsym_only <- fj_ocsym[!intersect_xen_ocsym$gene_symbol_oculina,] 
colnames(fj_gd1234a_only)[1] <- paste(c("Preferred_name"))

fj_gd1234a_only <- fj_gd1234a_only %>%
  full_join(gene2gene_symbol) %>%
  mutate(Fishers_oculina = if_else(is.na(Fishers_oculina), 0, 1)) %>%
  select(Preferred_name, Fishers_oculina)
write.csv(fj_gd1234a_only, file="./files/fj_gd1234a_ONLY_GO.csv", quote=F, row.names=FALSE)


fj_gd_xen <- fj_gd %>%
  drop_na(gene_symbol_xenia)
fj_gd_xen_only <- subset(fj_gd_xen, !(gene_symbol_xenia %in% intersect_xen_gd1234a$gene_symbol_xenia)) %>%
  select(-gene_symbol_oculina, -Fishers_oculina, -V1)
colnames(fj_gd_xen_only)[1] <- paste(c("gene_symbol"))

#head(xenia_id2sym)
#xenia_id2sym2 <- xenia_id2sym %>%
#  mutate(gene_symbol = gsub("_", "-", gene_symbol))

fj_gd_xen_only <- fj_gd_xen_only %>%
  full_join(xenia_id2sym2) %>%
  mutate(Fishers_xenia = if_else(is.na(Fishers_xenia), 0, 1)) %>%
  select(Protein_ID, Fishers_xenia)
write.csv(fj_gd_xen_only, file="./files/fj_gd_xen_ONLY_GO.csv", quote=F, row.names=FALSE)
```

Reformat GO output tables
```{r}
go_bp_sharedorthos <- read.csv("./files/MWU_BP_gd1234aXEN_shared_GO.csv", header=TRUE, sep=" ")
head(go_bp_sharedorthos)
write.table(go_bp_sharedorthos, "./files/MWU_BP_gd1234aXEN_shared_GO.txt", quote=F, sep="\t", row.names=FALSE)

go_bp_xenonly <- read.csv("./files/MWU_BP_fj_gd_xen_ONLY_GO.csv", header=TRUE, sep=" ")
head(go_bp_xenonly)
write.table(go_bp_xenonly, "./files/MWU_BP_fj_gd_xen_ONLY_GO.txt", quote=F, sep="\t", row.names=FALSE)

go_bp_gd1234aonly <- read.csv("./files/MWU_BP_fj_gd1234a_ONLY_GO.csv", header=TRUE, sep=" ")
head(go_bp_gd1234aonly)
write.table(go_bp_gd1234aonly, "./files/MWU_BP_fj_gd1234a_ONLY_GO.txt", quote=F, sep="\t", row.names=FALSE)
```



